<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2024-09-11" />

<title>cross-tissue DE analysis with fastTopics</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cross-tissue DE analysis with
fastTopics</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2024-09-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree163b42241dd04fd7c63504ec158bef8fae47d9c5targetblank163b422a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/163b42241dd04fd7c63504ec158bef8fae47d9c5" target="_blank">163b422</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree163b42241dd04fd7c63504ec158bef8fae47d9c5targetblank163b422a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/163b42241dd04fd7c63504ec158bef8fae47d9c5" target="_blank">163b422</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/test_magma.Rmd
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  code/run_magma/
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer/
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/positions.bed
    Untracked:  output/topic1/
    Untracked:  output/topic10/
    Untracked:  output/topic11/
    Untracked:  output/topic12/
    Untracked:  output/topic2/
    Untracked:  output/topic3/
    Untracked:  output/topic4/
    Untracked:  output/topic5/
    Untracked:  output/topic6/
    Untracked:  output/topic7/
    Untracked:  output/topic8/
    Untracked:  output/topic9/
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/cross_tissue_DE_u19_fastTopics.Rmd</code>) and HTML
(<code>docs/cross_tissue_DE_u19_fastTopics.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/163b42241dd04fd7c63504ec158bef8fae47d9c5/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">163b422</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-09-11
</td>
<td>
plotted heatmaps for enrichment results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/87c183bf15d834e52b1dac9c0dea02aee9496512/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">87c183b</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b7a8317f32ca76707a4f8e94d6329386c0f3b25b/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">b7a8317</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-25
</td>
<td>
updated result page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/4e0028dbbf61f4137f14f1e79a68e400bbfaed25/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">4e0028d</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/36aceab433e16b43c378bbc524e5f77699a48ee3/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">36aceab</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-23
</td>
<td>
organized results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/744d4f03155ef8face840a3392dbcd3d301ae64e/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">744d4f0</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/5c6b22f9accd707a1cba6fb39fc949ed504fe061/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">5c6b22f</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-22
</td>
<td>
updated GSEA results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/3d732525a5a3f44cfb65f6cc59a0e8a7aba1b1cf/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">3d73252</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/f26e8ad0eb30e310b0d00b20c7543f85a70f954f/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">f26e8ad</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
wflow_publish("analysis/cross_tissue_DE_u19_fastTopics.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/95bfdffcc0c78076c819eeaf141f1ddb96e56ab4/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">95bfdff</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/3e4f0154bf223596e8db26a44b2942a7bb9a99b9/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">3e4f015</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
wflow_publish("analysis/cross_tissue_DE_u19_fastTopics.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/2e475c99feb5c4f9f25e5ad8af0c20424fea954c/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">2e475c9</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/306ddb1d7705b9482cd8cec26092c1bc6621c16c/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">306ddb1</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
<td>
wflow_publish("analysis/cross_tissue_DE_u19_fastTopics.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/f812a40863cae53d9ee687fd66c34ad72e519b12/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">f812a40</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/cross_tissue_DE_u19_fastTopics.html" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/7ca45f65c382b0a20a2b14e62730ecd79f9140f9/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">7ca45f6</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
<td>
cross-tissue comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/7a452619bca02571aaaa2e8b1411ade96a41642c/analysis/cross_tissue_DE_u19_fastTopics.Rmd" target="_blank">7a45261</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-08
</td>
<td>
cross-tissue comparison with topic modeling
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="gom-de-analysis-on-u19-dataset" class="section level1">
<h1>GoM DE analysis on u19 dataset</h1>
<div id="model-fitting" class="section level2">
<h2>Model fitting</h2>
<p>Parameters:</p>
<p>N_updates = 150 N_topics = 12 ## Model evaluation</p>
<p>check the convergence
<img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-2-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Model overview:
  Number of data rows, n: 53647
  Number of data cols, m: 17420
  Rank/number of topics, k: 12
Evaluation of model fit (170 updates performed):
  Poisson NMF log-likelihood: -1.997995557900e+08
  Multinomial topic model log-likelihood: -1.995509032083e+08
  Poisson NMF deviance: +2.634951598369e+08
  Max KKT residual: +1.430262e-02</code></pre>
<pre><code>Set show.size.factors = TRUE, show.mixprops = TRUE and/or show.topic.reps = TRUE in print(...) for more information</code></pre>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-3-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-4-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualize-topics-with-structural-plots" class="section level2">
<h2>Visualize topics with structural plots</h2>
<ul>
<li><p>plot by tissue
<img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
<p>Past versions of unnamed-chunk-5-1.png</p>
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-5-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div></li>
<li><p>plot by tissue and cell-type</p></li>
</ul>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-6-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="characterize-topics-with-enrichment-analysis"
class="section level2">
<h2>Characterize topics with enrichment analysis</h2>
<p>Two ways to perform enrichment test:</p>
<ul>
<li>hypergeometric test (ORA from <a
href="https://cran.r-project.org/web/packages/WebGestaltR/WebGestaltR.pdf">WebGestaltR</a>)</li>
</ul>
<p>A test purely depends on occurrences, asking whether the genes
related to a topic occur more frequently in GO term compared to the
background genes.</p>
<ul>
<li>two-sample T test or regression (<a
href="https://cloufield.github.io/GWASTutorial/09_Gene_based_analysis/#annotate-snps">MAGMA</a>)</li>
</ul>
<p>It tests whether the genes in a set is more associated with
<strong>phenotype</strong> than those outside of a set.</p>
<div id="go-enrichment-results-ora-method" class="section level3">
<h3>GO enrichment results (ORA method)</h3>
<div id="top-genes-ranked-by-loadings" class="section level4">
<h4>Top genes ranked by loadings</h4>
<p>Input:</p>
<ul>
<li>Top 500 loading genes vs. all genes</li>
<li>Database: Biological_Process, Cellular_Component,
Molecular_Function</li>
</ul>
<p>Top loading genes are enriched in large number of GO terms, which
have broad functions.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-190858cf5ffe3166100d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-190858cf5ffe3166100d">{"x":{"filter":"none","vertical":false,"caption":"<caption>GO term counts<\/caption>","data":[["1"],[383],[183],[205],[110],[260],[253],[142],[244],[123],[97],[197],[134]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k1","targets":1},{"name":"k2","targets":2},{"name":"k3","targets":3},{"name":"k4","targets":4},{"name":"k5","targets":5},{"name":"k6","targets":6},{"name":"k7","targets":7},{"name":"k8","targets":8},{"name":"k9","targets":9},{"name":"k10","targets":10},{"name":"k11","targets":11},{"name":"k12","targets":12}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="topic-specific-genes-through-de-analysis"
class="section level4">
<h4>Topic-specific genes through DE analysis</h4>
<p>Input:</p>
<ul>
<li>Up-regulated genes specific to each topic (z &gt; 0 and lfsr &lt;
0.01) vs. all genes</li>
<li>Database: Biological_Process, Cellular_Component,
Molecular_Function</li>
</ul>
<p><strong>Volcano plots for GoM DE results</strong></p>
<p>Axis: the z-scores for posterior mean log-fold change estimates
vs. log-fold change</p>
<p>From volcano plots, we see several genes that encode different types
of cytokines are present in topic 6. Chemokine ligands (CCL4, CCL20,
etc.) are proteins that signal leukocyte migration, while cytokines
(IL17A, IL22) are interleukins that signal immune cells to defend
against pathogens.</p>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-8-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Number of up-regulated genes in each topic against the rest:&quot;</code></pre>
<pre><code>  t1   t2   t3   t4   t5   t6   t7   t8   t9  t10  t11  t12 
1602  122  594  562  391  381  267  505  111  626  465  489 </code></pre>
<p><strong>GO Enrichment result table</strong></p>
<p>All topics have fewer number of GO terms with enrichment, except for
topic 1.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8481a56e9d46967c7c27" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8481a56e9d46967c7c27">{"x":{"filter":"none","vertical":false,"caption":"<caption>GO term counts<\/caption>","data":[["1"],[428],[46],[26],[7],[85],[83],[40],[77],[8],[31],[7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k1","targets":1},{"name":"k2","targets":2},{"name":"k3","targets":3},{"name":"k4","targets":4},{"name":"k5","targets":5},{"name":"k6","targets":6},{"name":"k7","targets":7},{"name":"k8","targets":8},{"name":"k9","targets":9},{"name":"k11","targets":10},{"name":"k12","targets":11}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><strong>Visualizing GO enrichments with ComplexHeatmap</strong></p>
<p>Parameters: - -log10(FDR) as value input - No clustering due to
missing data - top 10 GO terms shown for each topic</p>
<p>Legend: - Color scale capped by FDR = <span
class="math inline">\(10^{-15}\)</span></p>
<p>BP results show k1 enriched for granulocyte activation and neutrophil
mediated immunity, as well as cell adhesion and motility. These pathways
are highly relevant to Asthma. Due to high number of GO terms
over-represented by k1 genes, we may repeat the topic modeling by
increasing topic number. Several topics like k2, 6, 7, 9, 10 are
strongly enriched with protein localization. Topics k3-6 have genes
over-represented in T-cell activation, while k10, 12 in B-cell
activation.</p>
<p>MF results show broad enrichment of molecular binding, with k7 highly
enriched for cell adhesion molecular binding.</p>
<pre><code>Selecting by FDR</code></pre>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/2e475c99feb5c4f9f25e5ad8af0c20424fea954c/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-11-1.png" target="_blank">2e475c9</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/68d9e18eb743d0da900d1db89f4c0cf4f6d68531/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-11-1.png" target="_blank">68d9e18</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="identify-topics-correlated-with-tissue-difference"
class="section level2">
<h2>Identify topics correlated with tissue difference</h2>
<ol style="list-style-type: decimal">
<li>We can test whether topic proportion is correlated with the tissue
of origin <span class="math display">\[
F  = \beta X_{\text{tissue}} + \text{Covariates} + \epsilon
\]</span></li>
<li>perform T-test to see whether topic proportions between two tissues
are significantly different</li>
</ol>
<div id="barplot-for-topic-proportions-in-cell-types"
class="section level3">
<h3>Barplot for topic proportions in cell types</h3>
<ul>
<li>The topic proportion between tissue are similar across T cell
subsets.</li>
<li><span class="math inline">\(CD8^+\)</span> T cells have the highest
proportion of k3 compared to others, similar as NK cells.</li>
<li>The topic k1 dominates lung monocytes, and this topic separates ILC3
cells from other T/NK cells in lungs.</li>
<li>The topic proportion that differs the most between tissue occurs in
B cells(eg. k4, k10, k12)</li>
</ul>
<pre><code>         
          Naive_B Memory_B Monocytes  ILC3 CD4_T CD8_T  Treg  Th17    NK Other
  lungs      1174     5287      1010   499  6980 12210  1336  2732  8067   145
  spleens    1710    10507        22    52   886   421    47    68   464    30</code></pre>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-12-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="perform-t-test-while-adjusting-for-confounders"
class="section level2">
<h2>Perform t-test while adjusting for confounders</h2>
<div id="procedure" class="section level3">
<h3>Procedure</h3>
<p>Test mean difference between tissue one donor at a time and then do
meta-analysis with Fisher’s method</p>
</div>
<div id="results" class="section level3">
<h3>Results</h3>
<p>X-axis denotes cell types and y-axis denotes the topics. For major
cell types, we saw majority of topics have significant differences in
proportions between tissue.</p>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/87c183bf15d834e52b1dac9c0dea02aee9496512/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-14-1.png" target="_blank">87c183b</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/95bfdffcc0c78076c819eeaf141f1ddb96e56ab4/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-14-1.png" target="_blank">95bfdff</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/2e475c99feb5c4f9f25e5ad8af0c20424fea954c/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-14-1.png" target="_blank">2e475c9</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="find-evidence-for-the-function-of-tfs-with-motif-enrichment-in-genes-loaded-on-topic-5"
class="section level3">
<h3>Find evidence for the function of TFs with motif enrichment in genes
loaded on topic 5</h3>
<pre><code>Warning: The dot-dot notation (`..count..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(count)` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/744d4f03155ef8face840a3392dbcd3d301ae64e/docs/figure/cross_tissue_DE_u19_fastTopics.Rmd/unnamed-chunk-15-1.png" target="_blank">744d4f0</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-05-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a9b9e8f1dbaee5155a90" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a9b9e8f1dbaee5155a90">{"x":{"filter":"none","vertical":false,"data":[["BICDL1","KIF5C","IL21R","RORA","ABTB2","GATA3","TTC39C","RALGAPA1","USP20","STK24","CD6","CTLA4","DNMBP","GPRIN3","ATP13A3","PLEKHG2","ZC3HAV1","ZEB1","CD226","FAM168B","GCN1","POFUT2","RELA","FNBP1","BAZ1A","ARID3B","TMEM181","LPIN1","CAPZA1","TNIP2","LRRC8C","ARID4B","GRAMD1B","ZNF267","FAM133B","AGO2","FGFR1","ARIH1","NCK2","SYTL3","CHD2","ARID5A","SLAMF1","KPNA6","SREBF2","FAM126B","SNX9","CDK6","CAST","RNF4","MAPKAPK2","RUNX3","LRRFIP2","CASP8","HERC1","ZFP36","PSME4","SURF4","NCOA3","SUPT5H","SRP54","MAP3K2","COPB1","CDC14A","YWHAZ","INTS6","RTCA","OTULIN","TP53BP2","USP9X","ZFYVE28","ZFP36L2","SIK3","MARK3","PSMA1","RB1CC1","BIRC2","ELK3","WNK1","RAB7A","EZR","HSP90AA1","SNRK","ZNF644","SPPL2A","IL32","LATS2","SMCHD1","SSH2","P2RY8","KAT6A","ARHGAP15","FMNL1","NFKBIA","DENND4A","CBLB","TANK","DDX3X","SMG1","TXNDC11","JAK1","STK17A","ZNF292","ITK","PLEKHB2","EP300","CTNNB1","NIPBL"],[" 38.2"," 18.4"," 13.9"," 48.8"," 12.0"," 15.8"," 15.2"," 27.0","  9.7"," 17.3"," 20.0","  6.7","  7.0"," 19.4"," 19.7"," 10.9"," 32.2"," 22.0"," 13.4"," 11.5","  5.8","  3.9","  3.7"," 46.8"," 11.3","  6.9","  3.5","  7.0"," 17.7","  8.6"," 12.4"," 12.6","  6.4","  4.4"," 18.9"," 21.6","  7.8","  9.9","  7.8"," 31.1"," 20.3","  4.7","  7.8","  8.5"," 14.7"," 10.3"," 20.8","  7.0","  6.9","  7.3"," 12.1"," 10.6","  6.1"," 11.3","  8.5","  7.0"," 23.6","  5.8","  6.2","  4.1","  5.7","  7.3","  5.0"," 14.6","  3.8","  7.1","  4.5"," 30.0","  7.5","  4.5","  8.5"," 16.3"," 14.7","  4.9","  3.3","  9.7","  3.1","  3.2"," 11.5"," 17.8"," 11.9","  6.8","  6.5","  4.8","  7.1","  6.3"," 29.5"," 19.9","  4.1","  4.4","  6.3","  4.1","  3.7","  5.4"," 20.5"," 18.5"," 10.9","  4.0","  4.5","  4.8","  5.3","  4.3","  3.6","  3.5","  5.2"," 11.1"," 15.4"," 10.8"],["8.3e-317"," 6.7e-73"," 1.5e-41"," 0.0e+00"," 8.1e-31"," 1.3e-53"," 1.4e-49","9.3e-158"," 3.7e-20"," 8.0e-65"," 2.2e-86"," 7.2e-10"," 7.3e-11"," 6.8e-82"," 3.1e-84"," 5.6e-26","1.2e-224","6.4e-105"," 1.1e-38"," 1.6e-28"," 1.5e-07"," 7.8e-04"," 1.6e-03"," 0.0e+00"," 6.8e-28"," 2.5e-10"," 2.8e-03"," 8.6e-11"," 4.2e-68"," 4.9e-16"," 2.7e-33"," 3.2e-34"," 3.9e-09"," 1.9e-04"," 6.9e-77","1.5e-100"," 3.3e-13"," 3.5e-21"," 3.7e-13","1.5e-209"," 2.1e-89"," 4.3e-05"," 3.4e-13"," 1.5e-15"," 2.5e-46"," 5.6e-23"," 3.5e-93"," 1.1e-10"," 2.0e-10"," 1.5e-11"," 1.6e-31"," 4.3e-24"," 4.1e-08"," 1.9e-27"," 1.7e-15"," 1.4e-10","8.9e-121"," 1.9e-07"," 2.1e-08"," 6.4e-04"," 4.5e-07"," 1.6e-11"," 1.3e-05"," 1.0e-45"," 1.5e-03"," 5.3e-11"," 1.1e-04","3.3e-195"," 3.6e-12"," 1.1e-04"," 1.4e-15"," 3.5e-57"," 2.4e-46"," 1.8e-05"," 4.8e-03"," 4.0e-20"," 8.6e-03"," 6.6e-03"," 1.2e-28"," 2.2e-68"," 1.0e-30"," 3.7e-10"," 2.5e-09"," 2.9e-05"," 7.4e-11"," 8.2e-09","2.4e-188"," 9.0e-86"," 8.1e-04"," 2.5e-04"," 2.2e-08"," 8.9e-04"," 2.9e-03"," 3.3e-06"," 1.2e-90"," 2.6e-73"," 3.7e-25"," 1.6e-03"," 3.9e-04"," 1.4e-04"," 9.9e-06"," 7.3e-04"," 6.0e-03"," 6.4e-03"," 2.6e-05"," 8.9e-26"," 3.2e-50"," 3.3e-24"],[true,false,true,false,true,false,true,true,false,true,true,true,false,false,false,true,true,false,false,true,false,false,true,true,false,true,false,false,false,true,true,false,false,false,true,true,false,true,true,true,false,true,true,true,false,true,true,true,false,true,false,false,true,true,false,true,true,false,true,false,false,true,true,true,false,false,true,true,false,false,false,false,true,false,true,true,false,true,true,false,false,false,false,true,false,false,false,false,true,true,true,false,false,false,false,true,false,false,true,false,false,true,true,false,true,false,false,true],[true,true,false,true,false,true,false,false,true,false,false,false,false,false,false,false,false,true,false,false,false,false,false,true,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,true,true,false,true,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,true,false,false,false,true,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false],[true,false,true,false,false,false,true,false,false,true,false,false,false,true,true,false,false,false,false,false,true,false,false,true,false,false,false,true,false,true,true,true,false,false,false,false,true,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,true,false,true,false,false,false,true,true,false,true,false,false,true,false,false,false,true,true,false,false,false,false,true,false,false,false,true,true,false,true,true,false,false,true,false,true,false,true,false,false,false,false,false,false,false,false,false,true,true,true,false,true,true,false,true,false],[true,false,false,false,false,false,true,true,false,false,false,true,true,false,false,false,false,false,true,true,false,true,false,false,true,false,true,false,true,false,false,true,true,true,false,false,true,false,false,true,true,false,false,false,true,false,false,false,false,true,true,true,false,true,false,true,false,true,false,false,true,true,false,false,true,true,false,false,true,false,true,true,false,false,false,false,true,false,false,true,true,false,false,false,true,false,true,false,false,false,false,true,true,true,true,false,true,false,false,false,false,true,false,false,false,true,true,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>z<\/th>\n      <th>lfsr<\/th>\n      <th>Treg<\/th>\n      <th>CD8_T<\/th>\n      <th>CD4_T<\/th>\n      <th>Th17<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"z","targets":1},{"name":"lfsr","targets":2},{"name":"Treg","targets":3},{"name":"CD8_T","targets":4},{"name":"CD4_T","targets":5},{"name":"Th17","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] catecolors_0.1        ComplexHeatmap_2.14.0 colorRamp2_0.1.0     
 [4] tidyr_1.3.1           dplyr_1.1.4           poolr_1.1-1          
 [7] cowplot_1.1.3         ggplot2_3.4.0         fastTopics_0.6-175   
[10] Matrix_1.6-5          workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] matrixStats_1.2.0   fs_1.6.4            RColorBrewer_1.1-3 
  [4] doParallel_1.0.17   progress_1.2.3      httr_1.4.7         
  [7] rprojroot_2.0.4     tools_4.2.0         bslib_0.7.0        
 [10] DT_0.33             utf8_1.2.4          R6_2.5.1           
 [13] irlba_2.3.5.1       BiocGenerics_0.44.0 uwot_0.2.2         
 [16] lazyeval_0.2.2      colorspace_2.1-0    GetoptLong_1.0.5   
 [19] withr_3.0.0         tidyselect_1.2.1    prettyunits_1.2.0  
 [22] processx_3.8.3      compiler_4.2.0      git2r_0.33.0       
 [25] cli_3.6.2           Cairo_1.6-2         plotly_4.10.4      
 [28] labeling_0.4.3      sass_0.4.9          scales_1.3.0       
 [31] SQUAREM_2021.1      quadprog_1.5-8      callr_3.7.3        
 [34] pbapply_1.7-2       mixsqp_0.3-54       stringr_1.5.1      
 [37] digest_0.6.35       rmarkdown_2.26      RhpcBLASctl_0.23-42
 [40] pkgconfig_2.0.3     htmltools_0.5.8.1   highr_0.10         
 [43] fastmap_1.1.1       invgamma_1.1        GlobalOptions_0.1.2
 [46] htmlwidgets_1.6.4   rlang_1.1.3         rstudioapi_0.15.0  
 [49] farver_2.1.1        shape_1.4.6         jquerylib_0.1.4    
 [52] generics_0.1.3      jsonlite_1.8.8      crosstalk_1.2.1    
 [55] gtools_3.9.5        magrittr_2.0.3      S4Vectors_0.36.2   
 [58] Rcpp_1.0.12         munsell_0.5.1       fansi_1.0.6        
 [61] lifecycle_1.0.4     stringi_1.7.6       whisker_0.4.1      
 [64] yaml_2.3.8          mathjaxr_1.6-0      Rtsne_0.17         
 [67] parallel_4.2.0      promises_1.3.0      ggrepel_0.9.5      
 [70] crayon_1.5.2        lattice_0.22-5      circlize_0.4.15    
 [73] hms_1.1.3           knitr_1.46          ps_1.7.6           
 [76] pillar_1.9.0        rjson_0.2.21        stats4_4.2.0       
 [79] codetools_0.2-19    glue_1.7.0          evaluate_0.23      
 [82] getPass_0.2-2       data.table_1.15.4   RcppParallel_5.1.7 
 [85] vctrs_0.6.5         png_0.1-8           httpuv_1.6.14      
 [88] foreach_1.5.2       gtable_0.3.5        purrr_1.0.2        
 [91] clue_0.3-65         ashr_2.2-63         cachem_1.0.8       
 [94] xfun_0.43           later_1.3.2         viridisLite_0.4.2  
 [97] truncnorm_1.0-9     tibble_3.2.1        iterators_1.0.14   
[100] IRanges_2.32.0      cluster_2.1.6      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
