<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-04-15" />

<title>Differential peak analyses</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Differential peak analyses</h1>
<h4 class="date">2025-04-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree47dd55f0534f1811f4127d273447de548afa1939targetblank47dd55fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/47dd55f0534f1811f4127d273447de548afa1939" target="_blank">47dd55f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree47dd55f0534f1811f4127d273447de548afa1939targetblank47dd55fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/47dd55f0534f1811f4127d273447de548afa1939" target="_blank">47dd55f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  PBMCSmall.zip
    Untracked:  Rplots.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/linking_lung_celltype_OCRs_with_asthma_genetics_archive.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RDS
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RData
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/cross_celltypes_edgeR.RDS
    Untracked:  output/cross_celltypes_lymphOnly_edgeR.RData
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_Th17_dispersion_plot.pdf
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/lung_specific_p2g_CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD4_T_all.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD8_T.txt
    Untracked:  output/lung_specific_p2g_GO.RDS
    Untracked:  output/lung_specific_p2g_Memory_B.txt
    Untracked:  output/lung_specific_p2g_NK.txt
    Untracked:  output/lung_specific_p2g_Naive_B.txt
    Untracked:  output/lung_specific_p2g_Treg.txt
    Untracked:  output/lung_specific_peak2genes.RDS
    Untracked:  output/no_bkg/
    Untracked:  output/positions.bed
    Untracked:  output/topic3/
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/h2g_enrichment_u19.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Modified:   analysis/variants_to_function_u19.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/differential_peak_analysis_u19.Rmd</code>) and HTML
(<code>docs/differential_peak_analysis_u19.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/47dd55f0534f1811f4127d273447de548afa1939/analysis/differential_peak_analysis_u19.Rmd" target="_blank">47dd55f</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-15
</td>
<td>
performed DA with edgeR
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<ul>
<li>Identify genomic regions with differential accessibility across
tissue or disease status</li>
<li>Fine-mapped variants within cell-type specific peaks may suggest
specific roles of immune subsets in asthma</li>
<li>Fine-mapped variants within cross-tissue/status DA peaks in specific
immune subsets may further provide the direction of the effects on
genetic risks of asthma</li>
</ul>
<div id="cell-composition-by-donor" class="section level2">
<h2>Cell composition by donor</h2>
<p>Since COB-8 sample has nearly no lymphocytes, we decided to remove
COB-8 from analyses.
<img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="approach" class="section level2">
<h2>Approach</h2>
<p><strong>Models for EdgeR</strong></p>
<p>Modeling mean-variance relationship using a quasi-NB GLM $$ <span
class="math display">\[\begin{eqnarray*} {\rm var}(y_{gi}) = \sigma
^2_g\mu _{gi} + \psi _g \mu _{gi}^2 \end{eqnarray*}\]</span></p>
<p>$$ <strong>MDS plot</strong> distances between samples correspond to
leading biological coefficient of variation (BCV) between those
samples</p>
</div>
<div id="cross-tissue-comparison" class="section level2">
<h2>Cross-tissue comparison</h2>
<p>Different models for cross-tissue comparison:</p>
<ul>
<li>Y ~ group + batch</li>
<li>Y ~ 0 + group + batch</li>
</ul>
<p>Procedures: 1. Filtered out samples with total libray size &lt; 20K
(107/143 samples retained) 2. Filtered out peaks with low read counts
(70% of reads retained) 3. Computed the scaling factor to normalize read
depth by TMM 4. Estimated dispersion based on model design 5. Performed
NB regression</p>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-7-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Notes:<br />
Memory B cells show more cross-tissue differences, while T subsets seem
to violate model assumptions due to large variation</p>
<div id="genomic-region-enrichment-with-rgreat" class="section level3">
<h3>Genomic region enrichment with rGREAT</h3>
<p><strong>Clustering of GO terms</strong></p>
<div class="figure">
<img
src="https://github.com/user-attachments/assets/c204a6dd-f905-4c63-a41a-412749cfe8dc"
alt="" />
<p class="caption">MemB GO BP terms</p>
</div>
<div class="figure">
<img
src="https://github.com/user-attachments/assets/886cdbf0-2c4a-477b-99c0-29c94aa1798d"
alt="" />
<p class="caption">MemB GO MF terms</p>
</div>
<p><strong>A table of significant GO terms</strong></p>
<p>Differential peaks from lung memB cells are enriched for neuron fate
commitment, T cell recombination and apoptotic processes</p>
<p>Differential peaks from lung memB cells are also enriched for immune
receptor binding, metabolism, oxysterol binding, etc</p>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>
Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
</div>
<div id="cross-tissue-comparison-for-asthma-fine-mapped-regions"
class="section level3">
<h3>Cross-tissue comparison for asthma fine-mapped regions</h3>
<pre><code>========================================
circlize version 0.4.15
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
<pre><code>
Attaching package: &#39;circlize&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:colorRamp2&#39;:

    add_transparency, col2value, colorRamp2, rand_color</code></pre>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-484ed42468d86de4a421" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-484ed42468d86de4a421">{"x":{"filter":"none","vertical":false,"caption":"<caption>Selected peaks with FDR &lt; 0.1<\/caption>","data":[["1","2"],["chr3:32963768-32964268_CCR4","chr10:6583286-6583786_PRKCQ-AS1"],["Memory_B","Memory_B"],["1.1e-05","4.0e-07"],[0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Peak<\/th>\n      <th>CellType<\/th>\n      <th>pval<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Peak","targets":1},{"name":"CellType","targets":2},{"name":"pval","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="cross-status-comparison-for-lung-immune-major-lymphocytes"
class="section level2">
<h2>cross-status comparison for lung immune major lymphocytes</h2>
<p>Lung T cells display hundreds of peaks with less chromatin
accessibility in case than control.</p>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-15-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="genomic-regions-enrichment-results" class="section level3">
<h3>Genomic regions enrichment results</h3>
<p>Differential peaks from control lung Tregs are enriched for multiple
terms related to regulation of chemokine production.</p>
</div>
<div id="cross-status-comparison-for-asthma-fine-mapped-regions"
class="section level3">
<h3>Cross-status comparison for asthma fine-mapped regions</h3>
<p>None of the peaks show significant differences in chromatin
accessibility between case and control.</p>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cell-type-specific-analyses" class="section level2">
<h2>Cell-type specific analyses</h2>
<ul>
<li>Model: Y ~ group + batch + donor</li>
<li>COB-8 sample and non-lymphocyte cells removed</li>
<li>Almost half of peaks filtered due to relatively low read counts
compared to others</li>
<li>Normalized by TMM</li>
</ul>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-20-8.png" width="672" style="display: block; margin: auto;" /></p>
<div id="cell-type-comparison-for-asthma-fine-mapped-regions"
class="section level3">
<h3>Cell-type comparison for asthma fine-mapped regions</h3>
<p><strong>EdgeR pseudobulk NB approch</strong></p>
<p>Heatmap plot for comparison p-values</p>
<pre><code>Warning: The heatmap has not been initialized. You might have different results
if you repeatedly execute this function, e.g. when row_km/column_km was
set. It is more suggested to do as `ht = draw(ht); row_order(ht)`.</code></pre>
<p><img src="figure/differential_peak_analysis_u19.Rmd/unnamed-chunk-23-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] circlize_0.4.15                   org.Hs.eg.db_3.16.0              
 [3] AnnotationDbi_1.60.2              ComplexHeatmap_2.14.0            
 [5] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.66.3                  
 [7] Biostrings_2.66.0                 XVector_0.38.0                   
 [9] ggrepel_0.9.6                     edgeR_3.40.2                     
[11] limma_3.58.0                      rGREAT_2.0.2                     
[13] scales_1.3.0                      colorRamp2_0.1.0                 
[15] rtracklayer_1.58.0                tidyr_1.3.1                      
[17] dplyr_1.1.4                       rhdf5_2.42.1                     
[19] SummarizedExperiment_1.28.0       Biobase_2.58.0                   
[21] RcppArmadillo_14.4.1-1            Rcpp_1.0.14                      
[23] Matrix_1.6-5                      GenomicRanges_1.50.2             
[25] GenomeInfoDb_1.34.9               IRanges_2.32.0                   
[27] S4Vectors_0.36.2                  BiocGenerics_0.44.0              
[29] sparseMatrixStats_1.14.0          MatrixGenerics_1.10.0            
[31] matrixStats_1.5.0                 data.table_1.17.0                
[33] stringr_1.5.1                     plyr_1.8.9                       
[35] magrittr_2.0.3                    ggplot2_3.5.2                    
[37] gtable_0.3.6                      gtools_3.9.5                     
[39] gridExtra_2.3                     devtools_2.4.5                   
[41] usethis_3.1.0                     ArchR_1.0.3                      
[43] workflowr_1.7.1                  

loaded via a namespace (and not attached):
  [1] BiocFileCache_2.6.1                     
  [2] crosstalk_1.2.1                         
  [3] BiocParallel_1.32.6                     
  [4] digest_0.6.37                           
  [5] foreach_1.5.2                           
  [6] htmltools_0.5.8.1                       
  [7] GO.db_3.16.0                            
  [8] memoise_2.0.1                           
  [9] cluster_2.1.8.1                         
 [10] doParallel_1.0.17                       
 [11] remotes_2.5.0                           
 [12] prettyunits_1.2.0                       
 [13] colorspace_2.1-1                        
 [14] blob_1.2.4                              
 [15] rappdirs_0.3.3                          
 [16] xfun_0.52                               
 [17] callr_3.7.3                             
 [18] crayon_1.5.3                            
 [19] RCurl_1.98-1.17                         
 [20] jsonlite_2.0.0                          
 [21] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 
 [22] iterators_1.0.14                        
 [23] glue_1.8.0                              
 [24] zlibbioc_1.44.0                         
 [25] GetoptLong_1.0.5                        
 [26] DelayedArray_0.24.0                     
 [27] TxDb.Hsapiens.UCSC.hg38.knownGene_3.16.0
 [28] pkgbuild_1.4.7                          
 [29] Rhdf5lib_1.20.0                         
 [30] shape_1.4.6                             
 [31] DBI_1.2.3                               
 [32] miniUI_0.1.1.1                          
 [33] xtable_1.8-4                            
 [34] progress_1.2.3                          
 [35] clue_0.3-66                             
 [36] bit_4.6.0                               
 [37] DT_0.33                                 
 [38] profvis_0.4.0                           
 [39] htmlwidgets_1.6.4                       
 [40] httr_1.4.7                              
 [41] RColorBrewer_1.1-3                      
 [42] ellipsis_0.3.2                          
 [43] farver_2.1.2                            
 [44] urlchecker_1.0.1                        
 [45] pkgconfig_2.0.3                         
 [46] XML_3.99-0.18                           
 [47] sass_0.4.9                              
 [48] dbplyr_2.5.0                            
 [49] locfit_1.5-9.12                         
 [50] labeling_0.4.3                          
 [51] tidyselect_1.2.1                        
 [52] rlang_1.1.5                             
 [53] later_1.4.2                             
 [54] munsell_0.5.1                           
 [55] tools_4.2.0                             
 [56] cachem_1.1.0                            
 [57] cli_3.6.4                               
 [58] generics_0.1.3                          
 [59] RSQLite_2.3.9                           
 [60] evaluate_1.0.3                          
 [61] fastmap_1.2.0                           
 [62] yaml_2.3.10                             
 [63] processx_3.8.6                          
 [64] knitr_1.50                              
 [65] bit64_4.0.5                             
 [66] fs_1.6.5                                
 [67] purrr_1.0.4                             
 [68] KEGGREST_1.38.0                         
 [69] whisker_0.4.1                           
 [70] mime_0.12                               
 [71] xml2_1.3.8                              
 [72] biomaRt_2.54.1                          
 [73] compiler_4.2.0                          
 [74] rstudioapi_0.17.1                       
 [75] filelock_1.0.3                          
 [76] curl_6.2.2                              
 [77] png_0.1-8                               
 [78] statmod_1.5.0                           
 [79] tibble_3.2.1                            
 [80] bslib_0.9.0                             
 [81] stringi_1.8.4                           
 [82] ps_1.9.0                                
 [83] GenomicFeatures_1.50.4                  
 [84] lattice_0.22-7                          
 [85] vctrs_0.6.5                             
 [86] pillar_1.10.2                           
 [87] lifecycle_1.0.4                         
 [88] rhdf5filters_1.10.1                     
 [89] jquerylib_0.1.4                         
 [90] GlobalOptions_0.1.2                     
 [91] bitops_1.0-9                            
 [92] httpuv_1.6.15                           
 [93] R6_2.6.1                                
 [94] BiocIO_1.8.0                            
 [95] promises_1.3.2                          
 [96] sessioninfo_1.2.2                       
 [97] codetools_0.2-20                        
 [98] pkgload_1.4.0                           
 [99] rprojroot_2.0.4                         
[100] rjson_0.2.23                            
[101] withr_3.0.2                             
[102] GenomicAlignments_1.34.1                
[103] Rsamtools_2.14.0                        
[104] GenomeInfoDbData_1.2.9                  
[105] parallel_4.2.0                          
[106] hms_1.1.3                               
[107] rmarkdown_2.29                          
[108] Cairo_1.6-2                             
[109] git2r_0.33.0                            
[110] getPass_0.2-2                           
[111] shiny_1.10.0                            
[112] restfulr_0.0.15                         </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
