<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2024-11-03" />

<title>A testrun for MAGMA</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">A testrun for MAGMA</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2024-11-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreee496376c46772915f65b9cbe01527aa45bde286atargetblanke496376a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/e496376c46772915f65b9cbe01527aa45bde286a" target="_blank">e496376</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreee496376c46772915f65b9cbe01527aa45bde286atargetblanke496376a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/e496376c46772915f65b9cbe01527aa45bde286a" target="_blank">e496376</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer/
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/positions.bed
    Untracked:  output/topic1/
    Untracked:  output/topic10/
    Untracked:  output/topic11/
    Untracked:  output/topic12/
    Untracked:  output/topic2/
    Untracked:  output/topic3/
    Untracked:  output/topic4/
    Untracked:  output/topic5/
    Untracked:  output/topic6/
    Untracked:  output/topic7/
    Untracked:  output/topic8/
    Untracked:  output/topic9/
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/gene_set_analysis_for_RNA_topics.Rmd</code>) and HTML
(<code>docs/gene_set_analysis_for_RNA_topics.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/e496376c46772915f65b9cbe01527aa45bde286a/analysis/gene_set_analysis_for_RNA_topics.Rmd" target="_blank">e496376</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-03
</td>
<td>
performed MAGMA joint analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="magma" class="section level1">
<h1>MAGMA</h1>
<p><strong>Gene analysis</strong> A linear principal component
regression model that estimates whether there is genetic effect of gene
g on the phenotype Y, conditional on all covariates. The model first
projects genotype matrix for a gene g onto its PCs, pruning away PCs
with very small eigenvalues. Then it performs F test in the regression
of Y on SNP matrix and covariates to estimate genetic effect.</p>
<p><span class="math display">\[
  Y = \alpha_{0g}\vec 1 + X_g^*\alpha_g  + W\beta_g + \epsilon_g
\]</span> When inividual geneotype matrix not available, MAGMA performs
gene test with mean <span class="math inline">\(X^2\)</span> statistics
and a gene p-value is then obtained by using a known approximation of
the sampling distribution. Please refer to the following paper for
details of approximation for the distribution of the weighted
combination of p-values. This model requires summary statistics and
reference LD panel.</p>
<p>Ref: Hou C (2005) A simple approximation for the distribution of the
weighted combination of non-independent or independent probabilities.
Stat Probabil Lett 73: 179–187.</p>
<p><strong>Competitive gene-set analysis (GSA)</strong></p>
<p>One-sided Two-sample T test or linear regression in equivalence is
applied to test whether the genes in a gene set are more strongly
associated with Y or not.</p>
<p>Let Z denote the association z-score. Let <span
class="math inline">\(\S_s\)</span> be an indicator variable with
element <span class="math inline">\(s_g = 1\)</span> defined as for gene
g in gene set s and 0 otherwise. The goal is to test whether <span
class="math inline">\(\beta_s\)</span> is greater than zero, which
represents the difference in association between genes in the gene set
and genes outside the gene set.</p>
<p><span class="math display">\[
Z = \beta_{0s}\vec 1 + S_s\beta_s + \epsilon
\]</span> This also be tested by unpaired two sample T-test, while two
samples can have unequal variances and sample sizes.</p>
<div id="testrun" class="section level2">
<h2>Testrun</h2>
<p>Procedure:</p>
<ol style="list-style-type: decimal">
<li>annotate SNPs and genes</li>
<li>gene-based analysis</li>
<li>gene-set analysis</li>
</ol>
</div>
<div id="gsa-marginal-associations" class="section level2">
<h2>GSA Marginal associations</h2>
<p>We evaluated each set of topic DE genes one at a time for whether
genes in each set have stronger association with the phenotype compared
to those outside of the set. Around half of the tests show significant
p-values, which seems to be inflated.</p>
<div id="changing-background-gene-set" class="section level3">
<h3>Changing background gene set</h3>
<ul>
<li>Total number of genes reduced from to 18K to ~16K</li>
<li>P values for the reduced background genes are very similar to the
full ones.</li>
</ul>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e3dd2f708ffc37673119" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e3dd2f708ffc37673119">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9"],["SET","SET","SET","SET","SET","SET","SET","SET","SET","SET","SET","SET"],[766,724,685,1689,187,888,750,593,491,358,529,155],[0.082533,0.12265,0.10569,-0.050649,0.031389,0.11296,0.14575,0.21682,0.1278,0.070092,0.07594099999999999,0.17778],[0.017551,0.02539,0.02131,-0.015506,0.0033596,0.02576,0.030685,0.040796,0.021952,0.010324,0.013523,0.017341],[0.041514,0.040226,0.041287,0.028165,0.073703,0.036235,0.040046,0.043091,0.045679,0.049731,0.042142,0.079803],[0.023409,0.0011502,0.0052389,0.96392,0.3351,0.00091399,0.00013694,2.4567e-07,0.002576,0.07936600000000001,0.035781,0.012955]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>VARIABLE<\/th>\n      <th>TYPE<\/th>\n      <th>NGENES<\/th>\n      <th>BETA<\/th>\n      <th>BETA_STD<\/th>\n      <th>SE<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"VARIABLE","targets":1},{"name":"TYPE","targets":2},{"name":"NGENES","targets":3},{"name":"BETA","targets":4},{"name":"BETA_STD","targets":5},{"name":"SE","targets":6},{"name":"P","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/gene_set_analysis_for_RNA_topics.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-on-top-100-genes-from-each-topic"
class="section level3">
<h3>Testing on top 100 genes from each topic</h3>
<p>The supplementary table from MAGMA paper shows the mean type 1 error
rates are well controlled for a set of size 100. The <a
href="https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=CP">MSigDB
canonical pathways</a> contains 1320 gene sets from a number of
different databases. I can look into the average size of the gene sets.
<img
src="https://github.com/user-attachments/assets/e4ea0389-c993-46a1-bc6b-c2a69630addf"
alt="MAGMA controlling for type 1 error rates for gene-set analysis" />
Selecting top 100 genes will reduce the number of overlapping genes
across topic DE genes. Indeed, we see more topics have insignificant
association p-values.</p>
</div>
<div id="enrichment-results-by-traits" class="section level3">
<h3>Enrichment results by traits</h3>
<p>Lung related immune diseases from UK Biobank: Blood clot, DVT,
emphysema (chronic bronchitis), asthma, rhinitis eczema/allergy
diagnosed by doctor</p>
<p>Sample size: 336,782.</p>
<p>Instead of all DE genes, I used top 100 up-regulated genes ranked by
z scores to test the enrichment for each topic across immune traits. For
Asthma, now we see only k3, k4, k5 and k12 (4 out of 12) topics show
significant enrichment after multiple testing correction. For other
immune diseases, allergy displays very strong enrichment for k4, k5 and
k12, but none of the others have enrichment signal.</p>
<pre><code>Loading required package: grid</code></pre>
<pre><code>========================================
ComplexHeatmap version 2.14.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9c8a3c9d2f6933833caa" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9c8a3c9d2f6933833caa">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD"],["k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9"],[0.11709,0.10652,0.11026,0.11226,0.10049,0.11051,0.11831,0.10728,0.10741,0.093942,0.10333,0.11077,0.12091,0.10999,0.11384,0.11592,0.10377,0.11415,0.12214,0.1107,0.1109,0.09700499999999999,0.1067,0.11438,0.10476,0.089812,0.09148000000000001,0.095376,0.088214,0.09120399999999999,0.1002,0.09458800000000001,0.08820600000000001,0.08678900000000001,0.088293,0.099009,0.10464,0.09142599999999999,0.091803,0.096085,0.088284,0.090527,0.10165,0.09038400000000001,0.093306,0.086614,0.088994,0.099554,0.097332,0.08364099999999999,0.089561,0.094883,0.080619,0.089796,0.09733600000000001,0.085784,0.088461,0.07869900000000001,0.081081,0.09475],[0.22491,0.2048,0.00014474,0.24919,0.5043,0.00075885,0.00055155,0.0013446,0.55702,0.60118,0.019129,0.16956,0.22138,0.27165,2.2962e-05,0.25108,0.33044,0.13559,7.4606e-06,1.7542e-08,0.012081,0.22042,0.016831,0.12772,0.6434800000000001,0.49338,0.047248,0.7049,0.063331,0.94279,0.68127,0.44429,0.52355,0.72293,0.7335700000000001,0.38192,0.33004,0.0072252,0.35038,0.47749,0.29041,0.70383,0.58123,0.36607,0.91495,0.43589,0.21459,0.34264,0.082482,0.11796,0.37695,0.052548,0.4206,0.27004,0.51655,0.59372,0.7008,0.54796,0.083135,0.79687]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>TRAIT<\/th>\n      <th>VARIABLE<\/th>\n      <th>SE<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"TRAIT","targets":1},{"name":"VARIABLE","targets":2},{"name":"SE","targets":3},{"name":"P","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/gene_set_analysis_for_RNA_topics.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" />
## GSA Joint association</p>
<p>We evaluated all sets of topic DE genes jointly to accounted for the
overlapping genes between topics.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-cdceef25ea5e4ed46489" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cdceef25ea5e4ed46489">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Asthma","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Allergy","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Blood_clot","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Chronic_bronchitis","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD","Asthma/COPD"],["k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9","k10","k11","k12","k1","k2","k3","k4","k5","k6","k7","k8","k9"],[0.043059,0.041612,0.042417,0.02818,0.07482800000000001,0.036885,0.041038,0.044195,0.047608,0.051532,0.042527,0.08070099999999999,0.044403,0.042911,0.04374,0.029059,0.077164,0.038036,0.042319,0.045575,0.049094,0.053141,0.043855,0.08322,0.035479,0.034749,0.03584,0.023767,0.065012,0.030592,0.034505,0.038393,0.040189,0.045666,0.036769,0.071282,0.035185,0.034953,0.036264,0.023735,0.066444,0.031147,0.034296,0.038338,0.041563,0.046873,0.038555,0.073268,0.034589,0.033941,0.035125,0.023105,0.059718,0.030125,0.033936,0.036491,0.039255,0.042377,0.03463,0.06866999999999999],[0.28455,0.040914,0.042745,0.9729100000000001,0.78974,0.029668,0.0059923,4.8452e-05,0.028649,0.23749,0.11964,0.075271,0.18082,0.125,0.0021039,0.53523,0.57207,0.21029,3.8932e-05,1.154e-09,0.015642,0.1953,0.0020577,0.07263799999999999,0.34537,0.44099,0.10706,0.07673099999999999,0.013212,0.35639,0.14189,0.6539199999999999,0.3062,0.62448,0.5571,0.69455,0.96175,0.25551,0.046764,0.16305,0.67686,0.46582,0.56153,0.20573,0.85036,0.8029500000000001,0.23234,0.32289,0.47919,0.032264,0.32598,0.69059,0.14868,0.47275,0.35064,0.76619,0.7513300000000001,0.92791,0.57029,0.52684]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>TRAIT<\/th>\n      <th>VARIABLE<\/th>\n      <th>SE<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"TRAIT","targets":1},{"name":"VARIABLE","targets":2},{"name":"SE","targets":3},{"name":"P","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/gene_set_analysis_for_RNA_topics.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="topic-45-de-genes-with-asthma-magma-z-scores"
class="section level2">
<h2>Topic 4/5 DE genes with Asthma MAGMA Z scores</h2>
<p>Input: 1. Topic specific genes overlapped with MAGMA risk genes (FDR
&lt; 0.05) 2. background genes for topic modeling (~17K)</p>
<p>Enrichment test: Fisher’s exact test</p>
<p>Topic 5 specific genes with high Asthma z scores are enriched for
Treg and Th2 differentiation, and T cell activation.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-3872b756936758f43822" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3872b756936758f43822">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["GOBP_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_CD4_POSITIVE_CD25_POSITIVE_ALPHA_BETA_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_POSITIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_CHEMOKINE_PRODUCTION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_RESPONSE_TO_CHEMOKINE","lung_TRM_genes"],["CTLA4,TOX,KLHL25","GATA3,IL4R","GATA3,RORA,IL4R","CTLA4","IL4R","KLHL25","GATA3","IL4R","GATA3","KLHL25","RORA","RORA","IL4R","IL4R","","","","","","","","","","","",""],[30.778,45.2137,8.9724,88.46129999999999,59.0806,59.0806,44.316,32.2469,29.5654,15.4248,10.7483,7.541,5.0577,3.7611,0,0,0,0,0,0,0,0,0,0,0,0],[0.0049,0.0155,0.0486,0.0861,0.0861,0.0861,0.0946,0.1057,0.1057,0.1729,0.2197,0.2788,0.3689,0.4437,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOTerm<\/th>\n      <th>GenesInSet<\/th>\n      <th>OR<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"GOTerm","targets":1},{"name":"GenesInSet","targets":2},{"name":"OR","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Topic 4 specific genes with high Asthma z scores are enriched for
Th17 immune response.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8afaa1e405584ae8f8d4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8afaa1e405584ae8f8d4">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["GOBP_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_CHEMOKINE_PRODUCTION","GOBP_POSITIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_T_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","lung_TRM_genes","GOBP_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_CD4_POSITIVE_CD25_POSITIVE_ALPHA_BETA_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_RESPONSE_TO_CHEMOKINE"],["IL6R,IL2,JAK2,PRKCQ","IL2,JAK2,PRKCQ","JAK2,PRKCQ","IL6R,IL2","IL6R,TNF","IL2,PHF14","IL2","IL6R,TNF","IL2","IL2","IL6R,IL2","IL2","IL2","","","","","","","","","","","","",""],[28.1767,30.4664,35.2365,18.7215,8.6676,6.9466,29.4704,6.4226,12.8098,13.4002,4.8086,16.3789,7.7501,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.0005999999999999999,0.0025,0.0165,0.0394,0.1287,0.1371,0.1371,0.1371,0.1717,0.1717,0.1717,0.1717,0.2512,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOTerm<\/th>\n      <th>GenesInSet<\/th>\n      <th>OR<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"GOTerm","targets":1},{"name":"GenesInSet","targets":2},{"name":"OR","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Topic 6 specific genes with high Asthma z scores are enriched for
Th17 immune response, lung TRM genes, Treg diff, and chemokine
production.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f28229f4ed8e6e8836fd" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f28229f4ed8e6e8836fd">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["GOBP_T_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_T_HELPER_17_TYPE_IMMUNE_RESPONSE","lung_TRM_genes","GOBP_POSITIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_CHEMOKINE_PRODUCTION","GOBP_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_CD4_POSITIVE_CD25_POSITIVE_ALPHA_BETA_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_RESPONSE_TO_CHEMOKINE","GOBP_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY"],["FCER1G,IL2,IFNG,RORA","IL2,RORA,PHB","IL2,IFNG","IL2,IFNG","IL10,TNF,IFNG","IL2,RORA","IL2,IFNG","TNF,IFNG","IFNG","IL10,IL2","IL2","CXCR4,DUSP1","IL2","IL10","IL2","","","","","","","","","","",""],[13.1773,26.3035,45.3825,35.0852,12.8314,24.1297,20.8534,11.17,62.9125,8.954599999999999,37.7772,8.017300000000001,17.1813,17.9955,11.8053,0,0,0,0,0,0,0,0,0,0,0],[0.0047,0.0047,0.0102,0.0111,0.0111,0.0163,0.0183,0.0509,0.054,0.0609,0.0633,0.0633,0.1117,0.1117,0.1476,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOTerm<\/th>\n      <th>GenesInSet<\/th>\n      <th>OR<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"GOTerm","targets":1},{"name":"GenesInSet","targets":2},{"name":"OR","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="testing-on-genes-having-high-allergy-magma-z-scores"
class="section level3">
<h3>Testing on genes having high Allergy MAGMA Z scores</h3>
<p>Input: 1. Topic specific genes overlapped with MAGMA risk genes (FDR
&lt; 0.05) 2. background genes for topic modeling (~17K)</p>
<p>Enrichment test: Fisher’s exact test</p>
<p>Topic 4/5/6 specific genes with high Allergy Z scores barely show
enrichment.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-03d159f7ea476c06a90f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-03d159f7ea476c06a90f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["GOBP_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_T_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","lung_TRM_genes","GOBP_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_CHEMOKINE_PRODUCTION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_CD4_POSITIVE_CD25_POSITIVE_ALPHA_BETA_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_RESPONSE_TO_CHEMOKINE"],["GATA3,IL4R","CTLA4","GATA3,RORA,IL4R","IL4R","GATA3","IL4R","GATA3","PDCD1","RORA","CTLA4","RORA","IL4R","DOCK10","IL4R","","","","","","","","","","","",""],[26.1942,51.7012,5.1442,34.4954,25.8931,18.8241,17.2492,11.5026,6.2704,5.4449,4.3998,2.9503,2.3715,2.194,0,0,0,0,0,0,0,0,0,0,0,0],[0.0881,0.2084,0.2084,0.2178,0.2229,0.2278,0.2278,0.2872,0.441,0.45,0.4931,0.6342,0.6891,0.6891,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOTerm<\/th>\n      <th>GenesInSet<\/th>\n      <th>OR<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"GOTerm","targets":1},{"name":"GenesInSet","targets":2},{"name":"OR","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8f4c1ae56d9bbc7da2af" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8f4c1ae56d9bbc7da2af">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["GOBP_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_CHEMOKINE_PRODUCTION","GOBP_T_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_TYPE_IMMUNE_RESPONSE","lung_TRM_genes","GOBP_POSITIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_RESPONSE_TO_CHEMOKINE","GOBP_NEGATIVE_REGULATION_OF_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_NEGATIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_17_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_CD4_POSITIVE_CD25_POSITIVE_ALPHA_BETA_REGULATORY_T_CELL_DIFFERENTIATION","GOBP_REGULATION_OF_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_T_HELPER_2_CELL_DIFFERENTIATION","GOBP_POSITIVE_REGULATION_OF_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY","GOBP_NEGATIVE_REGULATION_OF_CHEMOKINE_PRODUCTION","GOBP_REGULATION_OF_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY"],["IL6R,IL2,JAK2","IL6R,IL2","IL2,JAK2","IL6R,FOXP1,TNF","IL6R,FOXP1,IL2","IL2","IL6R,TNF","DOCK10,IL2","JAK2","IL2","IL2","IL2","IL2","CCR7","","","","","","","","","","","",""],[12.8597,11.9352,12.3207,6.2764,4.6865,18.9069,5.5252,4.4293,10.5053,10.5053,8.217599999999999,8.5923,4.9713,1.9411,0,0,0,0,0,0,0,0,0,0,0,0],[0.0548,0.09180000000000001,0.09180000000000001,0.09180000000000001,0.1536,0.2114,0.2114,0.2503,0.2503,0.2503,0.2601,0.2601,0.3754,0.7566000000000001,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOTerm<\/th>\n      <th>GenesInSet<\/th>\n      <th>OR<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"GOTerm","targets":1},{"name":"GenesInSet","targets":2},{"name":"OR","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-set-associations-for-grn-cell-type-specific-targets"
class="section level2">
<h2>Gene-set associations for GRN cell-type specific targets</h2>
<!-- ## Gene-set associations for known immune cell markers -->
<!-- Most immune cell markers were obtained from Gene ontology. The tissue-resident genes were from the single-cell lung paper and ChatGPT. After multiple correction (adj.P < 0.002), tissue-resident markers do not show significance for the gene-set association with Asthma. Neither do the genes involved in cell-cell adhesion or cell migration. The gene sets that are significantly enriched for higher association with Asthma risks are activation of T/B cells and T cell lineage differentiation. -->
<!-- ```{bash eval = FALSE} -->
<!-- cd /home/jinggu/cluster/data/MAGMA/testrun -->
<!-- source ~/.bashrc -->
<!-- traitID="ukb-a-446" -->
<!-- geneset=gene_sets/GO_gene_sets_MAGMA.txt -->
<!-- magma \ -->
<!--     --gene-results $traitID.genes.raw \ -->
<!--     --set-annot ${geneset} \ -->
<!--     --out ${traitID}_GO_enriched -->
<!-- ``` -->
<!-- ```{r} -->
<!-- df_enriched <- read.table(paste0(workdir, "/results/ukb-a-446/ukb-a-446_GO_enriched.gsa.out"), -->
<!--                     skip = 4, header = TRUE) -->
<!-- tbl <- -->
<!--   rbind( -->
<!--     df_top %>% -->
<!--       filter(NGENES > 10 & P < 0.05/12) %>% # correct for multiple testing -->
<!--       select(VARIABLE, NGENES, BETA, SE, P), -->
<!--     df_enriched %>% mutate(VARIABLE = FULL_NAME) %>% -->
<!--       filter(NGENES > 10) %>% #& P < 0.05/33) %>% # correct for multiple testing 0.05/33 = 0.002 -->
<!--       select(VARIABLE, NGENES, BETA, SE, P) %>%  -->
<!--       arrange(P) %>% mutate(P = format(P, scientific = TRUE, digits = 2)) -->
<!--   ) -->
<!-- DT::datatable(tbl) -->
<!-- #write.table(tbl, "../output/fastTopics/top100_topic_GO_immune.gsa.txt",  -->
<!--             #quote = F, row.names = F, sep = "\t") -->
<!-- ``` -->
<!-- ```{r eval = FALSE} -->
<!-- ### Perform permutation to obtain null p-values -->
<!-- ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc") -->
<!-- B = 5000 -->
<!-- for(i in 1:B){ -->
<!--   gene_set = sample(ncbi37$V1, 700, replace = FALSE) -->
<!--   write.table(matrix(c(paste0("set", i), gene_set), nrow = 1), -->
<!--               sprintf("%s/set%s_geneset.txt", workdir, i),  -->
<!--               quote = F, row.names = F, sep=" ", col.names = F) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r eval = FALSE} -->
<!-- f_null <- read.table(paste0(workdir, "/ukb-a-446_permutated700.gsa.out"),  -->
<!--                      skip = 3, header = T) -->
<!-- f_perm <- unlist(lapply(f_reduced$BETA, function(i){sum(f_null$BETA >= i)/5000})) -->
<!-- plot(y = -log10(f_reduced$P),  -->
<!--      x = -log10(f_perm),  -->
<!--      xlab = "-log10(P) after permutation", -->
<!--      ylab = "-log10(P) before permutation") -->
<!-- abline(a = 0, b = 1, col = "red") -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] colorRamp2_0.1.0      ComplexHeatmap_2.14.0 tidyr_1.3.1          
[4] dplyr_1.1.4           data.table_1.15.4     workflowr_1.7.1      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12         circlize_0.4.15     getPass_0.2-2      
 [4] png_0.1-8           ps_1.7.6            rprojroot_2.0.4    
 [7] digest_0.6.35       foreach_1.5.2       utf8_1.2.4         
[10] R6_2.5.1            stats4_4.2.0        evaluate_0.23      
[13] httr_1.4.7          highr_0.10          pillar_1.9.0       
[16] GlobalOptions_0.1.2 rlang_1.1.3         rstudioapi_0.15.0  
[19] whisker_0.4.1       callr_3.7.3         jquerylib_0.1.4    
[22] S4Vectors_0.36.2    GetoptLong_1.0.5    DT_0.33            
[25] rmarkdown_2.26      stringr_1.5.1       htmlwidgets_1.6.4  
[28] compiler_4.2.0      httpuv_1.6.14       xfun_0.43          
[31] pkgconfig_2.0.3     BiocGenerics_0.44.0 shape_1.4.6        
[34] htmltools_0.5.8.1   tidyselect_1.2.1    tibble_3.2.1       
[37] IRanges_2.32.0      codetools_0.2-19    matrixStats_1.2.0  
[40] fansi_1.0.6         withr_3.0.0         crayon_1.5.2       
[43] later_1.3.2         jsonlite_1.8.8      lifecycle_1.0.4    
[46] git2r_0.33.0        magrittr_2.0.3      cli_3.6.2          
[49] stringi_1.8.4       cachem_1.0.8        fs_1.6.4           
[52] promises_1.3.0      doParallel_1.0.17   bslib_0.7.0        
[55] generics_0.1.3      vctrs_0.6.5         rjson_0.2.21       
[58] RColorBrewer_1.1-3  Cairo_1.6-2         iterators_1.0.14   
[61] tools_4.2.0         glue_1.7.0          purrr_1.0.2        
[64] crosstalk_1.2.1     processx_3.8.3      parallel_4.2.0     
[67] fastmap_1.1.1       yaml_2.3.8          clue_0.3-65        
[70] colorspace_2.1-0    cluster_2.1.6       knitr_1.46         
[73] sass_0.4.9         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
