<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2025-03-26" />

<title>Heritability_enrichment_analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Heritability_enrichment_analysis</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2025-03-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreef200d074b0db9630bc96ff3e294daad288279adetargetblankf200d07a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/f200d074b0db9630bc96ff3e294daad288279ade" target="_blank">f200d07</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreef200d074b0db9630bc96ff3e294daad288279adetargetblankf200d07a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/f200d074b0db9630bc96ff3e294daad288279ade" target="_blank">f200d07</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Rplots.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/linking_lung_celltype_OCRs_with_asthma_genetics_archive.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  analysis/variants_to_function_u19.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RDS
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RData
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_Th17_dispersion_plot.pdf
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/no_bkg/
    Untracked:  output/positions.bed
    Untracked:  output/topic3/
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/h2g_enrichment_u19.Rmd</code>)
and HTML (<code>docs/h2g_enrichment_u19.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/f200d074b0db9630bc96ff3e294daad288279ade/analysis/h2g_enrichment_u19.Rmd" target="_blank">f200d07</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-03-26
</td>
<td>
tau* enrichment vs. heritability enrichment
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>characterize and compare our identified open chromatin peaks with
Wang et al.</li>
<li>estimate heritability enrichment for lung-specific immune cells</li>
</ul>
<div id="stratified-ld-score-regression" class="section level2">
<h2>Stratified LD-score regression</h2>
<ul>
<li><code>tau*</code> used to estimate cell-type specific enrichment
conditional on other annotations</li>
<li><code>_prop_h2</code>*<code>total_h2</code> used to estimate
heritability enrichment, a robust quantity regardless of annotation
groups</li>
</ul>
<p><code>_prop_h2</code>: %heritability explained by one annotation
<span class="math inline">\(C_c\)</span></p>
<p><span class="math display">\[h(C_c) = \sum_{j \in C_c}\hat
{Var}({\beta_j)}
         = \sum_{j \in C_c}\sum_{c&#39;:j \in C_{c&#39;}} \hat
\tau_c\]</span> For overlapping annotations, the heritability for each
annotation is calculated as the sum of the coefficients <span
class="math inline">\(\hat \tau_c\)</span> of all annotations, where
each variant j belongs to.</p>
</div>
<div id="characterize-peaks-and-compare-with-blood-datasets"
class="section level2">
<h2>Characterize peaks and compare with blood datasets</h2>
<ul>
<li>snATAC-seq from PBMC samples: Benaglio et al 2023</li>
<li>bulk ATAC-seq for FACS sorted Hematopoietic cells: Calderon et
al. </li>
</ul>
</div>
<div id="compare-with-single-cell-lung-datasets-wang-et-al2020"
class="section level2">
<h2>Compare with single-cell lung datasets (Wang et al2020)</h2>
<p>A barplot of summarizing peak counts by types show Wang et
al. identified more peaks in distal and exonic regions but fewer ones in
promoters compared to our dataset. The color indicates whether each
query peak in U19 overlaps with the one in Wang et al. We see more than
50% peaks in U19 overlapped with ones in Wang et al., while these
overlapped peaks take up less than 50% of peaks in Wang et al.
<img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="different-approaches-for-selecting-peaks-for-enrichment-test"
class="section level2">
<h2>Different approaches for selecting peaks for enrichment test</h2>
<div id="peak-overlapping-across-cell-types" class="section level3">
<h3>Peak overlapping across cell types</h3>
<p>We first grouped cells by tissue and cell-type pair and then called
peaks separately. This allows us to identify peaks specific to each pair
of tissue and cell type, as shown in the heatmap. The highest overlap
percent across all is 50%, and some spleen T subsets show more distinct
peaks compared to all other cell types.</p>
<pre><code>[1] &quot;Check whether all peaks called from individual cell types overlap with joint peak set:&quot;</code></pre>
<pre><code>       Lung_CD4_T    Lung_CD8.CD4_T        Lung_CD8_T     Lung_Memory_B 
             TRUE              TRUE              TRUE              TRUE 
          Lung_NK      Lung_Naive_B        Lung_Other   Lung_Th17.CD4_T 
             TRUE              TRUE              TRUE              TRUE 
        Lung_Treg      Spleen_CD4_T  Spleen_CD8.CD4_T      Spleen_CD8_T 
             TRUE              TRUE              TRUE              TRUE 
  Spleen_Memory_B         Spleen_NK    Spleen_Naive_B      Spleen_Other 
             TRUE              TRUE              TRUE              TRUE 
Spleen_Th17.CD4_T       Spleen_Treg 
             TRUE              TRUE </code></pre>
<p><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-type-specific-peaks" class="section level3">
<h3>Cell-type specific peaks</h3>
<ul>
<li>identify marker peaks for B, T, NK cells only
<ul>
<li>criteria: FDR &lt;= 0.2 &amp; Log2FC &gt;= 0.5</li>
</ul></li>
<li>pair-wise comparison between lymphocytes of lungs and spleens</li>
<li>Lower overlapping of marker peaks for T cells with other cell types
and across tissues</li>
</ul>
<pre><code>[1] &quot;A table of number of overlapping marker peaks between cell types:&quot;</code></pre>
<pre><code>          Lung_B Lung_NK Lung_T Spleen_B Spleen_NK Spleen_T
Lung_B     12153      64     18     6969        36        1
Lung_NK       64    8748    414       90      2231       41
Lung_T        18     414   6577       22       388      443
Spleen_B    6969      90     22    20941        53        1
Spleen_NK     36    2231    388       53      4747       56
Spleen_T       1      41    443        1        56     1129</code></pre>
</div>
<div id="preparing-inputs-for-ldsc" class="section level3">
<h3>Preparing inputs for LDSC</h3>
<ol style="list-style-type: decimal">
<li>Map peaks called from individual cell type to the joint peak set
<ul>
<li>Using the same set of peaks for all the analyses</li>
</ul></li>
<li>Concatenate peaks from CD4 T subsets
<ul>
<li>all peaks from the joint peak set are disjoint</li>
</ul></li>
</ol>
</div>
<div id="examine-the-peaks-beforeafter-liftover" class="section level3">
<h3>Examine the peaks before/after liftover</h3>
<p>The union set of peaks from U19 atac-seq data were first lifted over
to hg19. The following table shows the change in genome builds only made
1% of peaks have inconsistent width.</p>
</div>
</div>
<div id="enrichment-results" class="section level2">
<h2>Enrichment results</h2>
<div id="individual-enrichment-test-across-all-cell-types"
class="section level3">
<h3>Individual enrichment test across all cell types</h3>
<p>For individual test, the quantity of heritability enrichment is used
to demonstrate overall contribution of each annotation. We can use
p-values for cell-type specific coefficients to compare their
contributions across cell types.</p>
<p>Legends:</p>
<ul>
<li>x-axis: heritability enrichment</li>
<li>p-values on top of the data points: p-values for tau</li>
</ul>
<p><strong>Peaks called from individual cell type</strong></p>
<p>All lung and spleen lymphocytes show significant enrichment for
genetic risks of asthma and related diseases, but not for BMI and
height. B cells are less significant than T and NK cells.
<img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>cell-type specific peaks</strong></p>
<p>To avoid unstable estimation, we extended each cell-type specific
peak by 250bp. Spleen T cells were not shown due to very low number of
cell-type specific peaks. Among all cell-type specific peaks, Only lung
T cells show significant enrichment for Allergy and COA, but not
others.</p>
<p><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="joint-test-for-all-major-lymphocytes" class="section level3">
<h3>Joint test for all major lymphocytes</h3>
<p>Legends:</p>
<ul>
<li>x-axis: tau* enrichment calculated based on cell-type specific
coefficients (tau)</li>
<li>p-values on top of the data points: p-values for tau</li>
</ul>
<p><strong>Individual called peaks</strong></p>
<p>Lung T cells remain significant conditional on all other major
lymphocytes.</p>
<pre><code>Joining with `by = join_by(trait)`
Joining with `by = join_by(Category)`</code></pre>
<p><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-11-1.png" width="1344" style="display: block; margin: auto;" />
<strong>Cell-type specific peaks</strong></p>
<p>To estimate cell-type specific heritability, we need to sum up the
coefficients for SNPs uniquely contained in the annotation. The
approximate way is to estimate with cell-type specific peaks. Here we
estimated lung T specific peaks explain around 5% heritability of
Allergy and COA risks.</p>
<pre><code>Joining with `by = join_by(trait)`
Joining with `by = join_by(Category)`</code></pre>
<p><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-12-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="joint-test-for-lung-t-subsets" class="section level3">
<h3>Joint test for Lung T subsets</h3>
<p>I used peaks called from individual cell type to run joint enrichment
analyse. Regulatory T cells remain significantly enriched for genetic
risks for allergy and AOA, conditional on all other lung T subsets.</p>
<pre><code>Joining with `by = join_by(trait)`
Joining with `by = join_by(Category)`</code></pre>
<p><img src="figure/h2g_enrichment_u19.Rmd/unnamed-chunk-13-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.14.0       rtracklayer_1.58.0         
 [3] colorRamp2_0.1.0            rhdf5_2.42.1               
 [5] SummarizedExperiment_1.28.0 Biobase_2.58.0             
 [7] MatrixGenerics_1.10.0       Rcpp_1.0.14                
 [9] Matrix_1.6-5                GenomicRanges_1.50.2       
[11] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[13] S4Vectors_0.36.2            BiocGenerics_0.44.0        
[15] matrixStats_1.5.0           data.table_1.17.0          
[17] stringr_1.5.1               plyr_1.8.9                 
[19] magrittr_2.0.3              ggplot2_3.5.1              
[21] gtable_0.3.6                gtools_3.9.5               
[23] gridExtra_2.3               ArchR_1.0.2                
[25] dplyr_1.1.4                

loaded via a namespace (and not attached):
 [1] bitops_1.0-9             fs_1.6.5                 doParallel_1.0.17       
 [4] RColorBrewer_1.1-3       rprojroot_2.0.4          tools_4.2.0             
 [7] bslib_0.9.0              R6_2.6.1                 colorspace_2.1-1        
[10] rhdf5filters_1.10.1      GetoptLong_1.0.5         withr_3.0.2             
[13] tidyselect_1.2.1         compiler_4.2.0           git2r_0.33.0            
[16] cli_3.6.4                Cairo_1.6-2              DelayedArray_0.24.0     
[19] labeling_0.4.3           sass_0.4.9               scales_1.3.0            
[22] digest_0.6.37            Rsamtools_2.14.0         rmarkdown_2.29          
[25] XVector_0.38.0           pkgconfig_2.0.3          htmltools_0.5.8.1       
[28] fastmap_1.2.0            rlang_1.1.5              GlobalOptions_0.1.2     
[31] rstudioapi_0.17.1        shape_1.4.6              jquerylib_0.1.4         
[34] BiocIO_1.8.0             generics_0.1.3           farver_2.1.2            
[37] jsonlite_1.9.1           BiocParallel_1.32.6      RCurl_1.98-1.16         
[40] GenomeInfoDbData_1.2.9   munsell_0.5.1            Rhdf5lib_1.20.0         
[43] lifecycle_1.0.4          stringi_1.8.4            whisker_0.4.1           
[46] yaml_2.3.10              zlibbioc_1.44.0          parallel_4.2.0          
[49] promises_1.3.2           crayon_1.5.3             lattice_0.22-6          
[52] Biostrings_2.66.0        circlize_0.4.15          knitr_1.49              
[55] pillar_1.10.1            rjson_0.2.23             codetools_0.2-20        
[58] XML_3.99-0.18            glue_1.8.0               evaluate_1.0.3          
[61] vctrs_0.6.5              png_0.1-8                httpuv_1.6.15           
[64] foreach_1.5.2            clue_0.3-66              cachem_1.1.0            
[67] xfun_0.51                restfulr_0.0.15          later_1.4.1             
[70] tibble_3.2.1             iterators_1.0.14         GenomicAlignments_1.34.1
[73] workflowr_1.7.1          cluster_2.1.8.1         </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
