<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-01-29" />

<title>Identify batch effects on the full scATAC-seq dataset</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Identify batch effects on the full
scATAC-seq dataset</h1>
<h4 class="date">2025-01-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree39795518542e9b5b1bf085457d0da5265d3e9496targetblank3979551a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/39795518542e9b5b1bf085457d0da5265d3e9496" target="_blank">3979551</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree39795518542e9b5b1bf085457d0da5265d3e9496targetblank3979551a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/39795518542e9b5b1bf085457d0da5265d3e9496" target="_blank">3979551</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/no_bkg/
    Untracked:  output/positions.bed
    Untracked:  output/topic3/
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/identify_batch_effects_full_atac.Rmd</code>) and HTML
(<code>docs/identify_batch_effects_full_atac.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/39795518542e9b5b1bf085457d0da5265d3e9496/docs/identify_batch_effects_full_atac.html" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9e9d39278f07ab1d1dd5ae2113e9d23469f0b0bc/analysis/identify_batch_effects_full_atac.Rmd" target="_blank">9e9d392</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
<td>
identified batch effects in full ATAC-seq
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/a9d6863a68c4f20ac95e58648be0e1038a881bfe/docs/identify_batch_effects_full_atac.html" target="_blank">a9d6863</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/710e6c4924d714f4634006f4aff5f1f3bc99ebc1/analysis/identify_batch_effects_full_atac.Rmd" target="_blank">710e6c4</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-21
</td>
<td>
updated batch effects for full ATAC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/db57d0062c0b48f53801c9efdb024f0553b37360/docs/identify_batch_effects_full_atac.html" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/a0728215e3305f0e08dde8dba40824f6a786ad67/analysis/identify_batch_effects_full_atac.Rmd" target="_blank">a072821</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
<td>
identified batch effects in topics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>By running topic models on the full scATAC-seq data, we aim to
identify batch effects and compare them with those from the matched
sc-multiomics datasets. The topic model for the data set has 12 topics,
and these topics vary in proportion across cell types, which have been
previously defined by marker gene expression based on scRNA-seq
data.</p>
<pre><code>[1] &quot;check whether L matrix has the same order as metadata: &quot;</code></pre>
<pre><code>[1] TRUE</code></pre>
<div id="batch-information" class="section level2">
<h2>Batch information</h2>
<p>Samples are matched for tissues across batches: at least one lung and
one spleen in each batch</p>
<p>Batch 1 and 2 have only control samples, but the rest have both case
and control samples.</p>
<pre><code>               
                batch2 batch3 batch4 batch5 batch1
  Lung_COB-11        0      0      0   5275      0
  Lung_COB-5         0      0   3000      0      0
  Lung_COB-6         0      0   9523      0      0
  Lung_SMO-2         0      0      0      0   5908
  Lung_SMO-3         0      0      0      0   9816
  Lung_SMO-5      2996      0      0      0      0
  Lung_SMO-7      9172      0      0      0      0
  Lung_SMO-8         0   8068      0      0      0
  Lung_SMO-9         0   6432      0      0      0
  Spleen_COB-10      0      0      0   4352      0
  Spleen_COB-7       0      0  10486      0      0
  Spleen_COB-8       0      0   3933      0      0
  Spleen_COB-9       0      0      0   8609      0
  Spleen_SMO-1       0      0      0      0   4419
  Spleen_SMO-10      0   7313      0      0      0
  Spleen_SMO-4    2975      0      0      0      0</code></pre>
<pre><code>                
                 batch2 batch3 batch4 batch5 batch1
  case_COB-10         0      0      0   4352      0
  case_COB-11         0      0      0   5275      0
  case_COB-5          0      0   3000      0      0
  case_COB-7          0      0  10486      0      0
  case_SMO-8          0   8068      0      0      0
  control_COB-6       0      0   9523      0      0
  control_COB-8       0      0   3933      0      0
  control_COB-9       0      0      0   8609      0
  control_SMO-1       0      0      0      0   4419
  control_SMO-10      0   7313      0      0      0
  control_SMO-2       0      0      0      0   5908
  control_SMO-3       0      0      0      0   9816
  control_SMO-4    2975      0      0      0      0
  control_SMO-5    2996      0      0      0      0
  control_SMO-7    9172      0      0      0      0
  control_SMO-9       0   6432      0      0      0</code></pre>
</div>
<div id="strctural-plots" class="section level2">
<h2>Strctural plots</h2>
<ul>
<li>full scATAC-seq datasets (102277 cells)</li>
<li>peak matrix with a union set of peaks called from each tissue and
cell-type pair</li>
<li>250K peaks in total</li>
</ul>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-4-1.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-4-1.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="distribution-of-topic-proportions-across-batches"
class="section level2">
<h2>Distribution of topic proportions across batches</h2>
<div id="topic-proportions-by-batches" class="section level3">
<h3>Topic proportions by batches</h3>
<p>Lung:</p>
<ul>
<li>k6 - COB-5 from batch 4</li>
<li>k9 - SMO-8/9 from batch 3</li>
<li>k12 - COB-5/6 from batch 4, COB-11 from batch 5</li>
</ul>
<p>Spleen:</p>
<ul>
<li>k1 - SMO-10 from batch 3, COB-10 from batch 5</li>
<li>k6 - SMO-10 from batch 3, SMO-1 from batch 1, COB-10 from batch
5</li>
<li>k9 - SMO-10 from batch 3</li>
<li>k12 - SMO-1 from batch 1 and SMO-4 from batch 2</li>
</ul>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-5-1.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-5-1.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="topic-proportion-by-donors" class="section level3">
<h3>Topic proportion by donors</h3>
<p>Comparing each donor sample with the rest of donor samples separately
for each tissue</p>
<p>Case samples: COB-5, 7, 10, 11 and SMO-8</p>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-1.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-1.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-2.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-6-2.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-1.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-1.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-2.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-7-2.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="correct-batch-effects-by-removing-samples"
class="section level2">
<h2>Correct batch effects by removing samples</h2>
<p>After removing SMO-1/10 sample, we have no batch 3 and less batch
effects in spleens.</p>
<p><img src="figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/39795518542e9b5b1bf085457d0da5265d3e9496/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-8-1.png" target="_blank">3979551</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/db57d0062c0b48f53801c9efdb024f0553b37360/docs/figure/identify_batch_effects_full_atac.Rmd/unnamed-chunk-8-1.png" target="_blank">db57d00</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3         circlize_0.4.15       ComplexHeatmap_2.14.0
 [4] colorRamp2_0.1.0      tidyr_1.3.1           dplyr_1.1.4          
 [7] poolr_1.1-1           cowplot_1.1.3         ggplot2_3.5.1        
[10] fastTopics_0.6-193    Matrix_1.6-5          workflowr_1.7.1      

loaded via a namespace (and not attached):
 [1] matrixStats_1.1.0   fs_1.6.5            RColorBrewer_1.1-3 
 [4] doParallel_1.0.17   progress_1.2.3      httr_1.4.7         
 [7] rprojroot_2.0.4     tools_4.2.0         bslib_0.8.0        
[10] R6_2.5.1            irlba_2.3.5.1       BiocGenerics_0.44.0
[13] uwot_0.2.2          lazyeval_0.2.2      colorspace_2.1-1   
[16] GetoptLong_1.0.5    withr_3.0.2         tidyselect_1.2.1   
[19] prettyunits_1.2.0   processx_3.8.4      compiler_4.2.0     
[22] git2r_0.33.0        cli_3.6.3           plotly_4.10.4      
[25] labeling_0.4.3      sass_0.4.9          scales_1.3.0       
[28] SQUAREM_2021.1      quadprog_1.5-8      callr_3.7.3        
[31] pbapply_1.7-2       mixsqp_0.3-54       stringr_1.5.1      
[34] digest_0.6.37       rmarkdown_2.29      RhpcBLASctl_0.23-42
[37] pkgconfig_2.0.3     htmltools_0.5.8.1   fastmap_1.2.0      
[40] invgamma_1.1        htmlwidgets_1.6.4   rlang_1.1.4        
[43] GlobalOptions_0.1.2 rstudioapi_0.17.1   farver_2.1.2       
[46] shape_1.4.6         jquerylib_0.1.4     generics_0.1.3     
[49] jsonlite_1.8.9      gtools_3.9.5        magrittr_2.0.3     
[52] S4Vectors_0.36.2    Rcpp_1.0.13-1       munsell_0.5.1      
[55] lifecycle_1.0.4     stringi_1.8.4       whisker_0.4.1      
[58] yaml_2.3.10         mathjaxr_1.6-0      Rtsne_0.17         
[61] parallel_4.2.0      promises_1.3.2      ggrepel_0.9.6      
[64] crayon_1.5.3        lattice_0.22-6      hms_1.1.3          
[67] knitr_1.49          ps_1.8.1            pillar_1.10.0      
[70] rjson_0.2.23        stats4_4.2.0        codetools_0.2-20   
[73] glue_1.8.0          evaluate_1.0.1      getPass_0.2-2      
[76] data.table_1.16.4   RcppParallel_5.1.9  vctrs_0.6.5        
[79] png_0.1-8           httpuv_1.6.15       foreach_1.5.2      
[82] gtable_0.3.6        purrr_1.0.2         clue_0.3-66        
[85] ashr_2.2-63         cachem_1.1.0        xfun_0.49          
[88] later_1.4.1         viridisLite_0.4.2   truncnorm_1.0-9    
[91] tibble_3.2.1        iterators_1.0.14    IRanges_2.32.0     
[94] cluster_2.1.8      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
