<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-02-18" />

<title>Linking asthma fine-mapped regions with lung immune OCRs</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Linking asthma fine-mapped regions with
lung immune OCRs</h1>
<h4 class="date">2025-02-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree06b50ece6cd6159f3676fa3c07852e686a2a2c35targetblank06b50eca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/06b50ece6cd6159f3676fa3c07852e686a2a2c35" target="_blank">06b50ec</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree06b50ece6cd6159f3676fa3c07852e686a2a2c35targetblank06b50eca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/06b50ece6cd6159f3676fa3c07852e686a2a2c35" target="_blank">06b50ec</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/no_bkg/
    Untracked:  output/positions.bed
    Untracked:  output/topic3/
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd</code>)
and HTML
(<code>docs/linking_lung_celltype_OCRs_with_asthma_genetics.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/06b50ece6cd6159f3676fa3c07852e686a2a2c35/analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd" target="_blank">06b50ec</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
changed PIP cutoff
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/4d1b43908c72c655b81c81a011c09a73ee35d4bd/docs/linking_lung_celltype_OCRs_with_asthma_genetics.html" target="_blank">4d1b439</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/6fb29da947d3dfc9aa885e76eb9d372affe6cf77/analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd" target="_blank">6fb29da</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
linking OCRs to asthma genetics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/8d25a9cf2c28210bcc54a20d9cb1aaa08eba91b5/docs/linking_lung_celltype_OCRs_with_asthma_genetics.html" target="_blank">8d25a9c</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/a65bf34f613e4aabaa63bdf54d87b9cd2bbb9b21/analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd" target="_blank">a65bf34</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
linking OCRs to asthma genetics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/b9acdaf064dee333c2065063a91dc90f72af2067/docs/linking_lung_celltype_OCRs_with_asthma_genetics.html" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/6757e27af3faa6ab99fa0215d85f0a723d421ac8/analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd" target="_blank">6757e27</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
<td>
linking OCRs to asthma genetics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>For asthma fine-mapped regions, what cell types have open chromatin
at these regions?</p>
<ul>
<li>Shared or cell-type specific?</li>
<li>Present In lungs or spleens?</li>
</ul>
<p>For the selected regions, what genes are differentially expressed for
cell types with open chromatin?</p>
</div>
<div id="procedure" class="section level1">
<h1>Procedure</h1>
<p>What we have:</p>
<ul>
<li>asthma fine-mapped variants grouped by enhancers using multiple
functional data</li>
<li>normalized CA and GE matrix (Feature x cell)
<ul>
<li>TMM and log-normalization (column-wise)</li>
<li>scaled to zscore within each tissue (row-wise)</li>
</ul></li>
</ul>
<ol style="list-style-type: decimal">
<li>Define asthma candidate enhancers by ePIPs <span
class="math inline">\(\ge\)</span> 0.3</li>
</ol>
<ul>
<li>the selected cutoff shows strong enrichment with overlapping
OCRs</li>
<li>randomly sampling 10 enhancers from low ePIPs as negative ctrls</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Overlap with the joint set of peaks in our dataset</li>
</ol>
<ul>
<li>peaks need to contain fine-mapped variants</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Compute average levels across cell types</li>
</ol>
</div>
<div id="result" class="section level1">
<h1>Result</h1>
<div id="enrichment-test-for-enhancers-with-high-epips"
class="section level2">
<h2>Enrichment test for enhancers with high ePIPs</h2>
<p>At a cutoff of 0.3 for ePIPs, enhancers with high ePIPs show
significant enrichment for the overlapped OCRs.</p>
<pre><code>               high_ePIP low_ePIP
overlapped            31       77
not_overlapped        10       73</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  fisher_x
p-value = 0.007228
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.284247 7.180885
sample estimates:
odds ratio 
  2.923176 </code></pre>
</div>
<div id="size-distribution-for-ethans-list-of-enhancers"
class="section level2">
<h2>Size distribution for Ethan’s list of enhancers</h2>
<p><img src="figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b9acdaf064dee333c2065063a91dc90f72af2067/docs/figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-6-1.png" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="processing-chromatin-accessibility-separately-for-each-tissue"
class="section level2">
<h2>Processing chromatin accessibility separately for each tissue</h2>
<p>We computed z-scores separately for each tissue.</p>
<div
id="comparing-chromatin-accessibility-between-peaks-at-high-or-low-epips"
class="section level3">
<h3>Comparing chromatin accessibility between peaks at high or low
ePIPs</h3>
<ul>
<li>Lung CD4_T subsets show overall higher mean accessibility btw peaks
with high ePIP and those with low ePIPs
<ul>
<li>Treg has the highest mean difference</li>
</ul></li>
<li>Spleen subsets show not much differences between peaks with high
ePIPs and those with low ePIPs.</li>
</ul>
<p><img src="figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b9acdaf064dee333c2065063a91dc90f72af2067/docs/figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-8-1.png" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="examining-chromatin-accessibility-for-asthma-fine-mapped-regions"
class="section level2">
<h2>Examining chromatin accessibility for asthma fine-mapped
regions</h2>
<div id="heatmaps-for-accessibiltiy-across-cell-types-in-normal-tissues"
class="section level3">
<h3>Heatmaps for accessibiltiy across cell types in normal tissues</h3>
<ul>
<li>Top section: 10 randomly sampled peaks overlapping enhancers with
low ePIPs</li>
<li>Middle section: average accessibility for all peaks overlapped
regardless of PIPs</li>
<li>Bottom section: all peaks overlapping enhancers with high ePIPs</li>
</ul>
<p>Y-axis: peak coordinate in hg38, aoa-ePIP, coa-ePIP, nearest gene
<img src="figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-12-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b9acdaf064dee333c2065063a91dc90f72af2067/docs/figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-12-1.png" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Overall, we see more chromatin near asthma fine-mapped variants are
more open in lungs than spleens.</li>
<li>shared across all lymphocytes
<ul>
<li>IL2RA, SMAD3, BACH2, RPS26</li>
</ul></li>
<li>shared across T subsets
<ul>
<li>GAL3ST2, CXCR5, CCR7,</li>
</ul></li>
<li>specific to NK/CD8 T cells:
<ul>
<li>IRF1, TM9SF2, TSLP, LINC00298, TBX21,</li>
</ul></li>
<li>specific to Treg:
<ul>
<li>TNFRSF11A, SLC7A10 (both specific to lungs)</li>
</ul></li>
</ul>
</div>
<div
id="heatmaps-for-accessibiltiy-across-cell-types-in-case-and-control"
class="section level3">
<h3>Heatmaps for accessibiltiy across cell types in case and
control</h3>
<p><img src="figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-16-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b9acdaf064dee333c2065063a91dc90f72af2067/docs/figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-16-1.png" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Overall, chromatin levels in control are more open than those in
case. Majoriy of CD4 subsets have lower chromatin accessibility, and
NK/CD8+ T cells show higher accessibility near several regions, which
turns out to be open in negative controls as well.</p>
</div>
</div>
<div id="validate-cell-type-specific-ocrs-near-asthma-risk-enhancers"
class="section level2">
<h2>Validate cell-type specific OCRs near asthma risk enhancers</h2>
<p>Do their nearby genes show differential expression?</p>
<div
id="overlapping-de-genes-across-cell-types-with-the-nearest-genes-from-cell-type-specific-ocrs"
class="section level3">
<h3>Overlapping DE genes across cell types with the nearest genes from
cell-type specific OCRs</h3>
<p>Cell-type specific OCRs are defined as those have higher
accessibility in CD4 T subsets.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-c649a1170c5e876d06e5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c649a1170c5e876d06e5">{"x":{"filter":"none","vertical":false,"caption":"<caption>Cell-type marker genes with cell-type specific open chromains<\/caption>","data":[["1","2","3","4","5","6"],["LPP","TBX21","ZBTB10","CXCR5","CCR4","CCL20"],[0.77,1.2,0.91,4.22,2.78,4.47],[0.524,0.243,0.32,0.179,0.129,0.303],[0.34,0.102,0.186,0.013,0.018,0.031],["7.9e-164","8.8e-163","5.0e-115","7.0e-153"," 3.4e-80"," 0.0e+00"],["NK","NK","NK","Naive_B","Treg","Th17"],["LPP","TBX21","ZBTB10","CXCR5","CCR4","CCL20"],["high","high","low","low","high","high"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>cluster<\/th>\n      <th>gene<\/th>\n      <th>mean_cluster_accessibility<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Gene","targets":1},{"name":"avg_log2FC","targets":2},{"name":"pct.1","targets":3},{"name":"pct.2","targets":4},{"name":"p_val_adj","targets":5},{"name":"cluster","targets":6},{"name":"gene","targets":7},{"name":"mean_cluster_accessibility","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-for-the-expression-of-cell-type-specific-genes"
class="section level3">
<h3>Heatmap for the expression of cell-type specific genes</h3>
<p>The scaling of gene expression was performed separately for each
tissue, which agrees with the differential GE analyses done for lungs
only. As a result, we see consistency in the differential expression and
the heatmap of normalized expression.</p>
<pre><code>Warning: The input is a data frame-like object, convert it to a matrix.</code></pre>
<p><img src="figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/b9acdaf064dee333c2065063a91dc90f72af2067/docs/figure/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd/unnamed-chunk-20-1.png" target="_blank">b9acdaf</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.14.0             BSgenome.Hsapiens.UCSC.hg38_1.4.5
 [3] BSgenome_1.66.3                   Biostrings_2.66.0                
 [5] XVector_0.38.0                    edgeR_3.40.2                     
 [7] limma_3.58.0                      scales_1.3.0                     
 [9] rtracklayer_1.58.0                tidyr_1.3.1                      
[11] dplyr_1.1.4                       rhdf5_2.42.1                     
[13] SummarizedExperiment_1.28.0       Biobase_2.58.0                   
[15] MatrixGenerics_1.10.0             Rcpp_1.0.13-1                    
[17] Matrix_1.6-5                      GenomicRanges_1.50.2             
[19] GenomeInfoDb_1.34.9               IRanges_2.32.0                   
[21] S4Vectors_0.36.2                  BiocGenerics_0.44.0              
[23] matrixStats_1.1.0                 data.table_1.16.4                
[25] stringr_1.5.1                     plyr_1.8.9                       
[27] magrittr_2.0.3                    ggplot2_3.5.1                    
[29] gtable_0.3.6                      gtools_3.9.5                     
[31] gridExtra_2.3                     ArchR_1.0.2                      
[33] workflowr_1.7.1                  

loaded via a namespace (and not attached):
 [1] bitops_1.0-9             fs_1.6.5                 RColorBrewer_1.1-3      
 [4] doParallel_1.0.17        httr_1.4.7               rprojroot_2.0.4         
 [7] tools_4.2.0              bslib_0.8.0              DT_0.33                 
[10] R6_2.5.1                 colorspace_2.1-1         GetoptLong_1.0.5        
[13] rhdf5filters_1.10.1      withr_3.0.2              tidyselect_1.2.1        
[16] processx_3.8.4           compiler_4.2.0           git2r_0.33.0            
[19] cli_3.6.3                Cairo_1.6-2              DelayedArray_0.24.0     
[22] labeling_0.4.3           sass_0.4.9               callr_3.7.3             
[25] digest_0.6.37            Rsamtools_2.14.0         rmarkdown_2.29          
[28] pkgconfig_2.0.3          htmltools_0.5.8.1        fastmap_1.2.0           
[31] htmlwidgets_1.6.4        GlobalOptions_0.1.2      rlang_1.1.4             
[34] rstudioapi_0.17.1        farver_2.1.2             shape_1.4.6             
[37] jquerylib_0.1.4          BiocIO_1.8.0             generics_0.1.3          
[40] jsonlite_1.8.9           crosstalk_1.2.1          BiocParallel_1.32.6     
[43] RCurl_1.98-1.16          GenomeInfoDbData_1.2.9   munsell_0.5.1           
[46] Rhdf5lib_1.20.0          lifecycle_1.0.4          stringi_1.8.4           
[49] whisker_0.4.1            yaml_2.3.10              zlibbioc_1.44.0         
[52] parallel_4.2.0           promises_1.3.2           crayon_1.5.3            
[55] lattice_0.22-6           circlize_0.4.15          locfit_1.5-9.10         
[58] knitr_1.49               ps_1.8.1                 pillar_1.10.0           
[61] rjson_0.2.23             codetools_0.2-20         XML_3.99-0.17           
[64] glue_1.8.0               evaluate_1.0.1           getPass_0.2-2           
[67] png_0.1-8                foreach_1.5.2            vctrs_0.6.5             
[70] httpuv_1.6.15            purrr_1.0.2              clue_0.3-66             
[73] cachem_1.1.0             xfun_0.49                restfulr_0.0.15         
[76] later_1.4.1              tibble_3.2.1             iterators_1.0.14        
[79] GenomicAlignments_1.34.1 cluster_2.1.8            statmod_1.5.0           </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
