<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-04-22" />

<title>understand_lung_specific_peaks</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">understand_lung_specific_peaks</h1>
<h4 class="date">2025-04-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreeffabeb8ec8ea62df42ac1276f9a95f70c19c507dtargetblankffabeb8a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/ffabeb8ec8ea62df42ac1276f9a95f70c19c507d" target="_blank">ffabeb8</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreeffabeb8ec8ea62df42ac1276f9a95f70c19c507dtargetblankffabeb8a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/ffabeb8ec8ea62df42ac1276f9a95f70c19c507d" target="_blank">ffabeb8</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Rplots.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/linking_lung_celltype_OCRs_with_asthma_genetics_archive.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/test_differential_accessibility_MAST.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_MAST.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  ctwas_m6a_joint_m6a_PIP_0.8_genes.txt
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RDS
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RData
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  output/annotation_reference.txt
    Untracked:  output/cross_celltypes_edgeR.RDS
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_Th17_dispersion_plot.pdf
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/lung_specific_p2g_CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD4_T_all.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD8_T.txt
    Untracked:  output/lung_specific_p2g_GO.RDS
    Untracked:  output/lung_specific_p2g_Memory_B.txt
    Untracked:  output/lung_specific_p2g_NK.txt
    Untracked:  output/lung_specific_p2g_Naive_B.txt
    Untracked:  output/lung_specific_p2g_Treg.txt
    Untracked:  output/lung_specific_peak2genes.RDS
    Untracked:  output/positions.bed
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/differential_peak_analysis_u19.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/h2g_enrichment_u19.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Modified:   analysis/variants_to_function_u19.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/lung_specific_peaks_vs_DA_analyses.Rmd</code>) and HTML
(<code>docs/lung_specific_peaks_vs_DA_analyses.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ffabeb8ec8ea62df42ac1276f9a95f70c19c507d/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">ffabeb8</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
bias in peak calling
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objectives" class="section level2">
<h2>Objectives</h2>
<p>The objective is to understand why lung-specific peaks in enrichment
results do not show differential accessibility. What drives the
differences in heritabiltiy enrichment between tissues? We have several
hypotheses as below.</p>
<ol style="list-style-type: decimal">
<li>Peak calling may be too conservative for spleens<br />
</li>
<li>Less variability in pooling samples to call peaks<br />
</li>
<li>Peaks do not pass FDR threshold with multiple testing burdens</li>
</ol>
<p>We will use some positive controls to test these hypotheses: 1) Lung
peaks with high peak calling scores 2) Lung peaks that contain GWAS SNPs
for childhood-onset asthma with high z scores</p>
<div id="summarizing-peaks-uniquely-detected-in-each-tissue"
class="section level3">
<h3>Summarizing peaks uniquely detected in each tissue</h3>
<pre class="r"><code>load(&quot;output/u19_output/peakCalling/peaks_called_by_TissueCellType.RData&quot;)
union_peaks &lt;- proj@peakSet
d_main &lt;- list()
d_main[[&quot;Lung_NK&quot;]] &lt;- d_major$Lung_NK
d_main[[&quot;Spleen_NK&quot;]] &lt;- d_major$Spleen_NK
d_main[[&quot;Lung_B&quot;]] &lt;- 
  unique(GRanges(
    as.character(c(d_major[[&quot;Lung_Naive_B&quot;]], 
                 d_major[[&quot;Lung_Memory_B&quot;]]))
  ))
      
d_main[[&quot;Spleen_B&quot;]] &lt;- 
  unique(GRanges(
    as.character(c(d_major[[&quot;Spleen_Naive_B&quot;]], 
                 d_major[[&quot;Spleen_Memory_B&quot;]]))
  ))</code></pre>
<pre class="r"><code>ggVennDiagram(
  lapply(d_main[c(1,2)], as.character)) + 
  coord_flip()</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggVennDiagram(
  lapply(d_main[c(3,4)], as.character)) + 
  coord_flip()</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" />
Out of 168K peaks unique to lung B cells, we focused on ~7K peaks with
the highest normalized p-values in B cells compared to other cell
types.</p>
</div>
<div id="preparing-positive-control-peaks" class="section level3">
<h3>Preparing positive control peaks</h3>
<p>Set 1: top 5 lung-specific peaks with high peak calling quality</p>
<pre><code>Hits object with 0 hits and 0 metadata columns:
   queryHits subjectHits
   &lt;integer&gt;   &lt;integer&gt;
  -------
  queryLength: 16771 / subjectLength: 140126</code></pre>
<pre><code>GRanges object with 5 ranges and 13 metadata columns:
                seqnames              ranges strand |     score
                   &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
  Lung_Memory_B     chr4   73069096-73069596      * |   74.8021
  Lung_Memory_B     chr1 153966992-153967492      * |   71.2846
  Lung_Memory_B     chr1 228140414-228140914      * |   63.1717
  Lung_Memory_B     chr6   36679727-36680227      * |   57.4205
  Lung_Memory_B     chr1 153617517-153618017      * |   48.6860
                replicateScoreQuantile groupScoreQuantile Reproducibility
                             &lt;numeric&gt;          &lt;numeric&gt;       &lt;numeric&gt;
  Lung_Memory_B                  0.889              0.873               3
  Lung_Memory_B                  0.867              0.847               3
  Lung_Memory_B                  0.843              0.818               4
  Lung_Memory_B                  0.743              0.695               5
  Lung_Memory_B                  0.829              0.800               3
                       GroupReplicate distToGeneStart nearestGene    peakType
                          &lt;character&gt;       &lt;integer&gt; &lt;character&gt; &lt;character&gt;
  Lung_Memory_B Lung_Memory_B._.COB-5             408       COX18      Exonic
  Lung_Memory_B Lung_Memory_B._.COB-6             291     CREB3L4    Promoter
  Lung_Memory_B Lung_Memory_B._.SMO-7             701        GUK1    Intronic
  Lung_Memory_B Lung_Memory_B._.SMO-2            3516      CDKN1A    Intronic
  Lung_Memory_B Lung_Memory_B._.COB-5             780     S100A14    Promoter
                distToTSS  nearestTSS        GC       idx         N
                &lt;integer&gt; &lt;character&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;
  Lung_Memory_B       345  uc011cbc.2    0.5908      3863         0
  Lung_Memory_B       291  uc057lfa.1    0.5449     14742         0
  Lung_Memory_B       331  uc057qdb.1    0.6926     22837         0
  Lung_Memory_B      1205  uc063oeu.1    0.6647      5635         0
  Lung_Memory_B       780  uc001fce.4    0.5968     14621         0
  -------
  seqinfo: 23 sequences from an unspecified genome; no seqlengths</code></pre>
<p><strong>Comparing zscores for SNPs within peaks vs. peak
scores</strong></p>
<p>Peak scores were defined as -log10(p-value) output from MACS2
normalized by sequencing depth across cell types.</p>
<p>We found ~400 out of 168K peaks contain COA GWAS SNPs and their
z-scores mostly center around zero. For those with more deviated z
scores, they tend to locate in peaks with low quality.</p>
<p>Comparing lung and spleen specific peaks, we found more COA GWAS SNPs
in lung peaks have z scores deviated from zero.</p>
<pre class="r"><code># get a set of lung-specific peaks with SNPs in high z scores 
f &lt;- readRDS(&quot;~/cluster/projects/u19_multiomics/data/coa_gp3_finemapping_gwas_L5.rds&quot;)
f_gr &lt;- GRanges(seqnames = paste0(&quot;chr&quot;, f$chr),
                IRanges(start = f$pos,
                        end = f$pos+1))
mcols(f_gr) &lt;- f[, -4:-1]
#liftOver to hg38
ch=import.chain(&quot;~/resources/genomes/chain_files/hg19ToHg38.over.chain&quot;)
seqlevelsStyle(f_gr) &lt;- &quot;UCSC&quot;
f_hg38_gr&lt;- unlist(liftOver(f_gr, ch))

overlapped &lt;- mergeByOverlaps(gr_B, f_hg38_gr)
overlapped_sp &lt;- mergeByOverlaps(spleenB, f_hg38_gr)

gginput &lt;- rbind(data.frame(type = &quot;LungOnly&quot;, score = overlapped$zscore),
             data.frame(type = &quot;SpleenOnly&quot;, score = overlapped_sp$zscore))

plot(sort(overlapped$zscore),
      main = &quot;GWAS SNPs in specific peaks to lung (black) or spleen (red)&quot;,
         ylab = &quot;Sorted Z scores&quot;) + 
  points(sort(overlapped_sp$zscore), col = &quot;red&quot;)</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>integer(0)</code></pre>
<pre class="r"><code>plot(y = overlapped$score,
     x = overlapped$zscore,
     ylab = &quot;Peak score&quot;,
     xlab = &quot;SNP z-score&quot;)</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Set 2: top 50 peaks ranked by absolute z-scores (|Z| &gt; 2.3) of
asthma GWAS SNPs within peaks</p>
<pre class="r"><code>pset2 &lt;- overlapped[order(abs(overlapped$zscore), decreasing = TRUE)[1:50], ]$gr_B
head(pset2)</code></pre>
<pre><code>GRanges object with 6 ranges and 13 metadata columns:
                seqnames              ranges strand |     score
                   &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
   Lung_Naive_B    chr17   39894697-39895197      * |   4.92238
  Lung_Memory_B     chr2 241759819-241760319      * |   7.20963
  Lung_Memory_B     chr2 241759819-241760319      * |   7.20963
   Lung_Naive_B     chr8   80376294-80376794      * |   2.31590
   Lung_Naive_B     chr8   80376294-80376794      * |   2.31590
   Lung_Naive_B     chr8   80376294-80376794      * |   2.31590
                replicateScoreQuantile groupScoreQuantile Reproducibility
                             &lt;numeric&gt;          &lt;numeric&gt;       &lt;numeric&gt;
   Lung_Naive_B                  0.154              0.039               2
  Lung_Memory_B                  0.333              0.199               2
  Lung_Memory_B                  0.333              0.199               2
   Lung_Naive_B                  0.067              0.019               2
   Lung_Naive_B                  0.067              0.019               2
   Lung_Naive_B                  0.067              0.019               2
                       GroupReplicate distToGeneStart nearestGene    peakType
                          &lt;character&gt;       &lt;integer&gt; &lt;character&gt; &lt;character&gt;
   Lung_Naive_B  Lung_Naive_B._.SMO-2           24906       GSDMB      Distal
  Lung_Memory_B Lung_Memory_B._.SMO-2           16755     GAL3ST2    Intronic
  Lung_Memory_B Lung_Memory_B._.SMO-2           16755     GAL3ST2    Intronic
   Lung_Naive_B Lung_Naive_B._.COB-11          109074      ZBTB10      Distal
   Lung_Naive_B Lung_Naive_B._.COB-11          109074      ZBTB10      Distal
   Lung_Naive_B Lung_Naive_B._.COB-11          109074      ZBTB10      Distal
                distToTSS  nearestTSS        GC       idx         N
                &lt;integer&gt; &lt;character&gt; &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt;
   Lung_Naive_B     10990  uc060eul.1    0.4092      5676         0
  Lung_Memory_B      4328  uc061uud.1    0.5329     19166         0
  Lung_Memory_B      4328  uc061uud.1    0.5329     19166         0
   Lung_Naive_B     28971  uc064nyf.1    0.4032      5281         0
   Lung_Naive_B     28971  uc064nyf.1    0.4032      5281         0
   Lung_Naive_B     28971  uc064nyf.1    0.4032      5281         0
  -------
  seqinfo: 23 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="checking-differential-results-for-the-selected-peaks"
class="section level3">
<h3>Checking differential results for the selected peaks</h3>
<p>Set 1</p>
<p>From the table of differential peak analyses, we see the five peaks
only called in lung B cells with high quality do not have large fold
change compared with spleen.</p>
<pre class="r"><code>qlf &lt;- readRDS(&quot;output/u19_output/diff_peak_analysis/cross_tissue_B_edgeR.RDS&quot;)
print(&quot;Positive set 1:&quot;)</code></pre>
<pre><code>[1] &quot;Positive set 1:&quot;</code></pre>
<pre class="r"><code>top5_out &lt;- qlf$table[na.omit(match(findOverlaps(pset1, union_peaks)@to, rownames(qlf$table))), ]</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>top5_out</code></pre>
<pre><code>             logFC   logCPM           F     PValue
62816 -0.041441023 5.143313 0.332383890 0.79223709
14742  0.168538507 4.445024 3.166543964 0.08326005
22837 -0.008858412 4.484598 0.006249968 0.93740749
85884  0.293382484 3.643101 4.797172140 0.03479254
14621  1.758545986 2.058012 4.610849044 0.04054034</code></pre>
<p>We further looked into how read counts in CPMs differ in those peaks
across samples for each tissue. You can see there are reads mapped to
those peak regions in spleen, but somehow they were not called as
peaks.</p>
<pre class="r"><code>y &lt;- readRDS(&quot;output/u19_output/diff_peak_analysis/u19_B_DGEList_edgeR.RDS&quot;)

peak_boxplot(y, peak_idx = rownames(top5_out))</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" />
Set 2</p>
<p>Similar results were seen in the Set 2 peak.</p>
<pre class="r"><code>print(&quot;Positive set 2:&quot;)</code></pre>
<pre><code>[1] &quot;Positive set 2:&quot;</code></pre>
<pre class="r"><code>qlf$table[na.omit(match(findOverlaps(pset2, union_peaks)@to, rownames(qlf$table))), ]</code></pre>
<pre><code>            logFC   logCPM          F    PValue
205004 -0.0291875 2.162244 0.02736533 0.8694975</code></pre>
<pre class="r"><code>peak_boxplot(y, peak_idx = &quot;205004&quot;)</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-8-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="review-peak-calling-process" class="section level3">
<h3>Review peak calling process</h3>
<p>ArchR makes pseudo-bulk replicates by bootstrapping and then call
peaks with MACS2. It requires at least 2 pseudo-bulk replicates to have
a peak call at each locus and the numeric significance cutoff for peaks
to be at least 0.1. There could be bias from pseudo-bulk replicates that
somehow calls many tissue specific peaks that are false positives.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] edgeR_3.40.2                           
 [2] limma_3.58.0                           
 [3] ggrepel_0.9.6                          
 [4] tidyr_1.3.1                            
 [5] dplyr_1.1.4                            
 [6] liftOver_1.22.0                        
 [7] Homo.sapiens_1.3.1                     
 [8] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [9] org.Hs.eg.db_3.16.0                    
[10] GO.db_3.16.0                           
[11] OrganismDbi_1.40.0                     
[12] GenomicFeatures_1.50.4                 
[13] AnnotationDbi_1.60.2                   
[14] rtracklayer_1.58.0                     
[15] gwascat_2.30.0                         
[16] ggVennDiagram_1.5.2                    
[17] rhdf5_2.42.1                           
[18] SummarizedExperiment_1.28.0            
[19] Biobase_2.58.0                         
[20] MatrixGenerics_1.10.0                  
[21] Rcpp_1.0.14                            
[22] Matrix_1.6-5                           
[23] GenomicRanges_1.50.2                   
[24] GenomeInfoDb_1.34.9                    
[25] IRanges_2.32.0                         
[26] S4Vectors_0.36.2                       
[27] BiocGenerics_0.44.0                    
[28] matrixStats_1.5.0                      
[29] data.table_1.17.0                      
[30] stringr_1.5.1                          
[31] plyr_1.8.9                             
[32] magrittr_2.0.3                         
[33] ggplot2_3.5.2                          
[34] gtable_0.3.6                           
[35] gtools_3.9.5                           
[36] gridExtra_2.3                          
[37] ArchR_1.0.2                            
[38] workflowr_1.7.1                        

loaded via a namespace (and not attached):
 [1] colorspace_2.1-1         rjson_0.2.23             rprojroot_2.0.4         
 [4] XVector_0.38.0           fs_1.6.5                 rstudioapi_0.17.1       
 [7] farver_2.1.2             bit64_4.0.5              xml2_1.3.8              
[10] codetools_0.2-20         splines_4.2.0            snpStats_1.48.0         
[13] cachem_1.1.0             knitr_1.50               jsonlite_2.0.0          
[16] Rsamtools_2.14.0         dbplyr_2.5.0             png_0.1-8               
[19] graph_1.76.0             BiocManager_1.30.25      readr_2.1.5             
[22] compiler_4.2.0           httr_1.4.7               fastmap_1.2.0           
[25] cli_3.6.4                later_1.4.2              htmltools_0.5.8.1       
[28] prettyunits_1.2.0        tools_4.2.0              glue_1.8.0              
[31] GenomeInfoDbData_1.2.9   rappdirs_0.3.3           jquerylib_0.1.4         
[34] vctrs_0.6.5              Biostrings_2.66.0        rhdf5filters_1.10.1     
[37] xfun_0.52                ps_1.9.0                 lifecycle_1.0.4         
[40] restfulr_0.0.15          statmod_1.5.0            XML_3.99-0.18           
[43] getPass_0.2-2            zlibbioc_1.44.0          scales_1.3.0            
[46] BSgenome_1.66.3          VariantAnnotation_1.44.1 hms_1.1.3               
[49] promises_1.3.2           RBGL_1.74.0              parallel_4.2.0          
[52] yaml_2.3.10              curl_6.2.2               memoise_2.0.1           
[55] sass_0.4.9               biomaRt_2.54.1           stringi_1.8.4           
[58] RSQLite_2.3.9            BiocIO_1.8.0             filelock_1.0.3          
[61] BiocParallel_1.32.6      rlang_1.1.5              pkgconfig_2.0.3         
[64] bitops_1.0-9             evaluate_1.0.3           lattice_0.22-7          
[67] purrr_1.0.4              Rhdf5lib_1.20.0          labeling_0.4.3          
[70] GenomicAlignments_1.34.1 bit_4.6.0                processx_3.8.6          
[73] tidyselect_1.2.1         R6_2.6.1                 generics_0.1.3          
[76] DelayedArray_0.24.0      DBI_1.2.3                pillar_1.10.2           
[79] whisker_0.4.1            withr_3.0.2              survival_3.8-3          
[82] KEGGREST_1.38.0          RCurl_1.98-1.17          tibble_3.2.1            
[85] crayon_1.5.3             BiocFileCache_2.6.1      tzdb_0.5.0              
[88] rmarkdown_2.29           progress_1.2.3           locfit_1.5-9.12         
[91] blob_1.2.4               callr_3.7.3              git2r_0.33.0            
[94] digest_0.6.37            httpuv_1.6.15            munsell_0.5.1           
[97] bslib_0.9.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
