<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-04-22" />

<title>Converging heritability and differential accessibility analyses</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Converging heritability and differential
accessibility analyses</h1>
<h4 class="date">2025-04-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree4ae1c3f531b86e1e1670ae8c23b22571475ac0c0targetblank4ae1c3fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/4ae1c3f531b86e1e1670ae8c23b22571475ac0c0" target="_blank">4ae1c3f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstree4ae1c3f531b86e1e1670ae8c23b22571475ac0c0targetblank4ae1c3fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/4ae1c3f531b86e1e1670ae8c23b22571475ac0c0" target="_blank">4ae1c3f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Rplots.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_gene expression_u19.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/linking_lung_celltype_OCRs_with_asthma_genetics_archive.Rmd
    Untracked:  analysis/lung_specific_peaks_vs_DA_analyses_v2.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/test_differential_accessibility_MAST.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/archR/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/munge_gwas_sumstats.sh
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_MAST.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  ctwas_m6a_joint_m6a_PIP_0.8_genes.txt
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RDS
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RData
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  output/annotation_reference.txt
    Untracked:  output/cross_celltypes_edgeR.RDS
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_Th17_dispersion_plot.pdf
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/lung_specific_p2g_CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD4_T_all.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4.txt
    Untracked:  output/lung_specific_p2g_CD8.CD4_T.txt
    Untracked:  output/lung_specific_p2g_CD8_T.txt
    Untracked:  output/lung_specific_p2g_GO.RDS
    Untracked:  output/lung_specific_p2g_Memory_B.txt
    Untracked:  output/lung_specific_p2g_NK.txt
    Untracked:  output/lung_specific_p2g_Naive_B.txt
    Untracked:  output/lung_specific_p2g_Treg.txt
    Untracked:  output/lung_specific_peak2genes.RDS
    Untracked:  output/positions.bed
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.html
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/differential_peak_analysis_u19.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/h2g_enrichment_u19.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Modified:   analysis/variants_to_function_u19.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/lung_specific_peaks_vs_DA_analyses.Rmd</code>) and HTML
(<code>docs/lung_specific_peaks_vs_DA_analyses.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/4ae1c3f531b86e1e1670ae8c23b22571475ac0c0/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">4ae1c3f</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
<td>
updated QQ plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/d50372b5014af6f10654715c39a53e82a4581852/docs/lung_specific_peaks_vs_DA_analyses.html" target="_blank">d50372b</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/1f6073bdde7eb2a76f327922922aa2b61070ed5a/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">1f6073b</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
<td>
understand enrichment results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/9daf69a1994681007567767a359e99afdb30366e/docs/lung_specific_peaks_vs_DA_analyses.html" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/582e66338c5b8eba5183dcb713b89a11cc35771e/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">582e663</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
<td>
wflow_publish(c("analysis/lung_specific_peaks_vs_DA_analyses.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/e849411e663d7426ddc8a5ef5cdc39bb61b1e8a6/docs/lung_specific_peaks_vs_DA_analyses.html" target="_blank">e849411</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ac59580d8db59faca8ad9148f0952b6e8e0a35aa/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">ac59580</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
bias in peak calling
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/bfd99eccdea253a6f52830f4b65c1e8b51a019bd/docs/lung_specific_peaks_vs_DA_analyses.html" target="_blank">bfd99ec</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/7102fb601585ede25fee0d29ebf2ff037a3d1cdf/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">7102fb6</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
bias in peak calling
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/a4592e7a7cb6725ec6fe125ce5139e9e8b807352/docs/lung_specific_peaks_vs_DA_analyses.html" target="_blank">a4592e7</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ffabeb8ec8ea62df42ac1276f9a95f70c19c507d/analysis/lung_specific_peaks_vs_DA_analyses.Rmd" target="_blank">ffabeb8</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
<td>
bias in peak calling
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<p>The objective is to understand why lung-specific peaks in enrichment
results do not show differential accessibility. What drives the
differences in heritabiltiy enrichment between tissues? We have several
hypotheses as below.</p>
<ol style="list-style-type: decimal">
<li>Peak calling may be too conservative for spleens</li>
</ol>
<ul>
<li>peak calling p-value threshold: 0.1</li>
<li>Similar percent of unique peaks called in spleen and lung for NK and
B</li>
<li>Much lower percent for T cells due to low number of T cells in
spleens</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Ignoring donor variability in calling peaks over pseudobulks</li>
</ol>
<ul>
<li>Batch effects can potentially cause spurious peaks</li>
<li>ArchR creates pseudo-replicates and call peaks in a sample-aware
manner</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Peaks do not pass FDR threshold with multiple testing burdens in
differential analyses</li>
</ol>
</div>
<div id="method" class="section level1">
<h1>Method</h1>
<p>We will use some positive controls to test these hypotheses: 1) Lung
peaks with high peak calling scores 2) Lung peaks that contain GWAS SNPs
for childhood-onset asthma with high z scores</p>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="summarizing-peaks-called-in-each-tissue"
class="section level2">
<h2>Summarizing peaks called in each tissue</h2>
<p>Lung immune cells have lower fraction of peaks overlapped with blood
than with spleen immune cells.</p>
<p>Lung vs. Spleen</p>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-2-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/a4592e7a7cb6725ec6fe125ce5139e9e8b807352/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-2-1.png" target="_blank">a4592e7</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Lung vs. Blood</p>
<p>Lung and Blood peaks were called by different procedures. Lung peaks
have fixed size (500bp), while blood peaks have sizes ranging from 200
to 5K bp. To make comparison, I first made a union set of peaks and then
comparing peaks from each tissue against the union set.</p>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-3-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/a4592e7a7cb6725ec6fe125ce5139e9e8b807352/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-3-1.png" target="_blank">a4592e7</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="checking-differential-ca-results-for-top-peaks-called-only-in-lungs"
class="section level2">
<h2>Checking differential CA results for top peaks called only in
lungs</h2>
<p>*CA: chromatin accessibility</p>
<div id="preparing-positive-control-peaks" class="section level3">
<h3>Preparing positive control peaks</h3>
<p>top 5 lung-specific peaks ranked by peak calling significance</p>
<p>Selected cell types:</p>
<ul>
<li>B (Memory_B)</li>
<li>T (CD8+ T)</li>
</ul>
<p>From the table of differential peak analyses, we see the five peaks
only called in lung memory B cells with high quality do not have large
fold change compared with spleen, and did not pass p-value thresholds.
Noticeably, all five peaks have decent number of read counts in spleen,
but they were missing in peak calling for spleen samples.</p>
<pre><code>Loading required package: edgeR</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>
Attaching package: &#39;limma&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    plotMA</code></pre>
<pre><code>             logFC   logCPM         F     PValue
109120 -0.05184564 3.888861 0.1740904 0.78953868
191182  0.07945834 4.489175 0.6692661 0.41847111
25479   0.09573716 3.775123 0.4308406 0.51558103
62816  -0.04144102 5.143313 0.3323839 0.79223709
14742   0.16853851 4.445024 3.1665440 0.08326005</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-5-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From the table of differential peak analyses, we see the five peaks
only called in lung CD8_T cells with high quality did not pass p-value
thresholds. Three out of two peaks have comparable or higher mean reads
in spleen, but they were missing in spleen.</p>
<pre><code>            logFC   logCPM          F     PValue
198930 -0.1076247 4.648426 0.50855795 0.48138003
13336   0.1210320 4.167066 0.44920240 0.50793886
179651  5.9207691 1.575136 4.21681589 0.06188397
84783   0.3286604 2.323882 0.06753748 0.79924828
123952  0.1494897 3.231486 0.27642318 0.60322072</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-7-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="reviewing-peak-calling-process" class="section level2">
<h2>Reviewing peak calling process</h2>
<p>ArchR makes pseudo-bulk replicates for each cell grouping (cell-type)
by sampling cells and then pooling the data from each single cell. With
the pseudo-bulk replicates generated, ArchR call peaks for each cell
group with MACS2 in a sample-aware manner. Basically, for each cell
group, ArchR generates a list of peak summits with genomic positions and
score values using 5 pseudo-bulk replicates, and then extends 250 bp
from each summit to define peaks, which results in fixed length of 501
bp. The peaks called from each replicate can overlap with each other, so
ArchR performs two rounds of filtering to generate non-overlapping
peaks: 1) removal of peaks overlapping within each replicate 2) remove
of peaks overlapping across replicates. Lastly, ArchR retain peaks with
at least 2 pseudo-bulk replicates and above the numeric significance
cutoff for peak scores (0.1).</p>
<div id="factors-contributing-to-peaks-not-called-in-spleen"
class="section level3">
<h3>Factors contributing to peaks not called in spleen</h3>
<p>Out of 17.8K peaks only called in lung Memory B cells, ~ 70% (12.8K)
peaks were found to overlap with peaks originally called in spleen
Memory B replicates. Half peaks were not detected in at least two
replicates, while the other half were filtered out during the process of
removing overlapped peaks.</p>
<p>The genome tracks for top peaks overlapped with more than two spleen
replicates show that the peaks at center (lung-specific) have stronger
peaks nearby.</p>
<pre><code>[1] &quot;Number of peaks specifically called in spleen Memory B replicates:&quot;</code></pre>
<pre><code>  Sample nPeaks
1   Rep1   5835
2   Rep2   5877
3   Rep3   3429
4   Rep4   4940
5   Rep5   7021</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-8-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/a4592e7a7cb6725ec6fe125ce5139e9e8b807352/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-8-1.png" target="_blank">a4592e7</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-04-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>CD8_T cells</strong></p>
<p>Out of 23K peaks only called in lung CD8+ T cells, ~ 67% (15K) peaks
were found to overlap with peaks originally called in spleen CD8+ T
replicates. Two thirds of peaks were not detected in at least two
replicates, while the remaining third were filtered out during the
process of removing overlapped peaks.</p>
<pre><code>[1] &quot;Number of peaks specifically called in spleen CD8_T replicates:&quot;</code></pre>
<pre><code>  Sample nPeaks
1   Rep1   4985
2   Rep2   6917
3   Rep3   5066
4   Rep4   5659
5   Rep5   4062</code></pre>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-9-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="overlapping-peaks-only-called-in-lungs-with-coa-gwas-snps"
class="section level2">
<h2>Overlapping peaks only called in lungs with COA GWAS SNPs</h2>
<p><strong>Memory B cells</strong></p>
<p>Two sets of peaks:</p>
<ol style="list-style-type: decimal">
<li>lung-ambiguous: peaks found in more than two spleen replicates</li>
<li>lung-only: peaks with low coverage in spleens or did not pass
reproducibility</li>
</ol>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-10-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>CD8+ T cells</strong>
<img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-11-1.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The discrepancy between heritability enrichment and differential CA
analyses is due to the peak calling process that by selecting smaller
windows for defining peaks arises many peaks only called in one tissue.
Comparing lung-only peaks and lung-ambiguous peaks, we see more GWAS
SNPs with high z-scores in peaks that are clearly only called in lungs.
For the enrichment analyses, we can add the ambiguous peaks to both
tissues. Otherwise, we may focus on lung-blood comparison.</p>
</div>
<div id="overlapping-lung-specific-peaks-with-coa-gwas-snps"
class="section level2">
<h2>Overlapping lung-specific peaks with COA GWAS SNPs</h2>
<p><strong>Differential CA peaks from wilcoxon ranksum
tests</strong></p>
<p>We check differential CA peaks aggregated from different cell types
at each fdr threshold and plot quantile-quantile plots for GWAS SNPs
within those peaks. The LD blocks were labeled on SNPs at z-score
greater than 2.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f8d4fa1566d403c956dd" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f8d4fa1566d403c956dd">{"x":{"filter":"none","vertical":false,"caption":"<caption>Number of differential CA peaks above fdr threshold<\/caption>","data":[["1","2","3"],[0.4,0.5,0.6],[609,1058,2412]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>fdr<\/th>\n      <th>nPeaks<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"fdr","targets":1},{"name":"nPeaks","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/d50372b5014af6f10654715c39a53e82a4581852/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-12-2.png" target="_blank">d50372b</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/9daf69a1994681007567767a359e99afdb30366e/docs/figure/lung_specific_peaks_vs_DA_analyses.Rmd/unnamed-chunk-12-2.png" target="_blank">9daf69a</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-05-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] edgeR_3.40.2                           
 [2] limma_3.58.0                           
 [3] cowplot_1.1.3                          
 [4] ggrepel_0.9.6                          
 [5] eulerr_7.0.2                           
 [6] liftOver_1.22.0                        
 [7] Homo.sapiens_1.3.1                     
 [8] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [9] org.Hs.eg.db_3.16.0                    
[10] GO.db_3.16.0                           
[11] OrganismDbi_1.40.0                     
[12] GenomicFeatures_1.50.4                 
[13] AnnotationDbi_1.60.2                   
[14] rtracklayer_1.58.0                     
[15] gwascat_2.30.0                         
[16] rhdf5_2.42.1                           
[17] SummarizedExperiment_1.28.0            
[18] Biobase_2.58.0                         
[19] MatrixGenerics_1.10.0                  
[20] Rcpp_1.0.14                            
[21] Matrix_1.6-5                           
[22] GenomicRanges_1.50.2                   
[23] GenomeInfoDb_1.34.9                    
[24] IRanges_2.32.0                         
[25] S4Vectors_0.36.2                       
[26] BiocGenerics_0.44.0                    
[27] matrixStats_1.5.0                      
[28] data.table_1.17.0                      
[29] stringr_1.5.1                          
[30] plyr_1.8.9                             
[31] magrittr_2.0.3                         
[32] ggplot2_3.5.2                          
[33] gtable_0.3.6                           
[34] gtools_3.9.5                           
[35] gridExtra_2.3                          
[36] ArchR_1.0.2                            
[37] dplyr_1.1.4                            
[38] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] BiocFileCache_2.6.1      polylabelr_0.3.0         splines_4.2.0           
  [4] crosstalk_1.2.1          BiocParallel_1.32.6      digest_0.6.37           
  [7] htmltools_0.5.8.1        memoise_2.0.1            BSgenome_1.66.3         
 [10] tzdb_0.5.0               Biostrings_2.66.0        readr_2.1.5             
 [13] prettyunits_1.2.0        colorspace_2.1-1         blob_1.2.4              
 [16] rappdirs_0.3.3           xfun_0.52                callr_3.7.3             
 [19] crayon_1.5.3             RCurl_1.98-1.17          jsonlite_2.0.0          
 [22] graph_1.76.0             survival_3.8-3           VariantAnnotation_1.44.1
 [25] glue_1.8.0               polyclip_1.10-7          zlibbioc_1.44.0         
 [28] XVector_0.38.0           DelayedArray_0.24.0      Rhdf5lib_1.20.0         
 [31] scales_1.3.0             DBI_1.2.3                progress_1.2.3          
 [34] bit_4.6.0                DT_0.33                  htmlwidgets_1.6.4       
 [37] httr_1.4.7               pkgconfig_2.0.3          XML_3.99-0.18           
 [40] farver_2.1.2             sass_0.4.9               dbplyr_2.5.0            
 [43] locfit_1.5-9.12          tidyselect_1.2.1         labeling_0.4.3          
 [46] rlang_1.1.5              later_1.4.2              munsell_0.5.1           
 [49] tools_4.2.0              cachem_1.1.0             cli_3.6.4               
 [52] generics_0.1.3           RSQLite_2.3.9            evaluate_1.0.3          
 [55] fastmap_1.2.0            yaml_2.3.10              processx_3.8.6          
 [58] knitr_1.50               bit64_4.0.5              fs_1.6.5                
 [61] purrr_1.0.4              KEGGREST_1.38.0          RBGL_1.74.0             
 [64] whisker_0.4.1            xml2_1.3.8               biomaRt_2.54.1          
 [67] compiler_4.2.0           rstudioapi_0.17.1        filelock_1.0.3          
 [70] curl_6.2.2               png_0.1-8                tibble_3.2.1            
 [73] statmod_1.5.0            bslib_0.9.0              stringi_1.8.4           
 [76] ps_1.9.0                 lattice_0.22-7           vctrs_0.6.5             
 [79] pillar_1.10.2            lifecycle_1.0.4          rhdf5filters_1.10.1     
 [82] BiocManager_1.30.25      jquerylib_0.1.4          snpStats_1.48.0         
 [85] bitops_1.0-9             httpuv_1.6.15            R6_2.6.1                
 [88] BiocIO_1.8.0             promises_1.3.2           codetools_0.2-20        
 [91] rprojroot_2.0.4          rjson_0.2.23             withr_3.0.2             
 [94] GenomicAlignments_1.34.1 Rsamtools_2.14.0         GenomeInfoDbData_1.2.9  
 [97] parallel_4.2.0           hms_1.1.3                tidyr_1.3.1             
[100] rmarkdown_2.29           git2r_0.33.0             getPass_0.2-2           
[103] restfulr_0.0.15         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
