<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2024-11-07" />

<title>Modeling U19 atac-seq with fastTopics</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Modeling U19 atac-seq with fastTopics</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2024-11-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreec2d37841c1a99ff262cefe665cdaa71f5dd837f5targetblankc2d3784a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/c2d37841c1a99ff262cefe665cdaa71f5dd837f5" target="_blank">c2d3784</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreec2d37841c1a99ff262cefe665cdaa71f5dd837f5targetblankc2d3784a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/c2d37841c1a99ff262cefe665cdaa71f5dd837f5" target="_blank">c2d3784</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer/
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/positions.bed
    Untracked:  output/topic1/
    Untracked:  output/topic10/
    Untracked:  output/topic11/
    Untracked:  output/topic12/
    Untracked:  output/topic2/
    Untracked:  output/topic3/
    Untracked:  output/topic4/
    Untracked:  output/topic5/
    Untracked:  output/topic6/
    Untracked:  output/topic7/
    Untracked:  output/topic8/
    Untracked:  output/topic9/
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/u19_lung_atac_fastTopics.Rmd</code>) and HTML
(<code>docs/u19_lung_atac_fastTopics.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/c2d37841c1a99ff262cefe665cdaa71f5dd837f5/analysis/u19_lung_atac_fastTopics.Rmd" target="_blank">c2d3784</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
<td>
topic models on lung atac-seq
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/u19_lung_atac_fastTopics.html" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/3d533291a3b651a3db0288e84f150240ca7f1d42/analysis/u19_lung_atac_fastTopics.Rmd" target="_blank">3d53329</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
<td>
topic models on lung atac-seq
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="topic-modeling-on-u19-lung-atac-seq-dataset"
class="section level1">
<h1>Topic modeling on u19 lung atac-seq dataset</h1>
<p>The objective for this analysis is to identify topics that show
different proportions in cases than controls in lungs. I applied topic
models on atac-seq binary matrix with all cells from lungs. The
dimension for the matrix is 71635 (#cells) by 114334 (#peaks).</p>
<div id="model-fitting" class="section level2">
<h2>Model fitting</h2>
<p>Parameters:</p>
<p>N_updates = 20 N_topics = 12</p>
<p>check the convergence
<img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-1-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Model overview:
  Number of data rows, n: 71635
  Number of data cols, m: 114334
  Rank/number of topics, k: 12
Evaluation of model fit (210 updates performed):
  Poisson NMF log-likelihood: -6.309299075816e+08
  Multinomial topic model log-likelihood: -6.305963765040e+08
  Poisson NMF deviance: +9.181674071631e+08
  Max KKT residual: +7.256944e-04</code></pre>
<pre><code>Set show.size.factors = TRUE, show.mixprops = TRUE and/or show.topic.reps = TRUE in print(...) for more information</code></pre>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-2-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-3-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualize-topics-with-structural-plots" class="section level2">
<h2>Visualize topics with structural plots</h2>
<ul>
<li>plot by cell-types</li>
</ul>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-4-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>plot by status</li>
</ul>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-5-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>plot by batch</li>
</ul>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-6-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="identifying-topics-that-capture-batch-or-donor-effects"
class="section level2">
<h2>Identifying topics that capture batch or donor effects</h2>
<p>Comparing topic proportions across batches, we see certain topic has
large variation within one batch and large variation across batches.</p>
<p>Comparing topic proportion for each individual versus all other
samples, we see k8 topic is strongly contributed by SMO-9.</p>
<pre><code>        
         10102022 10192022 12122022 12142022 8092022
  COB-11        0        0        0     5275       0
  COB-5         0        0    14445        0       0
  COB-6         0        0     9523        0       0
  SMO-2         0        0        0        0    5908
  SMO-3         0        0        0        0    9816
  SMO-5      2996        0        0        0       0
  SMO-7      9172        0        0        0       0
  SMO-8         0     8068        0        0       0
  SMO-9         0     6432        0        0       0</code></pre>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-7-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-7-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-7-2.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="identify-topics-with-different-proportion-between-cell-type-and-status"
class="section level2">
<h2>Identify topics with different proportion between cell type and
status</h2>
<p>Top barplot shows topic proportions do not vary a lot across cell
types.</p>
<p>Bottom barplot shows Topic 7 has higher proportion in cases while
topic 8 shows the opposite across T and NK cells. Topic 5 has higher
proportion in cases while topic 9 shows the opposite across B cells.</p>
<pre><code>         
          Naive_B Memory_B    NK CD8_T CD4_T  Treg  Th17 Other
  case       1555    11037  2611  7580  3421   503   665   416
  control    3645     8470  7903 13063  6001  1015  2517  1233</code></pre>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-8-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-8-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-8-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-8-2.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="characterize-topics" class="section level2">
<h2>Characterize topics</h2>
<div
id="a-table-of-the-nearest-gene-from-top-50-peaks-ranked-by-loadings"
class="section level3">
<h3>A table of the nearest gene from top 50 peaks ranked by
loadings</h3>
<div class="datatables html-widget html-fill-item" id="htmlwidget-77c58e6275387d551690" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-77c58e6275387d551690">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50"],["ARRDC4","ZNF471","CXCR4","P2RX1","MIR4464","SPINT2","TPST1","CD55","LINC01237","IRS2","SMARCE1","OTUD1","POLD1","ZSCAN18","TCF25","KDM4B","IGLL5","ZNRF2","CA11","CENPM","GPBP1","TRMT1","PHF1","ZNF19","ELL2","DNAJB2","KLF2","PPP1R3E","MIR3680-1","MYLPF","CDC123","LOC101927502","ARL4A","KAT6B","NUP98","APOLD1","CDCA2","EVI5","SARAF","USP36","ZNF891","MYOSLID-AS1","RAB6A","OSBP","UBE2D3","TMEM59","ZNF837","BCL2L2-PABPN1","FBXL12","RAB4B-EGLN2"],[null,"MRPL10","CD8A","RIN3","RUNX3","DNAJB2","AIM2","KMT5C","CDCA2","NUP98","KAT6B","BCL2L2-PABPN1","OARD1","CTDP1","P4HB","CD248","PIP4P1","SP1","UBE2D3","PRAG1","IBA57-DT","ZDHHC5","CST7","ATP9B","BICDL1","ZDHHC19","MYLPF","FBXW7","MBOAT7","CBS","SLC25A42","ACTB","FXR2","TMEM141","HSPA8","CENPM","AKTIP","AKNA","OSBP","CAPS","SUCO","HNRNPL","PNKP","INPP5A","ZBTB25","FEM1A","TNFRSF1B","WDTC1","SIVA1","CDCA4"],["DCBLD2","SPATA6","NREP","LOC101927502","ANKRD50","BHLHE41","IKZF2","SYT3","PTK2","ZNF471","KIF13A","GPT2","PHC2","CDS1","SEPT10","TFAP2A","DLC1","KIF26B","PRELID3A","KIAA1549","PTPRM","TENT5B","ARRDC4","NBPF1","KLF4","MYO10","OSBPL1A","FZD4","LARGE1","SRGAP1","HES1","FZD5","PCDH9","FAM160A1","SMAD2","KLF15","CBS","NOS1AP","KLF5","PLAUR","TNFRSF21","IGF1R","EPS8","GXYLT1","KCTD12","NECTIN2","TSPAN13","TNFRSF19","ZBTB7C","RASSF8-AS1"],["ARRDC4","LOC101927502","ADAMTS10","RUNX3","DNM2","KDM2B","CPAMD8","C1orf21","IL2RB","ACOX3","PPCDC","DOT1L","CST7","PRF1","IGF1R","RIN3","CD55","RXFP2","ACTB","MRPL10","COQ10B","PNKP","NFE2L2","SQSTM1","TCF25","KLF2","CDCA2","EDEM1","GFOD1","CASP9","XG","KAT6B","NOTCH4","NF2","ZFPM1","C1orf21","CROCC","CLCN3","FEM1A","IKZF2","TET3","CFAP61","BCL2L2-PABPN1","GPBP1","TOM1L2","ITPKB","AUTS2","DCUN1D2","PARK7","MAP2K3"],["MATN1","INPP5A","TMEM141","FBXL12","LINC00899","CBS","ELL","CBS","CAPS","P4HB","KAT6B","GMFG","KDM4B","STIP1","CIC","PHF1","KMT5C","HDAC4","SLC25A42","GPBP1","AXIN1","HNRNPL","UBE2D3","MBOAT7","NUP98","MOB3A","DNAJB1","CYLD","MYOSLID-AS1","CRIP1","PABPC1","HSPA8","NAPA","IBA57-DT","ATP9B","LINC01003","HIP1","SNAPC2","RAB4B-EGLN2","NIFK","SIVA1","THRAP3","TCF25","LRCH4","ASF1B","PNKP","CDCA2","RAB3A","RRAS","OSBP"],["MAP3K21","HSPA1A","TPPP","KLF11","PPARA","NR4A1","ZFP36L2","SNTA1","AHDC1","HOOK2","MAN1C1","FZD5","EPAS1","CAPRIN2","SERHL2","CNDP2","NR4A1","TSPAN14","ELL2","STARD5","GPR157","HSPA1B","LOC283335","GACAT2","MED29","CD82","CBS","SPTSSA","PLEKHO2","CMIP","FANCE","PRKCZ","PRAG1","SETD9","NDRG1","MAP2K3","PLXND1","ANKRD9","ACTG1","PPM1J","RASSF3","DCUN1D2","MCRIP1","LRP8","MIR210HG","DUSP2","WDSUB1","AVPI1","TSPAN17","BCL7C"],[null,"CD80","PRAG1","NUP98","COTL1","CD55","PPCDC","MIR3680-1","RALGDS","KAT6B","QSOX2","CENPM","TNIP2","LINC01237","SLC25A42","SPINT2","KLF2","ARHGAP31","LINC00899","NDUFV2","TRPV3","ADAM12","SARAF","OTUD1","LOC101927502","BCL10","PHF1","COQ8A","HOOK2","MSH6","TMEM218","CXCR4","CTDP1","RNF139-AS1","KCTD6","RHOA","CROCC","CHRAC1","THRAP3","ZBED4","KDM2A","INSIG1","SLC39A13","PTMA","NPAT","USP28","WDR89","RAB10","MYOSLID-AS1","AIM2"],["MBP","TBC1D7-LOC100130357","LAPTM5","DOT1L","OARD1","CEP68","CLCN3","PTGES","BIN1","RBM47","LOC102724050","RAB32","PTPRE","IRAK2","PPP1R3E","P4HB","KLF2","ARRDC4","NAPA","OSBP","CAPS","HOOK2","TUBA1C","PHF12","PPCDC","MARK4","FBXO31","PIK3R1","CD9","COQ10B","KIFC1","KMT5C","RAB10","KLF4","STK40","NPAT","CXXC5","PNKP","FXR2","AFMID","EIF1","LOC102723672","KDM4B","NLRP3","TRIB1","EDEM1","ENSA","RPS6KA1","ZMIZ1-AS1","MIR6511B1"],[null,"NBPF1","PHF12","PLEC","ANKFY1","SOCS3","ZNF428","TAX1BP3","RBM14","LINC00910","NCBP2","MYCBPAP","HNRNPH1","ZNF566","NAA25","SMARCA4","TMEM250","PHC2","LOC101928597","SUDS3","OIP5","NPHP4","CCT6B","FAAP100","TEX264","UHRF1BP1","KCNN1","HLA-F-AS1","CLK3","FAM43A","COQ8B","CROCC","GGN","HPS4","ST13","LRRC58","NAT10","POC1A","XXYLT1","VRK3","RPL13","SRSF1","PAXIP1-AS2","GPR137","HIC1","DXO","PURA","EMC3","MPHOSPH9","PPP1R15B"],["IKZF3","TIAM2","KAT2B","CA5A","DUSP5","AKAP13","NEDD9","MIR3159","CPD","TEX48","KCTD16","FAM129A","FSIP1","IRF4","RASSF5","KSR1","TNIP2","STAT4","RABAC1","ITPKB","SBF2-AS1","WEE1","PGS1","MIR4464","RPS16","SNX9","CCDC12","CDC42EP3","FAM129A","PHLDA1","LMNA","TK2","ARL4C","SORL1","KLHL21","CSRNP1","XRCC1","LMNA","CPNE5","GALM","KDM6B","S100A10","TNFRSF1B","OGDH","TRDMT1","PIP4K2A","LOC101927851","FAM167A","AIM2","LOC105376365"],["SHLD1","CD6","IRS2","FAM160A1","ATP6V0A4","KCTD2","FBLN7","LOC100130476","TCF7","VIPR1","GTF2IRD1","SPINT2","MFGE8","PDP1","CDS1","USP3","PCBP3","TTC6","ADAMTS10","ESPN","BACH2","NDFIP1","MDS2","SVIL","TGFBR3","IL7R","IGSF9B","RAP1GAP2","CD6","KRT33A","GPR68","GRAMD4","ARHGEF40","RASA3","TNFRSF1B","CBS","ZC3H3","LINC01185","CSGALNACT1","ZC3H12D","NPIPA1","SYTL2","ACSF3","FAM178B","PABPC1","CENPM","LRRC8C-DT","NRP1","PTPRJ","PPIL4"],["KLK4","PCBP3","KCNC3","EEPD1","TCF25","PIP4P1","HDAC4","QSOX2","CLCN3","CD72","PRR16","FBXL12","KLHL32","MIR924HG","ZBTB25","DNAJB2","PPP1R3E","CD248","MYOSLID-AS1","KAT6B","LINC01237","TNIP2","ELL2","INPP5A","CD55","ITPKB","WDR89","ZDHHC5","MIR6511B1","SBF2-AS1","ZNF891","MATN1","MBOAT7","RPS6KA5","GRK5","LINC01003","TFAP2A","BCL2L2-PABPN1","LINC00941","FOXN2","NIFK","GRK5","RBFA","CYLD","CYP46A1","MAP3K21","ZNF19","LOC100130476","RTN4","UBE2D3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"k1","targets":1},{"name":"k2","targets":2},{"name":"k3","targets":3},{"name":"k4","targets":4},{"name":"k5","targets":5},{"name":"k6","targets":6},{"name":"k7","targets":7},{"name":"k8","targets":8},{"name":"k9","targets":9},{"name":"k10","targets":10},{"name":"k11","targets":11},{"name":"k12","targets":12}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="test-motif-enrichment-for-peaks-across-topics"
class="section level2">
<h2>Test motif enrichment for peaks across topics</h2>
<p>Top 500 loading peaks for each topic were tested for motif enrichment
compared to all peaks used in topic models with Homer.</p>
<div id="heatmap-for-top-20-enriche-motifs" class="section level3">
<h3>Heatmap for top 20 enriche motifs</h3>
<p>Topic 7 has similar motif enrichment patterns as topic 5, but topic 5
is strongly enriched for some additional motifs. Topic 8 and 9 that are
in lower proportion in cases than controls indeed have distinct
enrichment patterns.</p>
<pre><code>========================================
circlize version 0.4.15
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
<pre><code>
Attaching package: &#39;circlize&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:colorRamp2&#39;:

    add_transparency, col2value, colorRamp2, rand_color</code></pre>
<p><img src="figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-14-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/0e7f14478e6e6cb15a3f1d32a230384a7c49ea67/docs/figure/u19_lung_atac_fastTopics.Rmd/unnamed-chunk-14-1.png" target="_blank">0e7f144</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-69c09b2ce8e54ab33e45" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-69c09b2ce8e54ab33e45">{"x":{"filter":"none","vertical":false,"caption":"<caption>topic 7<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440"],["Sp5(Zf)","NFY(CCAAT)","Sp1(Zf)","KLF1(Zf)","Sp2(Zf)","KLF3(Zf)","KLF5(Zf)","KLF6(Zf)","KLF14(Zf)","Klf9(Zf)","Klf4(Zf)","En1(Homeobox)","Maz(Zf)","NRF(NRF)","KLF10(Zf)","NRF1(NRF)","Atf1(bZIP)","GFY-Staf(?,Zf)","CRE(bZIP)","Rfx5(HTH)","Ronin(THAP)","LHX9(Homeobox)","NFIL3(bZIP)","MYB(HTH)","AMYB(HTH)","HIF-1b(HLH)","NFkB-p65-Rel(RHD)","Atf7(bZIP)","Hoxa9(Homeobox)","ZNF143|STAF(Zf)","E-box(bHLH)","EKLF(Zf)","NFkB-p65(RHD)","GFY(?)","Elk4(ETS)","c-Jun-CRE(bZIP)","Rfx1(HTH)","Smad3(MAD)","Atf4(bZIP)","Arnt:Ahr(bHLH)","CREB5(bZIP)","Rfx2(HTH)","X-box(HTH)","RFX(HTH)","Pbx3(Homeobox)","Atf2(bZIP)","bHLHE41(bHLH)","Elk1(ETS)","PBX1(Homeobox)","ETV4(ETS)","JunD(bZIP)","BMYB(HTH)","MITF(bHLH)","ZKSCAN1(Zf)","GFX(?)","Pknox1(Homeobox)","Usf2(bHLH)","HIF-1a(bHLH)","Oct6(POU,Homeobox)","bHLHE40(bHLH)","c-Myc(bHLH)","Lhx3(Homeobox)","PU.1:IRF8(ETS:IRF)","Nanog(Homeobox)","CEBP:AP1(bZIP)","Oct4(POU,Homeobox)","Rfx6(HTH)","IRF3(IRF)","ELF1(ETS)","IRF1(IRF)","ZNF467(Zf)","Fli1(ETS)","ETV1(ETS)","PAX6(Paired,Homeobox)","NPAS2(bHLH)","SpiB(ETS)","GABPA(ETS)","PAX5(Paired,Homeobox),condensed","IRF4(IRF)","OCT:OCT(POU,Homeobox,IR1)","Meis1(Homeobox)","HLF(bZIP)","Egr2(Zf)","TFE3(bHLH)","ISRE(IRF)","E2F3(E2F)","E2F4(E2F)","ETS(ETS)","HRE(HSF)","Max(bHLH)","EBF(EBF)","CLOCK(bHLH)","Brn1(POU,Homeobox)","PRDM1(Zf)","IRF8(IRF)","Oct4:Sox17(POU,Homeobox,HMG)","Lhx2(Homeobox)","Isl1(Homeobox)","USF1(bHLH)","Npas4(bHLH)","Hoxa13(Homeobox)","Phox2b(Homeobox)","ETS:E-box(ETS,bHLH)","Brn2(POU,Homeobox)","IRF2(IRF)","PBX2(Homeobox)","PRDM10(Zf)","NFkB-p50,p52(RHD)","Pax7(Paired,Homeobox),long","STAT4(Stat)","Oct11(POU,Homeobox)","Etv2(ETS)","Otx2(Homeobox)","PU.1(ETS)","CHR(?)","SCRT1(Zf)","PAX5(Paired,Homeobox)","DLX1(Homeobox)","Hoxd13(Homeobox)","EWS:FLI1-fusion(ETS)","NFAT(RHD)","ZNF528(Zf)","E2F6(E2F)","OCT:OCT(POU,Homeobox)","Six2(Homeobox)","Hoxd10(Homeobox)","n-Myc(bHLH)","HIF2a(bHLH)","Egr1(Zf)","NPAS(bHLH)","Zfp57(Zf)","Stat3+il21(Stat)","Phox2a(Homeobox)","Six1(Homeobox)","ETS:RUNX(ETS,Runt)","Pax8(Paired,Homeobox)","ZNF669(Zf)","DLX2(Homeobox)","CEBP(bZIP)","E2F(E2F)","Oct2(POU,Homeobox)","Srebp1a(bHLH)","Pit1+1bp(Homeobox)","Tgif1(Homeobox)","E2F1(E2F)","AP-2gamma(AP2)","STAT1(Stat)","Tlx?(NR)","OCT:OCT(POU,Homeobox)","GRE(NR),IR3","Lhx1(Homeobox)","DUX4(Homeobox)","Hnf6b(Homeobox)","FoxD3(forkhead)","RUNX(Runt)","ZNF382(Zf)","MNT(bHLH)","Hoxb4(Homeobox)","TATA-Box(TBP)","Tcf7(HMG)","ZNF711(Zf)","Hoxd11(Homeobox)","PU.1-IRF(ETS:IRF)","OCT:OCT-short(POU,Homeobox)","Zfp809(Zf)","E2F7(E2F)","c-Myc(bHLH)","Hoxa10(Homeobox)","RUNX1(Runt)","Foxh1(Forkhead)","ERG(ETS)","STAT5(Stat)","ZNF41(Zf)","ZBTB33(Zf)","p53(p53)","p53(p53)","ETS1(ETS)","Nkx6.1(Homeobox)","DLX5(Homeobox)","BMAL1(bHLH)","Pit1(Homeobox)","HRE(HSF)","Dlx3(Homeobox)","p73(p53)","Mef2d(MADS)","PRDM14(Zf)","ZFX(Zf)","GATA3(Zf),DR4","Unknown(Homeobox)","Pdx1(Homeobox)","GATA3(Zf),DR8","RBPJ:Ebox(?,bHLH)","CEBP:CEBP(bZIP)","RORg(NR)","Tgif2(Homeobox)","AP-2alpha(AP2)","Srebp2(bHLH)","TCFL2(HMG)","Unknown-ESC-element(?)","Nkx2.5(Homeobox)","NFAT:AP1(RHD,bZIP)","Chop(bZIP)","Prop1(Homeobox)","MafF(bZIP)","Bcl11a(Zf)","Stat3(Stat)","Duxbl(Homeobox)","ZNF7(Zf)","EBF2(EBF)","Zfp281(Zf)","HOXA1(Homeobox)","Barx1(Homeobox)","Tcfcp2l1(CP2)","ELF3(ETS)","PSE(SNAPc)","Foxo1(Forkhead)","EBF1(EBF)","ZNF692(Zf)","ELF5(ETS)","Ets1-distal(ETS)","LEF1(HMG)","Fosl2(bZIP)","Jun-AP1(bZIP)","EBNA1(EBV-virus)","FOXP1(Forkhead)","Zic3(Zf)","RORa(NR)","TEAD1(TEAD)","Cux2(Homeobox)","TCF4(bHLH)","ZNF264(Zf)","BHLHA15(bHLH)","EWS:ERG-fusion(ETS)","Atoh1(bHLH)","GATA3(Zf)","RORgt(NR)","RORgt(NR)","STAT6(Stat)","GLI3(Zf)","WT1(Zf)","GATA:SCL(Zf,bHLH)","SPDEF(ETS)","Nkx2.1(Homeobox)","FXR(NR),IR1","ARE(NR)","p63(p53)","PR(NR)","RUNX2(Runt)","Tcf21(bHLH)","HOXA2(Homeobox)","Nkx2.2(Homeobox)","GSC(Homeobox)","Nr5a2(NR)","Twist(bHLH)","Bach2(bZIP)","PAX3:FKHR-fusion(Paired,Homeobox)","PPARE(NR),DR1","Twist2(bHLH)","bZIP:IRF(bZIP,IRF)","Znf263(Zf)","PPARa(NR),DR1","ZBTB18(Zf)","THRb(NR)","TEAD(TEA)","Gata1(Zf)","IRF:BATF(IRF:bZIP)","HOXB13(Homeobox)","RXR(NR),DR1","CArG(MADS)","Tbx20(T-box)","NeuroG2(bHLH)","TRPS1(Zf)","Tcf12(bHLH)","Tcf3(HMG)","MafB(bZIP)","ZNF341(Zf)","FOXK2(Forkhead)","OCT4-SOX2-TCF-NANOG(POU,Homeobox,HMG)","TEAD3(TEA)","NeuroD1(bHLH)","HINFP(Zf)","Mef2b(MADS)","Six4(Homeobox)","FOXM1(Forkhead)","Fra2(bZIP)","MafK(bZIP)","Gata2(Zf)","JunB(bZIP)","Zic2(Zf)","MafA(bZIP)","Tbox:Smad(T-box,MAD)","GRE(NR),IR3","Gata6(Zf)","Fra1(bZIP)","NF1-halfsite(CTF)","Nr5a2(NR)","Sox21(HMG)","Pitx1(Homeobox)","MyoG(bHLH)","RUNX-AML(Runt)","Bach1(bZIP)","TEAD4(TEA)","FOXK1(Forkhead)","Elf4(ETS)","Smad4(MAD)","Hoxa11(Homeobox)","ZNF322(Zf)","Ascl1(bHLH)","YY1(Zf)","HNF6(Homeobox)","Foxo3(Forkhead)","STAT6(Stat)","ZSCAN22(Zf)","Gfi1b(Zf)","Ap4(bHLH)","THRa(NR)","Foxa3(Forkhead)","Hoxc9(Homeobox)","TEAD2(TEA)","BATF(bZIP)","NF1(CTF)","Foxf1(Forkhead)","PRDM15(Zf)","FoxL2(Forkhead)","Nrf2(bZIP)","EHF(ETS)","Nur77(NR)","TR4(NR),DR1","ZNF415(Zf)","Sox10(HMG)","Fos(bZIP)","CUX1(Homeobox)","HNF4a(NR),DR1","Atf3(bZIP)","Sox7(HMG)","Gata4(Zf)","Sox6(HMG)","PGR(NR)","PRDM9(Zf)","VDR(NR),DR3","Mef2c(MADS)","AP-1(bZIP)","Bapx1(Homeobox)","ZNF189(Zf)","Rbpj1(?)","NF-E2(bZIP)","Sox17(HMG)","FOXA1(Forkhead)","SF1(NR)","ZNF165(Zf)","HEB(bHLH)","ZBTB12(Zf)","ERRg(NR)","Zic(Zf)","Tbr1(T-box)","Bcl6(Zf)","Hand2(bHLH)","Smad2(MAD)","CRX(Homeobox)","CDX4(Homeobox)","Mef2a(MADS)","ZNF652","GRHL2(CP2)","Ascl2(bHLH)","RAR:RXR(NR),DR0","Myf5(bHLH)","FOXA1(Forkhead)","Tbx6(T-box)","Sox9(HMG)","SCL(bHLH)","GLIS3(Zf)","ZNF416(Zf)","ZEB2(Zf)","Brachyury(T-box)","Erra(NR)","E2A(bHLH)","Slug(Zf)","Reverb(NR),DR2","Tbx21(T-box)","Esrrb(NR)","ZNF519(Zf)","Tbet(T-box)","Nkx3.1(Homeobox)","Tbx5(T-box)","Gli2(Zf)","Sox3(HMG)","Snail1(Zf)","Ptf1a(bHLH)","Foxa2(Forkhead)","Eomes(T-box)","Sox2(HMG)","THRb(NR)","MyoD(bHLH)","LRF(Zf)","E2A(bHLH),near_PU.1","Olig2(bHLH)","COUP-TFII(NR)","Sox4(HMG)","HIC1(Zf)","Hoxd12(Homeobox)","Sox15(HMG)","Zac1(Zf)","BORIS(Zf)","EAR2(NR)","ERE(NR),IR3","CTCF(Zf)","AR-halfsite(NR)","COUP-TFII(NR)","ZEB1(Zf)","Fox:Ebox(Forkhead,bHLH)","RARa(NR)","Hnf1(Homeobox)","Pitx1:Ebox(Homeobox,bHLH)","ZNF16(Zf)","NF1:FOXA1(CTF,Forkhead)","MYNN(Zf)","REST-NRSF(Zf)","Mouse_Recombination_Hotspot(Zf)","LXRE(NR),DR4","HNF1b(Homeobox)","Pax7(Paired,Homeobox),longest","ZNF768(Zf)","ZFP3(Zf)","RAR:RXR(NR),DR5","RARg(NR)","DMRT6(DM)","FOXA1:AR(Forkhead,NR)","ZNF317(Zf)","T1ISRE(IRF)","GATA(Zf),IR4","ZNF675(Zf)","DUX(Homeobox)","CTCF-SatelliteElement(Zf?)","DMRT1(DM)","p53(p53)","GATA(Zf),IR3","Cdx2(Homeobox)","NFE2L2(bZIP)","Pax7(Paired,Homeobox)","ZNF136(Zf)"],[1e-41,1e-41,9.999999999999999e-37,9.999999999999999e-37,1e-35,1e-31,1e-28,1e-25,9.999999999999999e-25,1e-22,1e-17,9.999999999999999e-12,1e-09,1e-09,1e-09,1e-08,1e-07,1e-06,1e-06,1e-05,1e-05,1e-05,1e-05,0.0001,0.0001,0.0001,0.0001,0.0001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.01,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],["75.00%","34.60%","55.80%","68.40%","80.20%","52.20%","72.20%","64.60%","80.40%","40.00%","27.00%","17.40%","64.40%","17.40%","25.80%","18.60%","13.80%","5.80%","12.00%","7.40%","5.20%","9.60%","7.80%","20.80%","16.00%","23.00%","2.60%","9.80%","16.60%","7.20%","6.00%","6.20%","8.80%","4.40%","21.60%","6.20%","5.00%","27.20%","3.00%","13.60%","5.80%","4.20%","3.20%","4.00%","4.00%","6.80%","26.00%","20.80%","2.00%","28.20%","3.60%","13.20%","12.40%","1.00%","1.60%","3.80%","7.20%","6.80%","2.80%","8.20%","15.40%","6.80%","2.60%","27.20%","4.60%","2.80%","14.40%","3.00%","18.00%","2.00%","21.80%","25.00%","24.60%","1.20%","13.60%","4.00%","19.00%","2.60%","3.20%","0.40%","13.80%","4.00%","13.00%","3.20%","1.20%","26.20%","20.20%","11.60%","1.60%","9.60%","4.00%","9.80%","2.00%","3.20%","3.80%","0.60%","3.60%","8.80%","9.00%","12.40%","8.60%","0.60%","1.20%","0.60%","1.60%","3.20%","4.80%","3.80%","0.20%","5.60%","1.80%","14.40%","3.40%","6.60%","2.60%","3.00%","6.60%","4.80%","5.00%","9.80%","4.00%","0.20%","21.60%","0.40%","5.20%","4.40%","11.80%","6.00%","27.20%","14.60%","14.00%","5.80%","0.80%","1.60%","2.60%","4.60%","1.60%","4.80%","3.60%","2.00%","1.40%","2.20%","0.80%","16.40%","11.00%","27.20%","1.60%","3.60%","0.20%","1.00%","3.00%","0.20%","2.60%","2.40%","5.20%","0.20%","9.40%","0.80%","6.00%","1.40%","40.60%","8.60%","10.20%","1.40%","7.00%","4.80%","6.60%","1.80%","6.40%","2.20%","20.20%","1.60%","0.20%","2.40%","0.60%","0.60%","17.00%","7.20%","2.20%","15.00%","2.00%","1.20%","1.60%","0.40%","0.80%","1.60%","21.40%","0.20%","1.80%","2.40%","0.20%","2.40%","0.40%","0.40%","18.00%","19.60%","1.60%","0.40%","6.20%","13.60%","0.60%","0.80%","0.80%","0.60%","4.60%","4.60%","0.20%","1.80%","9.40%","6.00%","0.60%","0.80%","1.00%","6.20%","1.20%","9.00%","15.60%","3.20%","8.20%","2.60%","2.00%","3.00%","2.40%","0.20%","1.20%","11.40%","0.40%","3.60%","1.00%","9.80%","6.60%","9.80%","4.80%","8.20%","3.40%","0.40%","0.40%","1.40%","1.00%","16.60%","0.40%","9.20%","18.20%","3.00%","1.00%","1.80%","9.40%","3.80%","8.40%","0.20%","13.80%","4.00%","3.40%","0.60%","1.60%","0.40%","6.00%","12.20%","0.80%","25.20%","5.80%","2.80%","7.40%","1.40%","1.20%","0.20%","2.40%","8.60%","1.00%","1.00%","9.20%","4.20%","9.80%","0.60%","1.20%","6.60%","1.60%","0.20%","3.60%","5.40%","12.40%","2.20%","0.20%","1.80%","3.20%","1.20%","1.20%","3.40%","9.20%","4.80%","0.60%","0.60%","1.60%","3.40%","13.40%","3.80%","7.00%","18.40%","9.40%","3.20%","0.20%","3.00%","2.20%","13.20%","14.00%","5.60%","2.20%","15.80%","1.40%","1.00%","1.40%","1.20%","0.60%","2.80%","9.60%","3.40%","0.40%","0.60%","1.20%","3.40%","3.00%","1.40%","6.60%","1.20%","0.20%","11.40%","0.20%","0.40%","5.80%","6.20%","3.20%","0.80%","1.40%","3.40%","0.80%","1.60%","4.80%","0.20%","2.20%","0.60%","1.00%","3.60%","11.60%","4.20%","7.40%","0.20%","1.80%","1.20%","1.60%","2.00%","19.80%","1.60%","4.00%","7.60%","3.80%","4.20%","1.20%","12.20%","7.80%","1.00%","0.80%","0.20%","0.40%","11.60%","0.20%","4.00%","1.40%","4.60%","2.20%","34.20%","28.00%","9.80%","8.00%","0.20%","15.00%","14.00%","4.80%","0.20%","2.40%","2.00%","4.80%","2.40%","10.40%","20.80%","1.20%","4.80%","7.00%","22.60%","0.80%","5.80%","2.00%","24.40%","5.60%","29.80%","11.80%","7.20%","10.00%","1.80%","21.40%","2.40%","1.80%","33.00%","7.60%","6.00%","0.20%","3.00%","19.20%","6.40%","11.20%","0.60%","10.60%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%","0.00%"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>motif<\/th>\n      <th>P-value<\/th>\n      <th>% of Target Sequences with Motif<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"motif","targets":1},{"name":"P-value","targets":2},{"name":"% of Target Sequences with Motif","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>`</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] circlize_0.4.15             catecolors_0.1             
 [3] rhdf5_2.42.1                SummarizedExperiment_1.28.0
 [5] Biobase_2.58.0              MatrixGenerics_1.10.0      
 [7] Rcpp_1.0.13-1               GenomicRanges_1.50.2       
 [9] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[11] S4Vectors_0.36.2            BiocGenerics_0.44.0        
[13] matrixStats_1.2.0           data.table_1.16.2          
[15] stringr_1.5.1               plyr_1.8.9                 
[17] magrittr_2.0.3              gtable_0.3.6               
[19] gtools_3.9.5                gridExtra_2.3              
[21] ArchR_1.0.2                 ComplexHeatmap_2.14.0      
[23] colorRamp2_0.1.0            tidyr_1.3.1                
[25] dplyr_1.1.4                 poolr_1.1-1                
[27] cowplot_1.1.3               ggplot2_3.5.1              
[29] fastTopics_0.6-193          Matrix_1.6-5               
[31] workflowr_1.7.1            

loaded via a namespace (and not attached):
 [1] Rtsne_0.17             colorspace_2.1-1       rjson_0.2.21          
 [4] rprojroot_2.0.4        XVector_0.38.0         GlobalOptions_0.1.2   
 [7] fs_1.6.5               clue_0.3-65            rstudioapi_0.15.0     
[10] farver_2.1.2           DT_0.33                ggrepel_0.9.6         
[13] fansi_1.0.6            mathjaxr_1.6-0         codetools_0.2-19      
[16] doParallel_1.0.17      cachem_1.1.0           knitr_1.48            
[19] jsonlite_1.8.9         Cairo_1.6-2            RhpcBLASctl_0.23-42   
[22] cluster_2.1.6          ashr_2.2-63            png_0.1-8             
[25] uwot_0.2.2             compiler_4.2.0         httr_1.4.7            
[28] fastmap_1.2.0          lazyeval_0.2.2         cli_3.6.3             
[31] later_1.3.2            htmltools_0.5.8.1      prettyunits_1.2.0     
[34] tools_4.2.0            glue_1.8.0             GenomeInfoDbData_1.2.9
[37] jquerylib_0.1.4        rhdf5filters_1.10.1    vctrs_0.6.5           
[40] crosstalk_1.2.1        iterators_1.0.14       xfun_0.49             
[43] ps_1.7.6               lifecycle_1.0.4        irlba_2.3.5.1         
[46] getPass_0.2-2          zlibbioc_1.44.0        scales_1.3.0          
[49] hms_1.1.3              promises_1.3.0         parallel_4.2.0        
[52] RColorBrewer_1.1-3     yaml_2.3.10            pbapply_1.7-2         
[55] sass_0.4.9             stringi_1.8.4          SQUAREM_2021.1        
[58] highr_0.11             foreach_1.5.2          truncnorm_1.0-9       
[61] shape_1.4.6            rlang_1.1.4            pkgconfig_2.0.3       
[64] bitops_1.0-7           evaluate_1.0.1         lattice_0.22-5        
[67] invgamma_1.1           Rhdf5lib_1.20.0        purrr_1.0.2           
[70] labeling_0.4.3         htmlwidgets_1.6.4      processx_3.8.3        
[73] tidyselect_1.2.1       R6_2.5.1               generics_0.1.3        
[76] DelayedArray_0.24.0    pillar_1.9.0           whisker_0.4.1         
[79] withr_3.0.2            RCurl_1.98-1.14        mixsqp_0.3-54         
[82] tibble_3.2.1           crayon_1.5.3           utf8_1.2.4            
[85] plotly_4.10.4          rmarkdown_2.29         GetoptLong_1.0.5      
[88] progress_1.2.3         callr_3.7.3            git2r_0.33.0          
[91] digest_0.6.37          httpuv_1.6.14          RcppParallel_5.1.9    
[94] munsell_0.5.1          viridisLite_0.4.2      bslib_0.8.0           
[97] quadprog_1.5-8        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
