<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-03-27" />

<title>Linking asthma fine-mapped regions with lung immune OCRs</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Linking asthma fine-mapped regions with
lung immune OCRs</h1>
<h4 class="date">2025-03-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreed3ed9d1828635a30cc4aebdb6e277cd4a50dbcc4targetblankd3ed9d1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/d3ed9d1828635a30cc4aebdb6e277cd4a50dbcc4" target="_blank">d3ed9d1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreed3ed9d1828635a30cc4aebdb6e277cd4a50dbcc4targetblankd3ed9d1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/d3ed9d1828635a30cc4aebdb6e277cd4a50dbcc4" target="_blank">d3ed9d1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  ArchRLogs/
    Untracked:  Rplots.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/ArchRLogs/
    Untracked:  analysis/GOBP_B_CELL_ACTIVATION_INVOLVED_IN_IMMUNE_RESPONSE.v2024.1.Hs.gmt
    Untracked:  analysis/Rplots.pdf
    Untracked:  analysis/cross_tissue_DE_u19_k10_fastTopics.Rmd
    Untracked:  analysis/differential_peak_analysis.Rmd
    Untracked:  analysis/full_atac_h2g_enrichment.Rmd
    Untracked:  analysis/identify_T_subsets.Rmd
    Untracked:  analysis/identify_batch_effects_full_atac_k8.Rmd
    Untracked:  analysis/linking_lung_celltype_OCRs_with_asthma_genetics_archive.Rmd
    Untracked:  analysis/run_fastTopics.R
    Untracked:  analysis/run_fastTopics_lungOnly.R
    Untracked:  analysis/test.pdf
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k10.Rmd
    Untracked:  analysis/u19_full_atac_fastTopics_k8.Rmd
    Untracked:  analysis/u19_lung_atac_fastTopics_no_correction.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  analysis/ukb-a-446.log
    Untracked:  analysis/ukb-a-446_GO_immune.log
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/compute_pancreas_factors.R
    Untracked:  code/link_variant_to_function.R
    Untracked:  code/make_plots.R
    Untracked:  code/run_GO_enrichment.R
    Untracked:  code/run_fastTopics.R
    Untracked:  code/run_magma.sh
    Untracked:  code/run_magma/
    Untracked:  code/run_topic_DE_analysis.R
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RDS
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.RData
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.csv
    Untracked:  data/CRE_0_asthma_fine-mapping_summary.txt
    Untracked:  data/DA_peaks_Tsub_vs_others.RDS
    Untracked:  data/DA_peaks_by_cell_type.RDS
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/Tsuo2022_meta_analysis_asthma_risk_genes.csv
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/annotated_scRNA_data.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_Th17_dispersion_plot.pdf
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/no_bkg/
    Untracked:  output/positions.bed
    Untracked:  output/topic3/
    Untracked:  output/u19_analysis
    Untracked:  output/u19_output
    Untracked:  test.pdf

Unstaged changes:
    Modified:   analysis/U19_motif_analysis.ipynb
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/gene_set_analysis_for_RNA_topics.Rmd
    Modified:   analysis/identify_batch_effects_full_atac.Rmd
    Modified:   analysis/identify_batch_effects_multiomics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/linking_lung_celltype_OCRs_with_asthma_genetics.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    analysis/test_magma.Rmd
    Modified:   analysis/u19_h2g_enrichment.Rmd
    Modified:   analysis/u19_lung_atac_fastTopics.Rmd
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/variants_to_function_u19.Rmd</code>) and HTML
(<code>docs/variants_to_function_u19.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/d3ed9d1828635a30cc4aebdb6e277cd4a50dbcc4/analysis/variants_to_function_u19.Rmd" target="_blank">d3ed9d1</a>
</td>
<td>
Jing Gu
</td>
<td>
2025-03-27
</td>
<td>
link variants to function
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>For asthma fine-mapped regions, what cell types have open chromatin
at these regions?</p>
<ul>
<li>Shared or cell-type specific?</li>
<li>Present In lungs or spleens?</li>
</ul>
<p>For the selected regions, what genes are differentially expressed for
cell types with open chromatin?</p>
<div id="peak-selections" class="section level2">
<h2>Peak selections</h2>
<p>What we have:</p>
<ul>
<li>asthma fine-mapped variants grouped by enhancers using multiple
functional data</li>
<li>normalized CA and GE matrix (Feature x cell)
<ul>
<li>TMM and log-normalization (column-wise)</li>
<li>scaled to zscore within each tissue (row-wise)</li>
</ul></li>
</ul>
<div id="approach-one" class="section level3">
<h3>Approach One</h3>
<ul>
<li>Overlapping with confident variants that contribute to asthma risk
enhancers
<ul>
<li>confident enhancer sequences: ePIPs <span
class="math inline">\(\ge\)</span> 0.5</li>
<li>peaks need to be within 250 bp from confidently fine-mapped variants
(PIP <span class="math inline">\(\ge\)</span> 0.3)</li>
<li>average chromatin accessibility overlapping enhancers as negative
controls</li>
</ul></li>
</ul>
</div>
<div id="approach-two-selected" class="section level3">
<h3>Approach Two (selected)</h3>
<ul>
<li>Identify peaks near confidently fine-mapped variants
<ul>
<li>peaks need to be within 250 bp from confidently fine-mapped variants
(PIP <span class="math inline">\(\ge\)</span> 0.3)</li>
<li>average chromatin accessibility overlapping across a set of peaks as
negative controls
<ul>
<li>67 variants with PIP <span class="math inline">\(\ge
0.3\)</span></li>
<li>67 randomly chosen variants</li>
</ul></li>
</ul></li>
</ul>
</div>
</div>
<div id="differential-accessibility-test" class="section level2">
<h2>Differential accessibility test</h2>
<ol style="list-style-type: decimal">
<li>Wilcoxon test with ArchR</li>
</ol>
<ul>
<li>corrected for number of fragments and TSS enrichment</li>
<li>pseudobulk level test</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Negative binomial tests with EdgeR</li>
</ol>
<ul>
<li>corrected for batch effects (2 instead of 5 batches)</li>
<li>Spurious p-values occur when overly correct for batch effects</li>
</ul>
<div id="procedures" class="section level3">
<h3>Procedures</h3>
<ol style="list-style-type: decimal">
<li>Overlap our joint set of peaks with candidate enhancers</li>
<li>Compute the distances from our peaks to variants contributing to
ePIPs</li>
<li>Filter out peaks not within 250 bp from any variant contributing to
ePIPs</li>
</ol>
</div>
</div>
</div>
<div id="result" class="section level1">
<h1>Result</h1>
<div id="raw-counts" class="section level2">
<h2>Raw counts</h2>
<div
id="pearson-correlation-for-peak-counts-for-major-cell-types-across-samples"
class="section level3">
<h3>Pearson correlation for peak counts for major cell types across
samples</h3>
<ul>
<li>Peaks from the same major cell type of both tissues cluster
together</li>
<li>Major cell types in COB-8 Spleen samples have low correlation with
others</li>
</ul>
</div>
</div>
<div id="directly-overlapping-peaks-with-fine-mapped-variants"
class="section level2">
<h2>Directly overlapping peaks with fine-mapped variants</h2>
<ul>
<li>49 peaks within 250bp from 67 confidently fine-mapped SNPs</li>
<li>confidently fine-mapped SNPs defined as PIP &gt;= 0.3 for either aoa
or coa</li>
</ul>
<p>Comparing the peaks selected with two approaches:</p>
<p>A: peaks near confidently fine-mapped SNPs that fall in asthma
putitative enhancer sequences</p>
<p>B: directly overlap peaks with confidently fine-mapped SNPs</p>
<p>Approach B was selected as it includes more peaks and filter out some
peaks annotated as long intergenic non-coding RNAs.
<img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="examining-chromatin-accessibility-for-asthma-fine-mapped-regions"
class="section level2">
<h2>Examining chromatin accessibility for asthma fine-mapped
regions</h2>
<div id="heatmaps-for-accessibiltiy-across-cell-types"
class="section level3">
<h3>Heatmaps for accessibiltiy across cell types</h3>
<p><strong>Legends</strong> - Top: AOA - Bottom: COA - average
accessibility across peaks near equal number of confidently fine-mapped
SNPs and random SNPs - Y-axis: peak coordinates in hg38, nearest gene -
Chromatin accessibility normalized within each tissue
<img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-11-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>We see different patterns of open chromatin across immune
subsets:</p>
<ol style="list-style-type: decimal">
<li>shared across B cells only: LRP1, CXCR5</li>
<li>shared across NK and CD8+ T: TBX21, GAL3ST2, IRF1</li>
<li>shared across B, NK and CD8+ T</li>
<li>shared across CD4+T subsets: CCL20, CCR4</li>
<li>unique to CD4+T subset: TNFRSF11A</li>
</ol>
</div>
<div
id="heatmaps-for-accessibiltiy-across-cell-types-in-case-and-control"
class="section level3">
<h3>Heatmaps for accessibiltiy across cell types in case and
control</h3>
<p><strong>Legends</strong> - Top: AOA - Bottom: COA - average
accessibility across peaks near equal number of confidently fine-mapped
SNPs and random SNPs - Y-axis: peak coordinates in hg38, nearest gene -
Chromatin accessibility normalized within each tissue</p>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-14-1.png" width="1344" style="display: block; margin: auto;" />
Overall, chromatin levels in control are more open than those in case.
Majoriy of CD4 subsets have lower chromatin accessibility in case
samples. B, NK and CD8 T cells have more regions with open chromatin in
case samples.</p>
</div>
</div>
<div id="validate-cell-type-specific-ocrs-near-asthma-risk-enhancers"
class="section level2">
<h2>Validate cell-type specific OCRs near asthma risk enhancers</h2>
<p>Do their nearby genes show differential expression?</p>
<div id="heatmap-for-the-expression-of-cell-type-specific-genes"
class="section level3">
<h3>Heatmap for the expression of cell-type specific genes</h3>
<p>The scaling of gene expression was performed separately for each
tissue, which agrees with the differential GE analyses done for lungs
only. As a result, we see consistency in the differential expression and
the heatmap of normalized expression.</p>
<pre><code>Warning: The input is a data frame-like object, convert it to a matrix.</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div
id="overlapping-de-genes-across-cell-types-with-the-nearest-genes-from-cell-type-specific-ocrs"
class="section level3">
<h3>Overlapping DE genes across cell types with the nearest genes from
cell-type specific OCRs</h3>
<p>Cell-type specific OCRs are defined as those have higher
accessibility in CD4 T subsets.</p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a1fcd62a0dd1ba32e054" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a1fcd62a0dd1ba32e054">{"x":{"filter":"none","vertical":false,"caption":"<caption>Cell-type marker genes with cell-type specific open chromains<\/caption>","data":[["1","2","3","4","5","6"],["LINC00299","TBX21","LRP1","CXCR5","CCR4","CCL20"],[2.27,1.2,5.59,4.22,2.78,4.47],[0.333,0.243,0.104,0.179,0.129,0.303],[0.074,0.102,0.003,0.013,0.018,0.031],[" 0.0e+00","8.8e-163","1.4e-159","7.0e-153"," 3.4e-80"," 0.0e+00"],["NK","NK","Other","Naive_B","Treg","Th17"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>avg_log2FC<\/th>\n      <th>pct.1<\/th>\n      <th>pct.2<\/th>\n      <th>p_val_adj<\/th>\n      <th>cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Gene","targets":1},{"name":"avg_log2FC","targets":2},{"name":"pct.1","targets":3},{"name":"pct.2","targets":4},{"name":"p_val_adj","targets":5},{"name":"cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="test-peak-to-gene-association-across-cells"
class="section level2">
<h2>Test peak to gene association across cells</h2>
<div id="do-rna-counts-show-over-dispersion" class="section level3">
<h3>Do RNA counts show over-dispersion</h3>
<ul>
<li>40 peaks are nearby SNPs with high PIPs</li>
<li>identified 480 genes within 500KB of each selected peaks</li>
</ul>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>integer(0)</code></pre>
<p>Check distribution for the selected peaks</p>
<p>Peak levels in CPMs show overdispersion across cell types.</p>
</div>
</div>
<div id="differential-accessibility-test-between-lung-and-spleen"
class="section level2">
<h2>Differential accessibility test between lung and spleen</h2>
<div id="wilcoxon-test-with-archr" class="section level3">
<h3>wilcoxon test with ArchR</h3>
<p>Cross-tissue DA tests on peaks within asthma fine-mapped regions show
three peaks pass the threshold under FDR = 0.1.</p>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>DataFrame with 3 rows and 7 columns
       seqnames     idx     start       end    Log2FC       FDR  MeanDiff
          &lt;Rle&gt; &lt;array&gt;   &lt;array&gt;   &lt;array&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
17751      chr1   17751 172746261 172746761  0.767918 0.0610773 0.0664645
183779     chr4    6672 121696793 121697293  0.575187 0.0506949 0.0909610
183803     chr4    6696 121951558 121952058  2.366109 0.0974141 0.0833075</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="negative-binomial-test-with-edger" class="section level3">
<h3>Negative Binomial test with EdgeR</h3>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-1469b5ed4906484ed424" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1469b5ed4906484ed424">{"x":{"filter":"none","vertical":false,"caption":"<caption>Selected peaks with FDR &lt; 0.5<\/caption>","data":[["1","2","3","4","5"],["chr2:241763749-241764249_GAL3ST2","chr3:32963768-32964268_CCR4","chr10:6583286-6583786_PRKCQ-AS1","chr10:6583286-6583786_PRKCQ-AS1","chr17:47741709-47742209_TBX21"],["Memory_B","Memory_B","Memory_B","Naive_B","CD8.CD4_T"],["3.1e-03","4.5e-04","1.2e-03","6.1e-03","6.4e-03"],[0.37,0.16,0.22,0.46,0.46]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Peak<\/th>\n      <th>CellType<\/th>\n      <th>pval<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Peak","targets":1},{"name":"CellType","targets":2},{"name":"pval","targets":3},{"name":"FDR","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="examining-specific-examples" class="section level3">
<h3>Examining specific examples</h3>
<pre><code>$Memory_B</code></pre>
<pre><code>Warning: The following aesthetics were dropped during statistical transformation: label.
ℹ This can happen when ggplot fails to infer the correct grouping structure in
  the data.
ℹ Did you forget to specify a `group` aesthetic or to convert a numerical
  variable into a factor?
The following aesthetics were dropped during statistical transformation: label.
ℹ This can happen when ggplot fails to infer the correct grouping structure in
  the data.
ℹ Did you forget to specify a `group` aesthetic or to convert a numerical
  variable into a factor?
The following aesthetics were dropped during statistical transformation: label.
ℹ This can happen when ggplot fails to infer the correct grouping structure in
  the data.
ℹ Did you forget to specify a `group` aesthetic or to convert a numerical
  variable into a factor?</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-28-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
$Naive_B</code></pre>
<pre><code>Warning: The following aesthetics were dropped during statistical transformation: label.
ℹ This can happen when ggplot fails to infer the correct grouping structure in
  the data.
ℹ Did you forget to specify a `group` aesthetic or to convert a numerical
  variable into a factor?</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-28-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8.CD4_T</code></pre>
<pre><code>Warning: The following aesthetics were dropped during statistical transformation: label.
ℹ This can happen when ggplot fails to infer the correct grouping structure in
  the data.
ℹ Did you forget to specify a `group` aesthetic or to convert a numerical
  variable into a factor?</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-28-3.png" width="768" style="display: block; margin: auto;" />
Two examples of peaks in Treg
<img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>             logFC   logCPM          LR     PValue
214930 -0.75579284 3.289744 0.689471943 0.40634405
223473 -1.08669249 4.285742 2.894344486 0.08889093
148373 -0.88022832 3.601859 1.427689983 0.23214180
46817  -0.06925518 3.052186 0.005844166 0.93906336</code></pre>
</div>
</div>
<div id="differential-accessibility-test-between-case-and-control"
class="section level2">
<h2>Differential accessibility test between case and control</h2>
<div id="wilcoxon-test-with-archr-1" class="section level3">
<h3>wilcoxon test with ArchR</h3>
<p>Cross-status DA tests on peaks within asthma fine-mapped regions show
none of the peaks pass thresholds under FDR control.
<img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-68d86de4a421697d58ab" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-68d86de4a421697d58ab">{"x":{"filter":"none","vertical":false,"data":[["1","2","3"],["chr2:227805460-227805960_CCL20","chr5:132472891-132473391_IRF1","chr17:47741709-47742209_TBX21"],["Th17.CD4_T","Th17.CD4_T","Th17.CD4_T"],["3.0e-03","3.1e-03","6.9e-04"],[0.37,0.37,0.25]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Peak<\/th>\n      <th>CellType<\/th>\n      <th>pval<\/th>\n      <th>adjP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Peak","targets":1},{"name":"CellType","targets":2},{"name":"pval","targets":3},{"name":"adjP","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="show-examples-of-peaks-passing-threshold"
class="section level3">
<h3>Show examples of peaks passing threshold</h3>
<pre><code># A tibble: 3 × 4
  Peak                           CellType       pval  adjP
  &lt;chr&gt;                          &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;
1 chr2:227805460-227805960_CCL20 Th17.CD4_T 0.00299  0.372
2 chr5:132472891-132473391_IRF1  Th17.CD4_T 0.00310  0.372
3 chr17:47741709-47742209_TBX21  Th17.CD4_T 0.000694 0.250</code></pre>
<pre><code>$Th17.CD4_T</code></pre>
<p><img src="figure/variants_to_function_u19.Rmd/unnamed-chunk-34-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] mapgen_0.5.9                      org.Hs.eg.db_3.16.0              
 [3] GenomicInteractions_1.32.0        InteractionSet_1.26.1            
 [5] Gviz_1.42.1                       GenomicFeatures_1.50.4           
 [7] AnnotationDbi_1.60.2              circlize_0.4.15                  
 [9] TidyDensity_1.5.0                 eulerr_7.0.2                     
[11] ComplexHeatmap_2.14.0             BSgenome.Hsapiens.UCSC.hg38_1.4.5
[13] BSgenome_1.66.3                   Biostrings_2.66.0                
[15] XVector_0.38.0                    ggrepel_0.9.6                    
[17] edgeR_3.40.2                      limma_3.58.0                     
[19] scales_1.3.0                      colorRamp2_0.1.0                 
[21] rtracklayer_1.58.0                tidyr_1.3.1                      
[23] dplyr_1.1.4                       rhdf5_2.42.1                     
[25] SummarizedExperiment_1.28.0       Biobase_2.58.0                   
[27] MatrixGenerics_1.10.0             Rcpp_1.0.14                      
[29] Matrix_1.6-5                      GenomicRanges_1.50.2             
[31] GenomeInfoDb_1.34.9               IRanges_2.32.0                   
[33] S4Vectors_0.36.2                  BiocGenerics_0.44.0              
[35] matrixStats_1.5.0                 data.table_1.17.0                
[37] stringr_1.5.1                     plyr_1.8.9                       
[39] magrittr_2.0.3                    ggplot2_3.5.1                    
[41] gtable_0.3.6                      gtools_3.9.5                     
[43] gridExtra_2.3                     ArchR_1.0.2                      
[45] workflowr_1.7.1                  

loaded via a namespace (and not attached):
  [1] utf8_1.2.4               tidyselect_1.2.1         RSQLite_2.3.9           
  [4] htmlwidgets_1.6.4        BiocParallel_1.32.6      munsell_0.5.1           
  [7] codetools_0.2-20         interp_1.1-6             statmod_1.5.0           
 [10] DT_0.33                  withr_3.0.2              colorspace_2.1-1        
 [13] filelock_1.0.3           knitr_1.50               rstudioapi_0.17.1       
 [16] labeling_0.4.3           git2r_0.33.0             GenomeInfoDbData_1.2.9  
 [19] polyclip_1.10-7          bit64_4.0.5              farver_2.1.2            
 [22] rprojroot_2.0.4          vctrs_0.6.5              generics_0.1.3          
 [25] xfun_0.51                biovizBase_1.46.0        BiocFileCache_2.6.1     
 [28] R6_2.6.1                 doParallel_1.0.17        clue_0.3-66             
 [31] locfit_1.5-9.12          AnnotationFilter_1.22.0  bitops_1.0-9            
 [34] rhdf5filters_1.10.1      cachem_1.1.0             DelayedArray_0.24.0     
 [37] promises_1.3.2           BiocIO_1.8.0             nnet_7.3-20             
 [40] Cairo_1.6-2              processx_3.8.6           ensembldb_2.22.0        
 [43] rlang_1.1.5              GlobalOptions_0.1.2      lazyeval_0.2.2          
 [46] dichromat_2.0-0.1        checkmate_2.3.2          yaml_2.3.10             
 [49] crosstalk_1.2.1          backports_1.4.1          httpuv_1.6.15           
 [52] Hmisc_5.2-3              tools_4.2.0              jquerylib_0.1.4         
 [55] RColorBrewer_1.1-3       base64enc_0.1-3          progress_1.2.3          
 [58] zlibbioc_1.44.0          purrr_1.0.4              RCurl_1.98-1.17         
 [61] ps_1.9.0                 prettyunits_1.2.0        rpart_4.1.24            
 [64] deldir_2.0-4             GetoptLong_1.0.5         cluster_2.1.8.1         
 [67] fs_1.6.5                 whisker_0.4.1            ProtGenerics_1.34.0     
 [70] hms_1.1.3                evaluate_1.0.3           XML_3.99-0.18           
 [73] jpeg_0.1-11              shape_1.4.6              compiler_4.2.0          
 [76] biomaRt_2.54.1           tibble_3.2.1             crayon_1.5.3            
 [79] htmltools_0.5.8.1        later_1.4.1              Formula_1.2-5           
 [82] DBI_1.2.3                dbplyr_2.5.0             rappdirs_0.3.3          
 [85] cli_3.6.4                parallel_4.2.0           igraph_2.1.4            
 [88] pkgconfig_2.0.3          getPass_0.2-2            GenomicAlignments_1.34.1
 [91] foreign_0.8-88           xml2_1.3.8               foreach_1.5.2           
 [94] bslib_0.9.0              VariantAnnotation_1.44.1 callr_3.7.3             
 [97] digest_0.6.37            rmarkdown_2.29           polylabelr_0.3.0        
[100] htmlTable_2.4.3          restfulr_0.0.15          curl_6.2.1              
[103] Rsamtools_2.14.0         rjson_0.2.23             lifecycle_1.0.4         
[106] jsonlite_1.9.1           Rhdf5lib_1.20.0          pillar_1.10.1           
[109] lattice_0.22-6           KEGGREST_1.38.0          fastmap_1.2.0           
[112] httr_1.4.7               glue_1.8.0               png_0.1-8               
[115] iterators_1.0.14         bit_4.6.0                stringi_1.8.4           
[118] sass_0.4.9               blob_1.2.4               latticeExtra_0.6-30     
[121] memoise_2.0.1            tidyverse_2.0.0         </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
