---
title: "A testrun for MAGMA"
output: html_document
date: '`r Sys.Date()`'
author: 'Jing Gu'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(data.table)
library(dplyr)
workdir <- "/home/jinggu/cluster/data/MAGMA/testrun"
```

# MAGMA

**Gene analysis**
A linear principal component regression model that estimates whether there is genetic effect of gene g on the phenotype Y, conditional on all covariates. The model first projects genotype matrix for a gene g onto its PCs, pruning away PCs with very small eigenvalues. Then it performs F test in the regression of Y on SNP matrix and covariates to estimate genetic effect.

$$
  Y = \alpha_{0g}\vec 1 + X_g^*\alpha_g  + W\beta_g + \epsilon_g
$$
When inividual geneotype matrix not available, MAGMA performs gene test with mean $X^2$ statistics and a gene p-value is then obtained by using a known approximation of the sampling distribution. Please refer to the following paper for details of approximation for the distribution of the weighted combination of p-values. This model requires summary statistics and reference LD panel.

Ref: Hou C (2005) A simple approximation for the distribution of the weighted combination of non-independent or independent probabilities. Stat Probabil Lett 73: 179â€“187.

**Competitive gene-set analysis**

One-sided Two-sample T test or linear regression in equivalence is applied to test whether the genes in a gene set are more strongly associated with Y or not. 

Let Z denote the association z-score. 
Let $\S_s$ be an indicator variable with element $s_g = 1$ defined as for gene g in gene set s and 0 otherwise. The goal is to test whether $\beta_s$ is greater than zero, which represents the difference in association between genes in the gene set and genes outside the gene set.

$$
Z = \beta_{0s}\vec 1 + S_s\beta_s + \epsilon
$$
This also be tested by unpaired two sample T-test, while two samples can have unequal variances and sample sizes. 

## Testrun

Procedure: 

1. annotate SNPs and genes
2. gene-based analysis
3. gene-set analysis
```{bash, eval = FALSE}
#zcat BBJ_HDLC.txt.gz | awk 'NR>1 && $2==3 {print $1,$2,$3}' > HDLC_chr3.magma.input.snp.chr.pos.txt
#zcat BBJ_HDLC.txt.gz | awk 'NR>1 && $2==3 {print $1,10^(-$11)}' >  HDLC_chr3.magma.input.p.txt
cd /home/jinggu/cluster/data/MAGMA/testrun
traitID="ukb-a-446"
module load bcftools
# extract SNPs and p-values
bcftools query -f'%ID %CHROM %POS\n' /project/xinhe/shared/IEU_vcf/$traitID.vcf.gz > ./${traitID}_snploc.txt
bcftools query -f'%ID [%LP]\n' /project/xinhe/shared/IEU_vcf/$traitID.vcf.gz > ./${traitID}_pval.txt
```

```{r eval = FALSE}
# convert p-values to absolute value
f <- fread(paste0(workdir, "/ukb-a-446_pval.txt"))
f$V2 <- 10^(-f$V2)
fwrite(f, paste0(workdir, "/ukb-a-446_input_p.txt"), 
       sep = " ", col.names = FALSE)
# subset coding genes to be the input genes for topic modeling
fit<-readRDS("output/fastTopics/fit_model_updates12_k150.RDS")
allGenes <- fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
subGenes <- allGenes[allGenes$V6 %in% rownames(fit$F), ]
write.table(subGenes, "/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc",
            quote = F, row.names = F, sep = "\t")
```

```{bash eval = FALSE}
cd /home/jinggu/cluster/data/MAGMA/testrun
traitID="ukb-a-446"
ncbi37=/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc
source ~/.bashrc
# annotate SNPs
snploc=${traitID}_snploc.txt
magma --annotate \
      --snp-loc $snploc \
      --gene-loc $ncbi37 \
      --out $traitID
      
# gene-based analysis
ref=/home/jinggu/cluster/data/MAGMA/reference_data/g1000_eur
magma \
    --bfile $ref \
    --pval ${traitID}_input_p.txt N=336782 \
    --gene-annot $traitID.genes.annot \
    --out $traitID
    
# gene-set analysis

```

```{r eval = FALSE}
de_list<-readRDS("output/fastTopics/GoM_topDE_gene_list.RDS")
names(de_list) <- paste0("k", 1:12)
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
for(i in 1:length(de_list)){
  gene_set <- matrix(c(names(de_list)[i], ncbi37$V1[ncbi37$V6 %in% rownames(de_list[[i]])]),
                     nrow = 1)
  write.table(gene_set,
              sprintf("%s/%s_geneset.txt", workdir, names(de_list)[i]), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
  

```

```{bash eval = FALSE}
source ~/.bashrc
traitID="ukb-a-446"
geneset=top100_geneset.txt
magma \
    --gene-results $traitID.genes.raw \
    --set-annot ${geneset} \
    --out ${traitID}_top100

```
## Gene-set associations for topic marker genes that are up-regulated

```{r}
f <- read.table(paste0(workdir, "/all_genes/ukb-a-446.gsa.out"), skip = 3, header = T)
#DT::datatable(f)
```
When corrected for multiple testing, tests will be significant if p-value lower than ~0.005. Around half of the tests show significant p-values, which makes us wonder if p-values are inflated. Then we try using the input genes for topic modeling rather than all genes as background so that they are more comparable. 

* Total number of genes reduced from to 18K to ~16K
* P values for the reduced background genes are very similar to the full ones.
```{r}
f_reduced <- read.table(paste0(workdir, "/ukb-a-446_topics.gsa.out"), skip = 3, header = T)
DT::datatable(f_reduced)

plot(y = -log10(f_reduced$P), 
     x = -log10(f$P),
     ylab = "-log10(P) for input genes for topic modeling (~16K)",
     xlab = "-log10(P) for all genes (~18K)")
abline(a = 0, b = 1, col = "red")
```

### Test enrichment using top 100 genes from each topic

The supplementary table from MAGMA paper shows the mean type 1 error rates are well controlled for a set of size 100. The [MSigDB canonical pathways](https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=CP) contains 1320 gene sets from a number of different databases. I can look into the average size of the gene sets.
![MAGMA controlling for type 1 error rates for gene-set analysis ](https://github.com/user-attachments/assets/e4ea0389-c993-46a1-bc6b-c2a69630addf)

```{r eval = FALSE}
de_list<-readRDS("output/fastTopics/GoM_topDE_gene_list.RDS")
names(de_list) <- paste0("k", 1:12)
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
for(i in 1:length(de_list)){
  top100 <- de_list[[i]] %>% arrange(desc(z)) %>% slice_head(n=100)
  gene_set <- matrix(c(names(de_list)[i], 
                       ncbi37$V1[ncbi37$V6 %in% rownames(top100)]),
                     nrow = 1)
  write.table(gene_set,
              sprintf("%s/%s_top100_geneset.txt", workdir, names(de_list)[i]), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
```

Instead of all DE genes, I used top 100 up-regulated genes ranked by z scores to test the enrichment for each topic. Now we see only k3, k4, k5 and k12 (4/12) topics show significant enrichment after multiple testing correction.
```{r}
df_top <- read.table(paste0(workdir, "/ukb-a-446_top100.gsa.out"), 
                     skip = 3, header = T)
DT::datatable(df_top)
```
### Check the top genes contributing to topic 5 that have high MAGMA Z-scores

Genes with high Z-scores (p < 0.05) are found to show enrichment in the following gene sets:

* Regulation of signal transduction (adj.P = 0.01) from GO
  - RORA, GATA3, CDC42SE2, SREBF2
* Interleukin-2 signaling pathway (adj.P = 0.003) from BioPlanet
  - CD247, RORA, GATA3, TAB2, FOXO1,IL21R, YARS (overlapping genes)
* Th17 Cell Differentiation (adj.P = 0.0002) and IBD (adj.P = 0.001) from KEGG
  - CD247, RORA, GATA3,IL21R

```{r}
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc")

f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k5.genes.out"),
                header = TRUE)
geneInfo <- ncbi37[, c(1, 6)]
colnames(geneInfo) <- c("GENE", "SYMBOL")
f <- left_join(f, geneInfo)
DT::datatable(
  f  %>% arrange(P) %>%
    mutate(P =format(P, format = "e", digits = 3)) %>%
    select(SYMBOL, CHR, START, STOP, ZSTAT, P) 
)
```
### Check the top genes contributing to topic 3 that have high MAGMA Z-scores

Genes with high Z-scores (p < 0.05) are found to show enrichment in the following GO terms:

1	Alpha-Beta T Cell Activation (GO:0046631)	0.00007389	0.007163	204.61	1946.45.
2	Regulation Of Natural Killer Cell Activation (GO:0032814)	0.0001340	0.007163	146.12	1303.03.   
3	Regulation Of Lymphocyte Activation (GO:0051249)	0.0001706	0.007163	127.84	1109.13.    
4	Stimulatory C-type Lectin Receptor Signaling Pathway (GO:0002223)	0.0001906	0.007163	120.31	1030.52.    
5	Cellular Response To Lectin (GO:1990858)	0.0001906	0.007163	120.31	1030.52.     
6	Antigen Receptor-Mediated Signaling Pathway (GO:0050851)	0.0002146	0.007163	30.59	258.42.     
7	Positive Regulation Of Natural Killer Cell Mediated Cytotoxicity (GO:0045954)	0.0002337	0.007163	107.63	899.96.    
8	Natural Killer Cell Mediated Immunity (GO:0002228)	0.0002570	0.007163	102.25	845.24.      
9	Positive Regulation Of Natural Killer Cell Mediated Immunity (GO:0002717)	0.0003066	0.007597	92.94	751.89.     
10	Innate Immune Response Activating Cell Surface Receptor Signaling Pathway (GO:0002220)	0.0003892	0.008678	81.77	642.05.      
```{r}
f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k3.genes.out"),
                header = TRUE)
geneInfo <- ncbi37[, c(1, 6)]
colnames(geneInfo) <- c("GENE", "SYMBOL")
f <- left_join(f, geneInfo)
DT::datatable(
  f  %>% arrange(P) %>%
    mutate(P =format(P, format = "e", digits = 3)) %>%
    select(SYMBOL, CHR, START, STOP, ZSTAT, P) 
)

# write.table(data.frame(f %>% filter(P < 0.05) %>% pull(SYMBOL)),
#             "../output/fastTopics/k4_asthma_gsa_signif_genes.txt",
#             row.names = F, col.names = F, quote = F)
```
### Check the top genes contributing to topic 4 that have high MAGMA Z-scores

Genes with high Z-scores (p < 0.05) are found to show enrichment in the following GO terms:

1	Regulation Of Blood Vessel Endothelial Cell Migration (GO:0043535) | 0.0006701 | AKT3, PRKCA, TNF, ETS1.       
2	Antigen Receptor-Mediated Signaling Pathway (GO:0050851)	| 0.006130 | CD28, BCL2, BTN3A3, SKAP1.        
3	Positive Regulation Of Leukocyte Cell-Cell Adhesion (GO:1903039)|	0.006130 | TNF, ETS1, SKAP1.       

```{r}
f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k4.genes.out"),
                header = TRUE)
geneInfo <- ncbi37[, c(1, 6)]
colnames(geneInfo) <- c("GENE", "SYMBOL")
f <- left_join(f, geneInfo)
DT::datatable(
  f  %>% arrange(P) %>%
    mutate(P =format(P, format = "e", digits = 3)) %>%
    select(SYMBOL, CHR, START, STOP, ZSTAT, P) 
)

# write.table(data.frame(f %>% filter(P < 0.05) %>% pull(SYMBOL)),
#             "../output/fastTopics/k4_asthma_gsa_signif_genes.txt",
#             row.names = F, col.names = F, quote = F)
```


## Gene-set associations for known immune cell markers

Most immune cell markers were obtained from Gene ontology. The tissue-resident genes were from the single-cell lung paper and ChatGPT. After multiple correction, tissue-resident markers do not show significance for the gene-set association with Asthma. Neither do the genes involved in cell-cell adhesion or cell migration. The gene sets that are significantly enriched for higher association with Asthma risks are activation of T/B cells and differentiation of T cell lineage differentiation.
```{bash eval = FALSE}
cd /home/jinggu/cluster/data/MAGMA/testrun
source ~/.bashrc
traitID="ukb-a-446"
geneset=gene_sets/GO_gene_sets_MAGMA.txt
magma \
    --gene-results $traitID.genes.raw \
    --set-annot ${geneset} \
    --out ${traitID}_GO_enriched

```

```{r}
df_enriched <- read.table(paste0(workdir, "/ukb-a-446_GO_enriched.gsa.out"),
                    skip = 4, header = TRUE)
tbl <-
  rbind(
    df_top %>%
      filter(NGENES > 10 & P < 0.05/12) %>% # correct for multiple testing
      select(VARIABLE, NGENES, BETA, SE, P),
    df_enriched %>% mutate(VARIABLE = FULL_NAME) %>%
      filter(NGENES > 10) %>% #& P < 0.05/33) %>% # correct for multiple testing 0.05/33 = 0.002
      select(VARIABLE, NGENES, BETA, SE, P) %>% 
      arrange(P) %>% mutate(P = format(P, scientific = TRUE, digits = 2))
  )
DT::datatable(tbl)
#write.table(tbl, "../output/fastTopics/top100_topic_GO_immune.gsa.txt", 
            #quote = F, row.names = F, sep = "\t")
```

```{r eval = FALSE}
### Perform permutation to obtain null p-values
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc")
B = 5000
for(i in 1:B){
  gene_set = sample(ncbi37$V1, 700, replace = FALSE)
  write.table(matrix(c(paste0("set", i), gene_set), nrow = 1),
              sprintf("%s/set%s_geneset.txt", workdir, i), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
```

```{r eval = FALSE}
f_null <- read.table(paste0(workdir, "/ukb-a-446_permutated700.gsa.out"), 
                     skip = 3, header = T)

f_perm <- unlist(lapply(f_reduced$BETA, function(i){sum(f_null$BETA >= i)/5000}))
  
plot(y = -log10(f_reduced$P), 
     x = -log10(f_perm), 
     xlab = "-log10(P) after permutation",
     ylab = "-log10(P) before permutation")
abline(a = 0, b = 1, col = "red")
```

