---
title: "A testrun for MAGMA"
output: html_document
date: '`r Sys.Date()`'
author: 'Jing Gu'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(data.table)
library(dplyr)
library(tidyr)
workdir <- "/home/jinggu/cluster/data/MAGMA"
```

# MAGMA

**Gene analysis**
A linear principal component regression model that estimates whether there is genetic effect of gene g on the phenotype Y, conditional on all covariates. The model first projects genotype matrix for a gene g onto its PCs, pruning away PCs with very small eigenvalues. Then it performs F test in the regression of Y on SNP matrix and covariates to estimate genetic effect.

$$
  Y = \alpha_{0g}\vec 1 + X_g^*\alpha_g  + W\beta_g + \epsilon_g
$$
When inividual geneotype matrix not available, MAGMA performs gene test with mean $X^2$ statistics and a gene p-value is then obtained by using a known approximation of the sampling distribution. Please refer to the following paper for details of approximation for the distribution of the weighted combination of p-values. This model requires summary statistics and reference LD panel.

Ref: Hou C (2005) A simple approximation for the distribution of the weighted combination of non-independent or independent probabilities. Stat Probabil Lett 73: 179â€“187.

**Competitive gene-set analysis**

One-sided Two-sample T test or linear regression in equivalence is applied to test whether the genes in a gene set are more strongly associated with Y or not. 

Let Z denote the association z-score. 
Let $\S_s$ be an indicator variable with element $s_g = 1$ defined as for gene g in gene set s and 0 otherwise. The goal is to test whether $\beta_s$ is greater than zero, which represents the difference in association between genes in the gene set and genes outside the gene set.

$$
Z = \beta_{0s}\vec 1 + S_s\beta_s + \epsilon
$$
This also be tested by unpaired two sample T-test, while two samples can have unequal variances and sample sizes. 

## Testrun

Procedure: 

1. annotate SNPs and genes
2. gene-based analysis
3. gene-set analysis
```{bash, eval = FALSE}
#zcat BBJ_HDLC.txt.gz | awk 'NR>1 && $2==3 {print $1,$2,$3}' > HDLC_chr3.magma.input.snp.chr.pos.txt
#zcat BBJ_HDLC.txt.gz | awk 'NR>1 && $2==3 {print $1,10^(-$11)}' >  HDLC_chr3.magma.input.p.txt
cd /home/jinggu/cluster/data/MAGMA/results
#traitID=ukb-a-446
module load bcftools
# extract SNPs and p-values
for i in {ukb-d-PNEUMONIA,} #ukb-a-446,ukb-b-20464,ukb-a-447}
do
echo $i
bcftools query -f'%ID %CHROM %POS\n' /project/xinhe/shared/IEU_vcf/$i.vcf.gz > ./$i/${i}_snploc.txt
bcftools query -f'%ID [%LP]\n' /project/xinhe/shared/IEU_vcf/$i.vcf.gz > ./$i/${i}_pval.txt
bcftools query -f'[%SS]\n' /project/xinhe/shared/IEU_vcf/$i.vcf.gz |uniq > ./$i/${i}_ss.txt
done
```

```{r eval = FALSE}
workdir="/home/jinggu/cluster/data/MAGMA"
for(i in c("ukb-b-20464", "ukb-a-447", "ukb-d-PNEUMONIA")) {
  # convert p-values to absolute value
  f <- fread(paste0(workdir, sprintf("/results//%s/%s_pval.txt", i, i)))
  f$V2 <- 10 ^ (-f$V2)
  fwrite(f,
         paste0(workdir, sprintf("/results/%s/%s_input_p.txt", i, i)),
         sep = " ",
         col.names = FALSE)
}
# subset coding genes to be the input genes for topic modeling
fit <- readRDS("output/fastTopics/fit_model_updates12_k150.RDS")
allGenes <-
  fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
subGenes <- allGenes[allGenes$V6 %in% rownames(fit$F),]
write.table(
  subGenes,
  "/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc",
  quote = F,
  row.names = F,
  sep = "\t"
)

```

```{bash eval = FALSE}
traitID="ukb-d-PNEUMONIA"
cd /home/jinggu/cluster/data/MAGMA/results/$traitID
ncbi37=/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc
source ~/.bashrc
# annotate SNPs
snploc=${traitID}_snploc.txt
magma --annotate \
      --snp-loc $snploc \
      --gene-loc $ncbi37 \
      --out $traitID
      
# gene-based analysis
ref=/home/jinggu/cluster/data/MAGMA/reference_data/g1000_eur
magma \
    --bfile $ref \
    --pval ${traitID}_input_p.txt N=361194 \
    --gene-annot $traitID.genes.annot \
    --out $traitID
    
# gene-set analysis

```

```{r eval = FALSE}
de_list<-readRDS("output/fastTopics/GoM_topDE_gene_list.RDS")
names(de_list) <- paste0("k", 1:12)
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
for(i in 1:length(de_list)){
  gene_set <- matrix(c(names(de_list)[i], ncbi37$V1[ncbi37$V6 %in% rownames(de_list[[i]])]),
                     nrow = 1)
  write.table(gene_set,
              sprintf("%s/%s_geneset.txt", workdir, names(de_list)[i]), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
  

```

```{bash eval = FALSE}
source ~/.bashrc
cd /home/jinggu/cluster/data/MAGMA
traitID="ukb-a-446"
geneset=/home/jinggu/cluster/data/MAGMA/testrun/gene_sets/top100_geneset.txt
for i in {ukb-d-PNEUMONIA,} #ukb-a-447,ukb-b-20464}
do
magma \
    --gene-results results/$i/$i.genes.raw \
    --set-annot ${geneset} \
    --out results/$i/${i}_top100
done
```
## Gene-set associations with risks for immune diseases

### Testing on all marker genes 

When corrected for multiple testing, tests will be significant if p-value lower than ~0.005. Around half of the tests show significant p-values, which makes us wonder if p-values are inflated. Then we try using the input genes for topic modeling rather than all genes as background so that they are more comparable. 

* Total number of genes reduced from to 18K to ~16K
* P values for the reduced background genes are very similar to the full ones.
```{r}
f <- read.table(paste0(workdir, "/results/ukb-a-446/all_genes/ukb-a-446.gsa.out"), skip = 3, header = T)
f_reduced <- read.table(paste0(workdir, "/results/ukb-a-446/ukb-a-446_topics.gsa.out"), skip = 3, header = T)
DT::datatable(f_reduced)

plot(y = -log10(f_reduced$P), 
     x = -log10(f$P),
     ylab = "-log10(P) for input genes for topic modeling (~16K)",
     xlab = "-log10(P) for all genes (~18K)")
abline(a = 0, b = 1, col = "red")
```

### Testing on top 100 genes from each topic

The supplementary table from MAGMA paper shows the mean type 1 error rates are well controlled for a set of size 100. The [MSigDB canonical pathways](https://www.gsea-msigdb.org/gsea/msigdb/human/genesets.jsp?collection=CP) contains 1320 gene sets from a number of different databases. I can look into the average size of the gene sets.
![MAGMA controlling for type 1 error rates for gene-set analysis ](https://github.com/user-attachments/assets/e4ea0389-c993-46a1-bc6b-c2a69630addf)

```{r eval = FALSE}
de_list<-readRDS("output/fastTopics/GoM_topDE_gene_list.RDS")
names(de_list) <- paste0("k", 1:12)
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/NCBI37.3.gene.loc")
for(i in 1:length(de_list)){
  top100 <- de_list[[i]] %>% arrange(desc(z)) %>% slice_head(n=100)
  gene_set <- matrix(c(names(de_list)[i], 
                       ncbi37$V1[ncbi37$V6 %in% rownames(top100)]),
                     nrow = 1)
  write.table(gene_set,
              sprintf("%s/%s_top100_geneset.txt", workdir, names(de_list)[i]), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
```
### Summarizing enrichment results by traits

Instead of all DE genes, I used top 100 up-regulated genes ranked by z scores to test the enrichment for each topic across immune traits. For Asthma, now we see only k3, k4, k5 and k12 (4 out of 12) topics show significant enrichment after multiple testing correction. For other immune diseases, allergy displays very strong enrichment for k4, k5 and k12, but none of the others have enrichment signal. 

```{r}
# A function to plot heatmap
library(ComplexHeatmap)
library(colorRamp2)
#enrich_out<-readRDS("output/fastTopics/GO_enrichment_results_GoM_DE.RDS")

plot_heatmap <- 
  function(input_tbl, 
           cluster_by_rows=FALSE, 
           cluster_by_cols=FALSE,
           max_colour_value=NULL){
    if(is.null(max_colour_value)){
      max_colour_value = max(input_mat, na.rm = TRUE)
    }
  col_fun = 
    colorRamp2(c(
      min(input_mat, na.rm = TRUE),
      max_colour_value), 
      c("white", "red"))
  ht <- 
  Heatmap(
    t(input_tbl),
    width=unit(10, "cm"),
    name = "-log10(P)",
    row_names_side = "left",
    cluster_rows = cluster_by_rows,
    cluster_columns = cluster_by_cols,
    na_col = "black",
    col = col_fun,
    #column_title = sprintf("%s", i)
  )
  draw(ht)
}
```

```{r}
traits <- c("Asthma", "Allergy", "COPD", "Asthma/COPD")
traitIDs <- c("ukb-a-446", "ukb-a-447", "ukb-b-20464", "ukb-d-PNEUMONIA")
tbl <- c()
for(trait in traitIDs) {
  df_top <- read.table(
    paste0(workdir, "/results/", 
           trait,"/",
           trait, "_top100.gsa.out"),
    skip = 3,
    header = T
  )
  tbl <- rbind(
    tbl,
    df_top %>% 
      mutate(TRAIT = trait) %>%
      select(-c(TYPE, NGENES))
  )
}
DT::datatable(
  tbl %>%  
    mutate(TRAIT = factor(TRAIT, labels = traits)) %>%
    select(TRAIT, VARIABLE, SE, P)
  )

input_mat <-
  data.frame(
    tbl %>%
      mutate(P = -log10(P),
             TRAIT = factor(TRAIT,
                            labels = traits)) %>%
      select(c(VARIABLE, TRAIT, P)) %>%
      pivot_wider(names_from = TRAIT,
                  values_from = P),
    row.names = 1
  )

plot_heatmap(
  input_mat, 
  cluster_by_rows = TRUE,
  cluster_by_cols = TRUE)
```




<!-- Genes with high Z-scores (p < 0.05) are found to show enrichment in the following gene sets: -->

<!-- * Regulation of signal transduction (adj.P = 0.01) from GO -->
<!--   - RORA, GATA3, CDC42SE2, SREBF2 -->
<!-- * Interleukin-2 signaling pathway (adj.P = 0.003) from BioPlanet -->
<!--   - CD247, RORA, GATA3, TAB2, FOXO1,IL21R, YARS (overlapping genes) -->
<!-- * Th17 Cell Differentiation (adj.P = 0.0002) and IBD (adj.P = 0.001) from KEGG -->
<!--   - CD247, RORA, GATA3,IL21R -->

<!-- ![image](https://github.com/user-attachments/assets/8516bf22-cffa-490e-837b-1624e381ebd4) -->

<!-- ```{r} -->
<!-- ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc") -->

<!-- f <- read.table(paste0(workdir, "/results/ukb-a-446/ukb-a-446_top100.gsa.k5.genes.out"), -->
<!--                 header = TRUE) -->
<!-- geneInfo <- ncbi37[, c(1, 6)] -->
<!-- colnames(geneInfo) <- c("GENE", "SYMBOL") -->
<!-- f <- left_join(f, geneInfo) -->
<!-- DT::datatable( -->
<!--   f  %>% arrange(P) %>% -->
<!--     mutate(P =format(P, format = "e", digits = 3)) %>% -->
<!--     select(SYMBOL, CHR, START, STOP, ZSTAT, P)  -->
<!-- ) -->
<!-- ``` -->
<!-- ### Check the top genes contributing to topic 3 that have high MAGMA Z-scores -->

<!-- Genes with high Z-scores (p < 0.05) are found to show enrichment in the following GO terms: -->

<!-- ![image](https://github.com/user-attachments/assets/b5c0dd96-211b-4b78-9851-17a39fa294bf) -->

<!-- ```{r} -->
<!-- f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k3.genes.out"), -->
<!--                 header = TRUE) -->
<!-- geneInfo <- ncbi37[, c(1, 6)] -->
<!-- colnames(geneInfo) <- c("GENE", "SYMBOL") -->
<!-- f <- left_join(f, geneInfo) -->
<!-- DT::datatable( -->
<!--   f  %>% arrange(P) %>% -->
<!--     mutate(P =format(P, format = "e", digits = 3)) %>% -->
<!--     select(SYMBOL, CHR, START, STOP, ZSTAT, P)  -->
<!-- ) -->

<!-- # write.table(data.frame(f %>% filter(P < 0.05) %>% pull(SYMBOL)), -->
<!-- #             "../output/fastTopics/k3_asthma_gsa_signif_genes.txt", -->
<!-- #             row.names = F, col.names = F, quote = F) -->
<!-- ``` -->
<!-- ### Check the top genes contributing to topic 4 that have high MAGMA Z-scores -->

<!-- Genes with high Z-scores (p < 0.05) are found to show enrichment in the following GO terms: -->

<!-- 1	Regulation Of Blood Vessel Endothelial Cell Migration (GO:0043535) | 0.0006701 | AKT3, PRKCA, TNF, ETS1.        -->
<!-- 2	Antigen Receptor-Mediated Signaling Pathway (GO:0050851)	| 0.006130 | CD28, BCL2, BTN3A3, SKAP1.         -->
<!-- 3	Positive Regulation Of Leukocyte Cell-Cell Adhesion (GO:1903039)|	0.006130 | TNF, ETS1, SKAP1.    -->

<!-- 4 Regulation Of Cytokine Production (GO:0001817) | 0.013 | TNF, CD28, CCR7, BTN3A3 -->

<!-- ![image](https://github.com/user-attachments/assets/e9d707f2-1325-45d5-948b-f0ceccb3ab20) -->

<!-- ```{r} -->
<!-- f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k4.genes.out"), -->
<!--                 header = TRUE) -->
<!-- geneInfo <- ncbi37[, c(1, 6)] -->
<!-- colnames(geneInfo) <- c("GENE", "SYMBOL") -->
<!-- f <- left_join(f, geneInfo) -->
<!-- DT::datatable( -->
<!--   f  %>% arrange(P) %>% -->
<!--     mutate(P =format(P, format = "e", digits = 3)) %>% -->
<!--     select(SYMBOL, CHR, START, STOP, ZSTAT, P)  -->
<!-- ) -->

<!-- # write.table(data.frame(f %>% filter(P < 0.05) %>% pull(SYMBOL)), -->
<!-- #             "../output/fastTopics/k4_asthma_gsa_signif_genes.txt", -->
<!-- #             row.names = F, col.names = F, quote = F) -->
<!-- ``` -->

<!-- ### Check the top genes contributing to topic 12 that have high MAGMA Z-scores -->

<!-- Genes with high Z-scores (p < 0.05) are found to show enrichment in the following GO terms: -->
<!-- ![image](https://github.com/user-attachments/assets/8b067355-0abf-4eb6-8a69-8f5917a42b21) -->
<!-- ```{r} -->
<!-- f <- read.table(paste0(workdir, "/ukb-a-446_top100_gsa.k12.genes.out"), -->
<!--                 header = TRUE) -->
<!-- geneInfo <- ncbi37[, c(1, 6)] -->
<!-- colnames(geneInfo) <- c("GENE", "SYMBOL") -->
<!-- f <- left_join(f, geneInfo) -->
<!-- DT::datatable( -->
<!--   f  %>% arrange(P) %>% -->
<!--     mutate(P =format(P, format = "e", digits = 3)) %>% -->
<!--     select(SYMBOL, CHR, START, STOP, ZSTAT, P)  -->
<!-- ) -->

<!-- # write.table(data.frame(f %>% filter(P < 0.05) %>% pull(SYMBOL)), -->
<!-- #             "../output/fastTopics/k12_asthma_gsa_signif_genes.txt", -->
<!-- #             row.names = F, col.names = F, quote = F) -->
<!-- ``` -->
## Assess topic enrichment for immune marker gene sets

Only K5 satisfies the following criteria: 

1. higher proportion in lungs than spleens  
2. significant MAGMA Z scores

Immune marker gene sets:

1. GO terms for T helper cell differentiation, immune response (Th2, Th17, Treg)     
2. ChatGPT: Th2 response genes, tissue-resident cell markers for lung lymphocytes     
3. Tissue-resident lymphocyte markers: S1PR1, LMNA, SELL, RGS1, KLRG1, CD69, DUSP6, RGCC, SOCS1, ITGAE, CCR7, CTLA4, PDCD1, IL2RB, ITGA1, CXCR3


```{r}
# get topic specific genes
de<-readRDS("output/fastTopics/GoM_DE_u19.RDS")
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc")
de_list<-list()
for(k in 1:12){
  dat <- data.frame(postmean = de$postmean[,k],
                  z        = de$z[,k],
                  lfsr     = de$lfsr[,k])
  
  dat <- subset(dat,lfsr < 0.01)
  dat <- dat[order(dat$postmean,decreasing = TRUE),]
  de_list[[k]]<-dat
}
topic5 <- de_list[[5]]
```

### Testing on all up-regulated topic marker genes

Enrichment test: Fisher's exact test

Topic 5 specific genes are enriched for tissue-resident genes, Treg/Th17 differentiation, and Th17 immune response.

```{r}
test_topic_enrichment <-
  function(topic_genes,
           gene_sets, 
           bkg_genes) {
    enrichTest <-
      sapply(1:length(gene_sets), function(i) {
        overlaps <- intersect(topic_genes, gene_sets[[i]])
        numOverlaps <- sum(topic_genes %in% gene_sets[[i]])
        # make a contingency table
        if (!is.null(numOverlaps)) {
          X = matrix(
            c(
              numOverlaps,
              length(gene_sets[[i]]) - numOverlaps,
              length(topic_genes) - numOverlaps,
              length(bkg_genes$V1) - (length(gene_sets[[i]]) - numOverlaps)
            ),
            byrow = TRUE,
            ncol = 2
          )
          lm <- fisher.test(X) 
          # summarize results
          return(c(
            gsub("_geneset.txt", "", names(gene_sets)[i]),
            paste(overlaps, collapse = ","),
            lm$p.value,
            lm$estimate
          ))
        }
      })
    enrichTest <- data.frame(t(enrichTest))
    colnames(enrichTest) <- c("GOTerm", "GenesInSet","P", "OR")
    return(enrichTest)
  }
```


```{r}
files <- c(
  list.files(paste0(workdir, "/gene_sets"),
             pattern = "CELL_DIFFERENTIATION"),
  list.files(paste0(workdir, "/gene_sets"), pattern =
               "T_HELPER_2_CELL_CYTOKINE_PRODUCTION"),
  list.files(paste0(workdir, "/gene_sets"), pattern =
               "IMMUNE_RESPONSE_geneset"),
  "Tissue_resident_lymphocytes_geneset.txt"
)

genesets <- list()
for(i in files){
  f <- read.table(sprintf("%s/gene_sets/%s", workdir, i))
  genes <- ncbi37[ncbi37$V1 %in% as.character(f[, -1]), ]$V6
  genesets[[i]] <- genes
}

genesets$GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION_geneset.txt <-
  c(genesets$GOBP_T_HELPER_2_CELL_CYTOKINE_PRODUCTION_geneset.txt,
    "IL25", "IL5", "IL6", "IL9", "IL13")


topic5 <- de_list[[5]]
totalGenes <- read.table("output/fastTopics/GO_background_genes.txt")
# replace TNFRSF4 with its ligand 
rownames(topic5) <- gsub("TNFRSF4", "TNFSF4", rownames(topic5))


enrich_out <- 
  test_topic_enrichment(
    topic_genes = rownames(topic5),
    gene_sets = genesets,
    bkg_genes = totalGenes
  )


DT::datatable(
  enrich_out %>% 
    mutate(P = round(as.numeric(P), 4),
           OR = round(as.numeric(OR), 4)) %>%
    arrange(P)
)

```
### Testing on genes having high Asthma MAGMA Z scores

Input: Topic 5 genes with MAGMA Z score >= 1.64, background genes for topic modeling (~17K)
Enrichment test: Fisher's exact test

Topic 5 specific genes with high Asthma z scores are enriched for tissue-resident genes, Treg/Th17/Th2 differentiation, and Th17 immune response.
```{r}
f_zscores <- fread(paste0(workdir, "/results/ukb-a-446/ukb-a-446_top100.gsa.genes.out"))
topic5_ids <- ncbi37$V1[ncbi37$V6 %in% rownames(topic5)]
topic5_topZ <-
  f_zscores %>% 
  filter(GENE %in% topic5_ids) %>%
  arrange(desc(ZSTAT))

topic5_topZ$SYMBOL <- 
  unlist(lapply(topic5_topZ$GENE, function(i){ncbi37$V6[ncbi37$V1 == i]}))

enrich_out <- 
  test_topic_enrichment(
    topic_genes = topic5_topZ$SYMBOL[topic5_topZ$ZSTAT >= 1.64],
    gene_sets = genesets,
    bkg_genes = totalGenes
  )


DT::datatable(
  enrich_out %>% 
    mutate(P = round(as.numeric(P), 4),
           OR = round(as.numeric(OR), 4)) %>%
    arrange(P)
)
```
### Testing on genes having high Allergy MAGMA Z scores

Input: Topic 5 genes with MAGMA Z score >= 1.64, background genes for topic modeling (~17K)
Enrichment test: Fisher's exact test

Topic 5 specific genes with high allergy z scores are enriched for T cell activation involved in immune response.

```{r}
f_zscores <- fread(paste0(workdir, "/results/ukb-a-447/ukb-a-447_top100.gsa.genes.out"))
topic5_ids <- ncbi37$V1[ncbi37$V6 %in% rownames(topic5)]
topic5_topZ <-
  f_zscores %>% 
  filter(GENE %in% topic5_ids) %>%
  arrange(desc(ZSTAT))

topic5_topZ$SYMBOL <- 
  unlist(lapply(topic5_topZ$GENE, function(i){ncbi37$V6[ncbi37$V1 == i]}))

enrich_out <- 
  test_topic_enrichment(
    topic_genes = topic5_topZ$SYMBOL[topic5_topZ$ZSTAT >= 1.64],
    gene_sets = genesets,
    bkg_genes = totalGenes
  )

DT::datatable(
  enrich_out %>% 
    mutate(P = round(as.numeric(P), 4),
           OR = round(as.numeric(OR), 4)) %>%
    arrange(P)
)
```


## Gene-set associations for known immune cell markers

Most immune cell markers were obtained from Gene ontology. The tissue-resident genes were from the single-cell lung paper and ChatGPT. After multiple correction (adj.P < 0.002), tissue-resident markers do not show significance for the gene-set association with Asthma. Neither do the genes involved in cell-cell adhesion or cell migration. The gene sets that are significantly enriched for higher association with Asthma risks are activation of T/B cells and T cell lineage differentiation.
```{bash eval = FALSE}
cd /home/jinggu/cluster/data/MAGMA/testrun
source ~/.bashrc
traitID="ukb-a-446"
geneset=gene_sets/GO_gene_sets_MAGMA.txt
magma \
    --gene-results $traitID.genes.raw \
    --set-annot ${geneset} \
    --out ${traitID}_GO_enriched

```

```{r}
df_enriched <- read.table(paste0(workdir, "/results/ukb-a-446/ukb-a-446_GO_enriched.gsa.out"),
                    skip = 4, header = TRUE)
tbl <-
  rbind(
    df_top %>%
      filter(NGENES > 10 & P < 0.05/12) %>% # correct for multiple testing
      select(VARIABLE, NGENES, BETA, SE, P),
    df_enriched %>% mutate(VARIABLE = FULL_NAME) %>%
      filter(NGENES > 10) %>% #& P < 0.05/33) %>% # correct for multiple testing 0.05/33 = 0.002
      select(VARIABLE, NGENES, BETA, SE, P) %>% 
      arrange(P) %>% mutate(P = format(P, scientific = TRUE, digits = 2))
  )
DT::datatable(tbl)
#write.table(tbl, "../output/fastTopics/top100_topic_GO_immune.gsa.txt", 
            #quote = F, row.names = F, sep = "\t")
```

```{r eval = FALSE}
### Perform permutation to obtain null p-values
ncbi37<-fread("/home/jinggu/cluster/data/MAGMA/gene_loc/u19.fastTopics.gene.loc")
B = 5000
for(i in 1:B){
  gene_set = sample(ncbi37$V1, 700, replace = FALSE)
  write.table(matrix(c(paste0("set", i), gene_set), nrow = 1),
              sprintf("%s/set%s_geneset.txt", workdir, i), 
              quote = F, row.names = F, sep=" ", col.names = F)
}
```

```{r eval = FALSE}
f_null <- read.table(paste0(workdir, "/ukb-a-446_permutated700.gsa.out"), 
                     skip = 3, header = T)

f_perm <- unlist(lapply(f_reduced$BETA, function(i){sum(f_null$BETA >= i)/5000}))
  
plot(y = -log10(f_reduced$P), 
     x = -log10(f_perm), 
     xlab = "-log10(P) after permutation",
     ylab = "-log10(P) before permutation")
abline(a = 0, b = 1, col = "red")
```

